{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Original Summary:\n",
    "Notebook to process the raw ATAC data and call peaks (first unfiltered with Homer, then reproducible with IDR). I'll then use the H3K27ac data to call positive and negative peaks. \n",
    "I will need some packages found in the deep lift conda env Zeyang had me make... I've added that env to my jupyter notebook, so just need to change the kernel to that one I think.\n",
    "\n",
    "Adjusted on 1/21/21 to generate peaks for both control and NASH treatment in this notebook!\n",
    "\n",
    "Output files:\n",
    "- strain_treat_annot.txt: original peak annotation file \n",
    "- strain_treat_filt_annot.txt: filtered peaks with H3K27ac > 16 and \"intron\" or \"Intergenic\"\n",
    "- strain_treat_filt_annot_cut.bed: bed file of ^ with chrY and chrUn peaks removed\n",
    "\n",
    "#### 1/31 Problem Solving: \n",
    "There was very little overlap (even with mergePeaks d=1) between the C57 control and NASH ATAC peaks which is super wrong. Normally we'd see 50-80% overlapping peaks so I'm going to figure out how this happened here. It looks like my raw idr peaks overlap well with Hunter's, so the issue is probably after the step to generate these. \n",
    "\n",
    "### PROBLEM WAS d =1 (now should be given)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "%matplotlib inline\n",
    "\n",
    "import csv\n",
    "import os\n",
    "\n",
    "from scipy.stats.stats import pearsonr,spearmanr\n",
    "\n",
    "from os import listdir\n",
    "import fnmatch"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make Genome Browser Hub"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Hunter changed these tag directories a bit so skipping this (BUT STILL DOESN'T WORK EITHER)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812 00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811 01_balbc_Kupffer_ATAC_control_young_balb10A_TDT_JSS_20161013 01_balbc_Kupffer_ATAC_control_young_balb10B_TDT_JSS_20161013 01_balbc_Kupffer_ATAC_control_young_balb10C_TDT_JSS_20161013 02_aj_Kupffer_ATAC_control_young_aj10A_TDT_JSS_20161013 02_aj_Kupffer_ATAC_control_young_aj10B_TDT_JSS_20161013 02_aj_Kupffer_ATAC_control_young_aj10C_TDT_JSS_20161013\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Colors that will be used (change with -color or -gradient options):\n",
      "\tIndex\tColor\tNeg. color\tTag Directory\n",
      "\t1\t255,150,150\t(255,180,180)\t00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812\n",
      "\t2\t150,150,255\t(180,180,255)\t00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811\n",
      "\t3\t150,255,150\t(180,255,180)\t01_balbc_Kupffer_ATAC_control_young_balb10A_TDT_JSS_20161013\n",
      "\t4\t255,200,150\t(255,210,180)\t01_balbc_Kupffer_ATAC_control_young_balb10B_TDT_JSS_20161013\n",
      "\t5\t150,255,220\t(180,255,240)\t01_balbc_Kupffer_ATAC_control_young_balb10C_TDT_JSS_20161013\n",
      "\t6\t200,150,255\t(210,180,255)\t02_aj_Kupffer_ATAC_control_young_aj10A_TDT_JSS_20161013\n",
      "\t7\t200,200,150\t(210,210,170)\t02_aj_Kupffer_ATAC_control_young_aj10B_TDT_JSS_20161013\n",
      "\t8\t150,200,200\t(170,210,210)\t02_aj_Kupffer_ATAC_control_young_aj10C_TDT_JSS_20161013\n",
      "\n",
      "\n",
      "\tOnce finished, you will want to upload the following hub URL:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-control/hub.txt\n",
      "\n",
      "\tIf loading to the Wash U Epigenome Browser, use:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-control/washU.hub.txt\n",
      "\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 131\n",
      "!!! Could not open autofile 00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812/00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812/00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 141\n",
      "!!! Could not open autofile 00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811/00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811/00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 50\n",
      "!!! Could not open autofile 01_balbc_Kupffer_ATAC_control_young_balb10A_TDT_JSS_20161013/01_balbc_Kupffer_ATAC_control_young_balb10A_TDT_JSS_20161013.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘01_balbc_Kupffer_ATAC_control_young_balb10A_TDT_JSS_20161013/01_balbc_Kupffer_ATAC_control_young_balb10A_TDT_JSS_20161013.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 71\n",
      "!!! Could not open autofile 01_balbc_Kupffer_ATAC_control_young_balb10B_TDT_JSS_20161013/01_balbc_Kupffer_ATAC_control_young_balb10B_TDT_JSS_20161013.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘01_balbc_Kupffer_ATAC_control_young_balb10B_TDT_JSS_20161013/01_balbc_Kupffer_ATAC_control_young_balb10B_TDT_JSS_20161013.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 51\n",
      "!!! Could not open autofile 01_balbc_Kupffer_ATAC_control_young_balb10C_TDT_JSS_20161013/01_balbc_Kupffer_ATAC_control_young_balb10C_TDT_JSS_20161013.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘01_balbc_Kupffer_ATAC_control_young_balb10C_TDT_JSS_20161013/01_balbc_Kupffer_ATAC_control_young_balb10C_TDT_JSS_20161013.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 57\n",
      "!!! Could not open autofile 02_aj_Kupffer_ATAC_control_young_aj10A_TDT_JSS_20161013/02_aj_Kupffer_ATAC_control_young_aj10A_TDT_JSS_20161013.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘02_aj_Kupffer_ATAC_control_young_aj10A_TDT_JSS_20161013/02_aj_Kupffer_ATAC_control_young_aj10A_TDT_JSS_20161013.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 56\n",
      "!!! Could not open autofile 02_aj_Kupffer_ATAC_control_young_aj10B_TDT_JSS_20161013/02_aj_Kupffer_ATAC_control_young_aj10B_TDT_JSS_20161013.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘02_aj_Kupffer_ATAC_control_young_aj10B_TDT_JSS_20161013/02_aj_Kupffer_ATAC_control_young_aj10B_TDT_JSS_20161013.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 54\n",
      "!!! Could not open autofile 02_aj_Kupffer_ATAC_control_young_aj10C_TDT_JSS_20161013/02_aj_Kupffer_ATAC_control_young_aj10C_TDT_JSS_20161013.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘02_aj_Kupffer_ATAC_control_young_aj10C_TDT_JSS_20161013/02_aj_Kupffer_ATAC_control_young_aj10C_TDT_JSS_20161013.ucsc.bigWig’: No such file or directory\n",
      "\n",
      "\tAll finished: you will want to upload the following hub URL:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-control/hub.txt\n",
      "\n",
      "\tIf loading to the Wash U Epigenome Browser, use:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-control/washU.hub.txt\n",
      "\n",
      "\n",
      "\tTo edit track settings, edit files in:\n",
      "\t\t/homer_data/www/html/hubs//C57Kupffer-ATAC-control\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/h1bennet/strains/data/ATAC/control/\n",
    "tag_dir=$(ls -d *)\n",
    "echo $tag_dir\n",
    "makeMultiWigHub.pl C57Kupffer-ATAC-control2 mm10 -url http://homer.ucsd.edu/hubs/ -webdir /homer_data/www/html/hubs/ -d $tag_dir -force #&>/dev/null &\n",
    "\n",
    "#normally get a link from this (then put that into the UCSC genome browser)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919 00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921 01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3C_JSS_TDT_160926 01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3D_JSS_TDT_160928 02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3A_JSS_TDT_160926 02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3B_JSS_TDT_160926\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Colors that will be used (change with -color or -gradient options):\n",
      "\tIndex\tColor\tNeg. color\tTag Directory\n",
      "\t1\t255,150,150\t(255,180,180)\t00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919\n",
      "\t2\t150,150,255\t(180,180,255)\t00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921\n",
      "\t3\t150,255,150\t(180,255,180)\t01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3C_JSS_TDT_160926\n",
      "\t4\t255,200,150\t(255,210,180)\t01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3D_JSS_TDT_160928\n",
      "\t5\t150,255,220\t(180,255,240)\t02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3A_JSS_TDT_160926\n",
      "\t6\t200,150,255\t(210,180,255)\t02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3B_JSS_TDT_160926\n",
      "\n",
      "\n",
      "\tOnce finished, you will want to upload the following hub URL:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-AMLN30wk/hub.txt\n",
      "\n",
      "\tIf loading to the Wash U Epigenome Browser, use:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-AMLN30wk/washU.hub.txt\n",
      "\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 59\n",
      "!!! Could not open autofile 00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919/00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919/00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 58\n",
      "!!! Could not open autofile 00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921/00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921/00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 57\n",
      "!!! Could not open autofile 01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3C_JSS_TDT_160926/01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3C_JSS_TDT_160926.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3C_JSS_TDT_160926/01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3C_JSS_TDT_160926.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 56\n",
      "!!! Could not open autofile 01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3D_JSS_TDT_160928/01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3D_JSS_TDT_160928.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3D_JSS_TDT_160928/01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3D_JSS_TDT_160928.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 56\n",
      "!!! Could not open autofile 02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3A_JSS_TDT_160926/02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3A_JSS_TDT_160926.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3A_JSS_TDT_160926/02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3A_JSS_TDT_160926.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 56\n",
      "!!! Could not open autofile 02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3B_JSS_TDT_160926/02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3B_JSS_TDT_160926.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3B_JSS_TDT_160926/02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3B_JSS_TDT_160926.ucsc.bigWig’: No such file or directory\n",
      "\n",
      "\tAll finished: you will want to upload the following hub URL:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-AMLN30wk/hub.txt\n",
      "\n",
      "\tIf loading to the Wash U Epigenome Browser, use:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-AMLN30wk/washU.hub.txt\n",
      "\n",
      "\n",
      "\tTo edit track settings, edit files in:\n",
      "\t\t/homer_data/www/html/hubs//C57Kupffer-ATAC-AMLN30wk\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/h1bennet/strains/data/ATAC/AMLN_30week/\n",
    "tag_dir=$(ls -d *)\n",
    "echo $tag_dir\n",
    "makeMultiWigHub.pl C57Kupffer-ATAC-AMLN30wk mm10 -url http://homer.ucsd.edu/hubs/ -webdir /homer_data/www/html/hubs/ -d $tag_dir -force #&>/dev/null &"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Call Peaks"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 1: Unfiltered peaks from HOMER"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Control"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "File names:\n",
    "- 00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812\n",
    "- 00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812_round2\n",
      "00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811_round2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/hmummey/data/ATAC/raw_peaks/Control\n",
    "for data in ./*NCoRWT*; do\n",
    "    strain=\"$(cut -d '_' -f2 <<<\"$data\")\"\n",
    "    treatment=\"$(cut -d '_' -f5 <<<\"$data\")\"    \n",
    "    name=\"$(cut -d '/' -f2 <<<\"$data\")_round2\"\n",
    "    echo $name\n",
    "    \n",
    "    findPeaks $data -style factor -L 0 -C 0 -fdr 0.9 -o /home/hmummey/data/ATAC/init_peak_calling/${name}.txt -size 200 &>/dev/null &\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812',\n",
       "       '00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811'],\n",
       "      dtype='<U54')"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tag_diretory_array = np.array(fnmatch.filter(listdir('/home/hmummey/data/ATAC/raw_peaks/Control'), '*NCoRWT*'))\n",
    "tag_diretory_array.sort()\n",
    "tag_diretory_array"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### AMLN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919_round2\n",
      "00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921_round2\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/hmummey/data/ATAC/raw_peaks/AMLN_30week\n",
    "for data in ./*NCoRWT*; do\n",
    "    strain=\"$(cut -d '_' -f2 <<<\"$data\")\"\n",
    "    treatment=\"$(cut -d '_' -f5 <<<\"$data\")\"    \n",
    "    name=\"$(cut -d '/' -f2 <<<\"$data\")_round2\"\n",
    "    echo $name\n",
    "    findPeaks $data -style factor -L 0 -C 0 -fdr 0.9 -o /home/hmummey/data/ATAC/init_peak_calling/${name}.txt -size 200 &>/dev/null &\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919',\n",
       "       '00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921'],\n",
       "      dtype='<U60')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tag_diretory_array2 = np.array(fnmatch.filter(listdir('/home/hmummey/data/ATAC/raw_peaks/AMLN_30week'), '*NCoRWT*'))\n",
    "tag_diretory_array2.sort()\n",
    "tag_diretory_array2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Creating a file with all chip_names (control and AMLN)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['NCoRWT_AMLNDiet', 'NCoRWT_control'], dtype='<U15')"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chip_names = []\n",
    "for i in tag_diretory_array:\n",
    "    strain = i.split('_')[1]\n",
    "    treat = i.split('_')[4]\n",
    "    chip_names.append('_'.join([strain, treat]))\n",
    "    \n",
    "for i in tag_diretory_array2:\n",
    "    strain = i.split('_')[1]\n",
    "    treat = i.split('_')[4]\n",
    "    chip_names.append('_'.join([strain, treat]))\n",
    "    \n",
    "chip_names = np.unique(np.array(chip_names))\n",
    "chip_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "chip_file = '/home/hmummey/data/ATAC/chip_names_list_round2.txt' #exactly the same as prev file\n",
    "with open(chip_file, 'w') as f:\n",
    "    for c in chip_names:\n",
    "        f.write(c + '\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 2: Reproducible peaks from IDR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NCoRWT_AMLNDiet\n",
      "./00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919_round2.txt ./00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921_round2.txt\n",
      "\n",
      "NCoRWT_control\n",
      "./00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812_round2.txt ./00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811_round2.txt\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /home/hmummey/data/ATAC/init_peak_calling/\n",
    "while IFS='' read -r IP\n",
    "do\n",
    "    strain=\"$(cut -d '_' -f1 <<<\"$IP\")\"\n",
    "    treatment=\"$(cut -d '_' -f2 <<<\"$IP\")\"\n",
    "    round=\"round2\"\n",
    "    #name=\"$(cut -d '/' -f2 <<<\"$data\" | cut -d '.' -f1)\"\n",
    "\n",
    "    echo $IP\n",
    "    echo ./*${strain}*${treatment}*${round}*\n",
    "    #echo *${strain}*${treatment}*\n",
    "    echo \n",
    "\n",
    "    #mkdir ../IDR/${IP}\n",
    "    \n",
    "    #python3 /home/zes017/run_idr_homerPeaks.py -threshold 0.05 ./*${strain}*${treatment}*${round}* ../IDR/${IP}*${round} #&>/dev/null &\n",
    "done < '/home/hmummey/data/ATAC/chip_names_list_round2.txt'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Count Tags of Histone Marks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/hmummey/data/ATAC/annotated_peak_calling/00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919_round2_00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921_round2_idr.tsv\n",
      "C57\n",
      "AMLNDiet\n",
      "AMLN\n",
      "/home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423\n",
      "\n",
      "/home/hmummey/data/ATAC/annotated_peak_calling/00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812_round2_00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811_round2_idr.tsv\n",
      "C57\n",
      "control\n",
      "cont\n",
      "/home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\tPeak file = /home/hmummey/data/ATAC/annotated_peak_calling/00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919_round2_00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921_round2_idr.tsv\n",
      "\tGenome = mm10\n",
      "\tOrganism = mouse\n",
      "\tPeak Region set to 1000\n",
      "\tWill normalize tag counts to 1e7 per experiment\n",
      "\tTag Directories:\n",
      "\t\t/home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423\n",
      "\t\t/home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 0\n",
      "\t\tpeakfile formatted lines: 46223\n",
      "\t\tDuplicated Peak IDs: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 46223\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tResizing peaks...\n",
      "\tReading Positions...\n",
      "\t-----------------------\n",
      "\tFinding Closest TSS...\n",
      "\tAnnotating:........................\n",
      "\t\tAnnotation\tNumber of peaks\tTotal size (bp)\tLog2 Ratio (obs/exp)\tLogP enrichment (+values depleted)\n",
      "\t\t3UTR\t478.0\t20049228\t0.490\t-27.641\n",
      "\t\tmiRNA\t2.0\t24864\t2.244\t-2.694\n",
      "\t\tncRNA\t155.0\t3372538\t1.436\t-59.688\n",
      "\t\tTTS\t761.0\t27011103\t0.730\t-87.291\n",
      "\t\tpseudo\t6.0\t528823\t-0.582\t1.567\n",
      "\t\tExon\t838.0\t34014877\t0.537\t-55.326\n",
      "\t\tIntron\t17742.0\t942718221\t0.148\t-145.129\n",
      "\t\tIntergenic\t14964.0\t1662058715\t-0.915\t7764.186\n",
      "\t\tPromoter\t10843.0\t29761709\t4.423\t-24180.779\n",
      "\t\t5UTR\t426.0\t2270504\t3.466\t-641.441\n",
      "\t\tsnoRNA\t0.0\t19\t-0.000\t0.000\n",
      "\t\trRNA\t3.0\t5631\t4.972\t-8.906\n",
      "\tNOTE: If this part takes more than 2 minutes, there is a good chance\n",
      "\t\tyour machine ran out of memory: consider hitting ctrl+C and rerunning\n",
      "\t\tthe command with \"-noann\"\n",
      "\tTo capture annotation stats in a file, use \"-annStats <filename>\" next time\n",
      "\tAnnotating:............................\n",
      "\t\tAnnotation\tNumber of peaks\tTotal size (bp)\tLog2 Ratio (obs/exp)\tLogP enrichment (+values depleted)\n",
      "\t\t3UTR\t478.0\t20049228\t0.490\t-27.695\n",
      "\t\tOther\t58.0\t7955052\t-1.219\t30.480\n",
      "\t\tRC?\t0.0\t10979\t-0.247\t0.186\n",
      "\t\tRNA\t3.0\t114021\t0.632\t-1.185\n",
      "\t\tmiRNA\t2.0\t24864\t2.245\t-2.695\n",
      "\t\tncRNA\t155.0\t3372538\t1.437\t-59.726\n",
      "\t\tTTS\t761.0\t27011103\t0.731\t-87.409\n",
      "\t\tLINE\t1255.0\t543776315\t-2.879\t6279.064\n",
      "\t\tLINE?\t0.0\t8168\t-0.187\t0.139\n",
      "\t\tsrpRNA\t2.0\t43307\t1.444\t-1.783\n",
      "\t\tSINE\t1464.0\t196566416\t-1.188\t710.843\n",
      "\t\tRC\t0.0\t65629\t-1.080\t1.114\n",
      "\t\ttRNA\t9.0\t266576\t0.992\t-3.184\n",
      "\t\tDNA?\t0.0\t142206\t-1.771\t2.414\n",
      "\t\tpseudo\t6.0\t528823\t-0.581\t1.566\n",
      "\t\tDNA\t262.0\t28691693\t-0.894\t66.079\n",
      "\t\tExon\t838.0\t34014877\t0.537\t-55.428\n",
      "\t\tIntron\t14038.0\t598538808\t0.466\t-881.695\n",
      "\t\tIntergenic\t11163.0\t828470746\t-0.333\t452.966\n",
      "\t\tPromoter\t10843.0\t29761709\t4.424\t-24184.659\n",
      "\t\t5UTR\t426.0\t2270504\t3.466\t-641.590\n",
      "\t\tsnoRNA\t0.0\t19\t-0.000\t0.000\n",
      "\t\tLTR?\t4.0\t192563\t0.291\t-0.885\n",
      "\t\tscRNA\t2.0\t602998\t-2.355\t6.083\n",
      "\t\tCpG-Island\t1354.0\t3194561\t4.642\t-3079.909\n",
      "\t\tLow_complexity\t81.0\t19402996\t-2.024\t138.239\n",
      "\t\tLTR\t2834.0\t313078376\t-0.907\t776.356\n",
      "\t\tSimple_repeat\t115.0\t57560484\t-3.087\t627.309\n",
      "\t\tsnRNA\t5.0\t237524\t0.311\t-0.975\n",
      "\t\tUnknown\t8.0\t2407459\t-2.353\t21.588\n",
      "\t\tSINE?\t0.0\t29758\t-0.590\t0.505\n",
      "\t\tSatellite\t36.0\t4603878\t-1.118\t16.378\n",
      "\t\trRNA\t21.0\t166488\t2.894\t-26.257\n",
      "\tCounting Tags in Peaks from each directory...\n",
      "\tRatio for /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 : 0.53334107033446\n",
      "\tRatio for /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423 : 0.519990812802319\n",
      "\tOrganism: mouse\n",
      "\tLoading Gene Informaiton...\n",
      "\tOutputing Annotation File...\n",
      "\tDone annotating peaks file\n",
      "\n",
      "\n",
      "\tPeak file = /home/hmummey/data/ATAC/annotated_peak_calling/00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812_round2_00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811_round2_idr.tsv\n",
      "\tGenome = mm10\n",
      "\tOrganism = mouse\n",
      "\tPeak Region set to 1000\n",
      "\tWill normalize tag counts to 1e7 per experiment\n",
      "\tTag Directories:\n",
      "\t\t/home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915\n",
      "\t\t/home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 0\n",
      "\t\tpeakfile formatted lines: 48664\n",
      "\t\tDuplicated Peak IDs: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 48664\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tResizing peaks...\n",
      "\tReading Positions...\n",
      "\t-----------------------\n",
      "\tFinding Closest TSS...\n",
      "\tAnnotating:........................\n",
      "\t\tAnnotation\tNumber of peaks\tTotal size (bp)\tLog2 Ratio (obs/exp)\tLogP enrichment (+values depleted)\n",
      "\t\t3UTR\t527.0\t20049228\t0.556\t-37.818\n",
      "\t\tmiRNA\t0.0\t24864\t-0.531\t0.444\n",
      "\t\tncRNA\t137.0\t3372538\t1.184\t-38.620\n",
      "\t\tTTS\t744.0\t27011103\t0.624\t-64.421\n",
      "\t\tpseudo\t10.0\t528823\t0.081\t-0.751\n",
      "\t\tExon\t936.0\t34014877\t0.622\t-80.245\n",
      "\t\tIntron\t19387.0\t942718221\t0.202\t-289.845\n",
      "\t\tIntergenic\t16404.0\t1662058715\t-0.857\t7419.080\n",
      "\t\tPromoter\t10043.0\t29761709\t4.239\t-21009.656\n",
      "\t\t5UTR\t462.0\t2270504\t3.509\t-707.970\n",
      "\t\tsnoRNA\t0.0\t19\t-0.000\t0.000\n",
      "\t\trRNA\t2.0\t5631\t4.313\t-5.352\n",
      "\tNOTE: If this part takes more than 2 minutes, there is a good chance\n",
      "\t\tyour machine ran out of memory: consider hitting ctrl+C and rerunning\n",
      "\t\tthe command with \"-noann\"\n",
      "\tTo capture annotation stats in a file, use \"-annStats <filename>\" next time\n",
      "\tAnnotating:................................\n",
      "\t\tAnnotation\tNumber of peaks\tTotal size (bp)\tLog2 Ratio (obs/exp)\tLogP enrichment (+values depleted)\n",
      "\t\t3UTR\t527.0\t20049228\t0.557\t-37.869\n",
      "\t\tOther\t41.0\t7955052\t-1.794\t52.724\n",
      "\t\tRC?\t1.0\t10979\t2.350\t-1.725\n",
      "\t\tRNA\t1.0\t114021\t-1.027\t0.926\n",
      "\t\tmiRNA\t0.0\t24864\t-0.530\t0.444\n",
      "\t\tncRNA\t137.0\t3372538\t1.185\t-38.643\n",
      "\t\tTTS\t744.0\t27011103\t0.624\t-64.499\n",
      "\t\tLINE\t1757.0\t543786595\t-2.467\t5722.826\n",
      "\t\tLINE?\t0.0\t8168\t-0.197\t0.146\n",
      "\t\tsrpRNA\t1.0\t43307\t0.370\t-0.618\n",
      "\t\tSINE\t1485.0\t196568847\t-1.242\t797.996\n",
      "\t\tRC\t1.0\t65629\t-0.230\t0.397\n",
      "\t\ttRNA\t9.0\t266845\t0.916\t-2.919\n",
      "\t\tDNA?\t0.0\t142206\t-1.824\t2.541\n",
      "\t\tpseudo\t10.0\t528823\t0.082\t-0.752\n",
      "\t\tDNA\t426.0\t28691693\t-0.267\t10.085\n",
      "\t\tExon\t936.0\t34014877\t0.623\t-80.343\n",
      "\t\tIntron\t15230.0\t598538808\t0.510\t-1135.219\n",
      "\t\tIntergenic\t11971.0\t828534192\t-0.307\t410.103\n",
      "\t\tPromoter\t10042.0\t29761709\t4.239\t-21009.068\n",
      "\t\t5UTR\t462.0\t2270504\t3.509\t-708.095\n",
      "\t\tsnoRNA\t0.0\t19\t-0.000\t0.000\n",
      "\t\tLTR?\t4.0\t192563\t0.217\t-0.797\n",
      "\t\tscRNA\t3.0\t602998\t-1.845\t5.147\n",
      "\t\tCpG-Island\t1044.0\t3194146\t4.193\t-2061.773\n",
      "\t\tLow_complexity\t126.0\t19407316\t-1.461\t96.623\n",
      "\t\tLTR\t3389.0\t313097892\t-0.723\t566.572\n",
      "\t\tSimple_repeat\t213.0\t57569025\t-2.272\t490.618\n",
      "\t\tsnRNA\t10.0\t237524\t1.236\t-4.432\n",
      "\t\tUnknown\t12.0\t2407459\t-1.842\t17.562\n",
      "\t\tSINE?\t0.0\t29758\t-0.615\t0.532\n",
      "\t\tSatellite\t60.0\t4626809\t-0.463\t5.211\n",
      "\t\trRNA\t22.0\t166488\t2.887\t-27.326\n",
      "\tCounting Tags in Peaks from each directory...\n",
      "\tRatio for /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 : 1.05618536475626\n",
      "\tRatio for /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423 : 0.442167204732569\n",
      "\tOrganism: mouse\n",
      "\tLoading Gene Informaiton...\n",
      "\tOutputing Annotation File...\n",
      "\tDone annotating peaks file\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#merged file combines peak files from all the strains (for one condition)\n",
    "#alt: look at induced peaks by merging from one strain (both conditions)\n",
    "#HERE WE AREN'T USING MERGED FILES, JUST USE THE PREVIOUSLY MADE .TSV FILES\n",
    "\n",
    "for peak_file in /home/hmummey/data/ATAC/annotated_peak_calling/*_idr.tsv; do\n",
    "    echo $peak_file\n",
    "\n",
    "    #strain=\"$(echo $peak_file | cut -d '/' -f7 | cut -d '_' -f2)\"\n",
    "    strain=\"C57\"\n",
    "    treat=\"$(echo $peak_file | cut -d '/' -f7 | cut -d '_' -f5)\"\n",
    "    \n",
    "    #treat (AMLNDiet) doesn't match his dir name: /home/h1bennet/strains/data/H3K27Ac/AMLN_30week\n",
    "    #but it does match the file names: 00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423\n",
    "    #so modifying this a bit (cut down and then use * later)\n",
    "    treat2=\"$(echo $treat | cut -b 1-4)\"\n",
    "\n",
    "    echo $strain\n",
    "    echo $treat\n",
    "    echo $treat2\n",
    "    \n",
    "    files=''\n",
    "    for h in 'H3K27Ac'; do\n",
    "        files+=\"$(echo /home/h1bennet/strains/data/$h/$treat2*/*${strain}*${h}*)\"\n",
    "\n",
    "        files+=' '\n",
    "    done\n",
    "    echo $files\n",
    "    echo\n",
    "    annotatePeaks.pl $peak_file mm10 -size -500,500 -norm 1e7 -d $files > /home/hmummey/data/ATAC/annotated_peak_calling/${strain}_${treat}_round2_annot.txt #2>/dev/null &\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "strain = 'C57_control_round2'\n",
    "path = '/home/hmummey/data/ATAC/annotated_peak_calling/'\n",
    "annot_file = path + strain + '_annot.txt'\n",
    "annot_df = pd.read_csv(annot_file, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "48664\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812_round2_00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811_round2_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423)</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "      <th>Focus Ratio/Region Size</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Detailed Annotation</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>...</th>\n",
       "      <th>Entrez ID</th>\n",
       "      <th>Nearest Unigene</th>\n",
       "      <th>Nearest Refseq</th>\n",
       "      <th>Nearest Ensembl</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Alias</th>\n",
       "      <th>Gene Description</th>\n",
       "      <th>Gene Type</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 Tag Count in 1000 bp (9468035.0 Total, normalization factor = 1.06, effective total = 1e7)</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423 Tag Count in 1000 bp (22615879.0 Total, normalization factor = 0.44, effective total = 1e7)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>65_35545</td>\n",
       "      <td>chr8</td>\n",
       "      <td>71633967</td>\n",
       "      <td>71634967</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>28.0</td>\n",
       "      <td>intron (NM_001029873, intron 39 of 41)</td>\n",
       "      <td>intron (NM_001029873, intron 39 of 41)</td>\n",
       "      <td>-3361.0</td>\n",
       "      <td>...</td>\n",
       "      <td>102466799.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NR_106035</td>\n",
       "      <td>ENSMUSG00000098976</td>\n",
       "      <td>Mir6769b</td>\n",
       "      <td>Gm27546|mmu-mir-6769b</td>\n",
       "      <td>microRNA 6769b</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>6.34</td>\n",
       "      <td>3.54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>65_22643</td>\n",
       "      <td>chr5</td>\n",
       "      <td>113991974</td>\n",
       "      <td>113992974</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>43.0</td>\n",
       "      <td>intron (NM_198109, intron 1 of 14)</td>\n",
       "      <td>L1MB8|LINE|L1</td>\n",
       "      <td>1283.0</td>\n",
       "      <td>...</td>\n",
       "      <td>231637.0</td>\n",
       "      <td>Mm.389682</td>\n",
       "      <td>NM_198109</td>\n",
       "      <td>ENSMUSG00000042121</td>\n",
       "      <td>Ssh1</td>\n",
       "      <td>AW551225|Gm1394|Gm1395|SSH-1|SSH-1L</td>\n",
       "      <td>slingshot protein phosphatase 1</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>67.60</td>\n",
       "      <td>63.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>65_14830</td>\n",
       "      <td>chr9</td>\n",
       "      <td>57765476</td>\n",
       "      <td>57766476</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>59.0</td>\n",
       "      <td>promoter-TSS (NM_007713)</td>\n",
       "      <td>promoter-TSS (NM_007713)</td>\n",
       "      <td>-116.0</td>\n",
       "      <td>...</td>\n",
       "      <td>102414.0</td>\n",
       "      <td>Mm.25720</td>\n",
       "      <td>NM_007713</td>\n",
       "      <td>ENSMUSG00000032316</td>\n",
       "      <td>Clk3</td>\n",
       "      <td>AI256811</td>\n",
       "      <td>CDC-like kinase 3</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>260.88</td>\n",
       "      <td>279.01</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>65_28521</td>\n",
       "      <td>chr7</td>\n",
       "      <td>97154713</td>\n",
       "      <td>97155713</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>35.0</td>\n",
       "      <td>intron (NM_001162477, intron 1 of 9)</td>\n",
       "      <td>intron (NM_001162477, intron 1 of 9)</td>\n",
       "      <td>73462.0</td>\n",
       "      <td>...</td>\n",
       "      <td>14389.0</td>\n",
       "      <td>Mm.42033</td>\n",
       "      <td>NM_010248</td>\n",
       "      <td>ENSMUSG00000004508</td>\n",
       "      <td>Gab2</td>\n",
       "      <td>AI463667|D130058I17Rik|p97</td>\n",
       "      <td>growth factor receptor bound protein 2-associa...</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>46.47</td>\n",
       "      <td>48.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>65_16538</td>\n",
       "      <td>chr10</td>\n",
       "      <td>95501369</td>\n",
       "      <td>95502369</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>54.0</td>\n",
       "      <td>promoter-TSS (NM_001359477)</td>\n",
       "      <td>promoter-TSS (NM_001359477)</td>\n",
       "      <td>80.0</td>\n",
       "      <td>...</td>\n",
       "      <td>67270.0</td>\n",
       "      <td>Mm.30008</td>\n",
       "      <td>NM_026065</td>\n",
       "      <td>ENSMUSG00000062981</td>\n",
       "      <td>Mrpl42</td>\n",
       "      <td>2700009F22Rik|2900055D03Rik|D10Ertd322e|HSPC20...</td>\n",
       "      <td>mitochondrial ribosomal protein L42</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>106.67</td>\n",
       "      <td>99.05</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812_round2_00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811_round2_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423)  \\\n",
       "0                                           65_35545                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "1                                           65_22643                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "2                                           65_14830                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "3                                           65_28521                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "4                                           65_16538                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "\n",
       "     Chr      Start        End Strand  Peak Score  Focus Ratio/Region Size  \\\n",
       "0   chr8   71633967   71634967      +        1000                     28.0   \n",
       "1   chr5  113991974  113992974      +        1000                     43.0   \n",
       "2   chr9   57765476   57766476      +        1000                     59.0   \n",
       "3   chr7   97154713   97155713      +        1000                     35.0   \n",
       "4  chr10   95501369   95502369      +        1000                     54.0   \n",
       "\n",
       "                               Annotation  \\\n",
       "0  intron (NM_001029873, intron 39 of 41)   \n",
       "1      intron (NM_198109, intron 1 of 14)   \n",
       "2                promoter-TSS (NM_007713)   \n",
       "3    intron (NM_001162477, intron 1 of 9)   \n",
       "4             promoter-TSS (NM_001359477)   \n",
       "\n",
       "                      Detailed Annotation  Distance to TSS  ...    Entrez ID  \\\n",
       "0  intron (NM_001029873, intron 39 of 41)          -3361.0  ...  102466799.0   \n",
       "1                           L1MB8|LINE|L1           1283.0  ...     231637.0   \n",
       "2                promoter-TSS (NM_007713)           -116.0  ...     102414.0   \n",
       "3    intron (NM_001162477, intron 1 of 9)          73462.0  ...      14389.0   \n",
       "4             promoter-TSS (NM_001359477)             80.0  ...      67270.0   \n",
       "\n",
       "   Nearest Unigene Nearest Refseq     Nearest Ensembl Gene Name  \\\n",
       "0              NaN      NR_106035  ENSMUSG00000098976  Mir6769b   \n",
       "1        Mm.389682      NM_198109  ENSMUSG00000042121      Ssh1   \n",
       "2         Mm.25720      NM_007713  ENSMUSG00000032316      Clk3   \n",
       "3         Mm.42033      NM_010248  ENSMUSG00000004508      Gab2   \n",
       "4         Mm.30008      NM_026065  ENSMUSG00000062981    Mrpl42   \n",
       "\n",
       "                                          Gene Alias  \\\n",
       "0                              Gm27546|mmu-mir-6769b   \n",
       "1                AW551225|Gm1394|Gm1395|SSH-1|SSH-1L   \n",
       "2                                           AI256811   \n",
       "3                         AI463667|D130058I17Rik|p97   \n",
       "4  2700009F22Rik|2900055D03Rik|D10Ertd322e|HSPC20...   \n",
       "\n",
       "                                    Gene Description       Gene Type  \\\n",
       "0                                     microRNA 6769b           ncRNA   \n",
       "1                    slingshot protein phosphatase 1  protein-coding   \n",
       "2                                  CDC-like kinase 3  protein-coding   \n",
       "3  growth factor receptor bound protein 2-associa...  protein-coding   \n",
       "4                mitochondrial ribosomal protein L42  protein-coding   \n",
       "\n",
       "  /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 Tag Count in 1000 bp (9468035.0 Total, normalization factor = 1.06, effective total = 1e7)  \\\n",
       "0                                               6.34                                                                                                                                         \n",
       "1                                              67.60                                                                                                                                         \n",
       "2                                             260.88                                                                                                                                         \n",
       "3                                              46.47                                                                                                                                         \n",
       "4                                             106.67                                                                                                                                         \n",
       "\n",
       "   /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423 Tag Count in 1000 bp (22615879.0 Total, normalization factor = 0.44, effective total = 1e7)  \n",
       "0                                               3.54                                                                                                                                          \n",
       "1                                              63.23                                                                                                                                          \n",
       "2                                             279.01                                                                                                                                          \n",
       "3                                              48.20                                                                                                                                          \n",
       "4                                              99.05                                                                                                                                          \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(annot_df))\n",
    "annot_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_control_young_C571A_170915</th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_control_young_C572A_180423</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2.875780</td>\n",
       "      <td>2.182692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6.100137</td>\n",
       "      <td>6.005175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8.032762</td>\n",
       "      <td>8.129335</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5.568944</td>\n",
       "      <td>5.620586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6.750473</td>\n",
       "      <td>6.644577</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   00_C57_Kupffer_H3K27Ac_control_young_C571A_170915  \\\n",
       "0                                           2.875780   \n",
       "1                                           6.100137   \n",
       "2                                           8.032762   \n",
       "3                                           5.568944   \n",
       "4                                           6.750473   \n",
       "\n",
       "   00_C57_Kupffer_H3K27Ac_control_young_C572A_180423  \n",
       "0                                           2.182692  \n",
       "1                                           6.005175  \n",
       "2                                           8.129335  \n",
       "3                                           5.620586  \n",
       "4                                           6.644577  "
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "histone_tags1 = annot_df.iloc[:,-2:]\n",
    "histone_tags1.columns = [i.split(' Tag')[0].split('/')[-1] for i in histone_tags1.columns.values]\n",
    "histone_tags2 = np.log2(histone_tags1+1) #convert to log2 scale\n",
    "histone_tags2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_control_young_C571A_170915</th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_control_young_C572A_180423</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>48664.000000</td>\n",
       "      <td>48664.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>5.149920</td>\n",
       "      <td>4.984465</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>1.923150</td>\n",
       "      <td>2.074316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>3.657640</td>\n",
       "      <td>3.363171</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>5.324811</td>\n",
       "      <td>5.131343</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>6.707497</td>\n",
       "      <td>6.676098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>15.399944</td>\n",
       "      <td>14.987281</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       00_C57_Kupffer_H3K27Ac_control_young_C571A_170915  \\\n",
       "count                                       48664.000000   \n",
       "mean                                            5.149920   \n",
       "std                                             1.923150   \n",
       "min                                             0.000000   \n",
       "25%                                             3.657640   \n",
       "50%                                             5.324811   \n",
       "75%                                             6.707497   \n",
       "max                                            15.399944   \n",
       "\n",
       "       00_C57_Kupffer_H3K27Ac_control_young_C572A_180423  \n",
       "count                                       48664.000000  \n",
       "mean                                            4.984465  \n",
       "std                                             2.074316  \n",
       "min                                             0.000000  \n",
       "25%                                             3.363171  \n",
       "50%                                             5.131343  \n",
       "75%                                             6.676098  \n",
       "max                                            14.987281  "
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "histone_tags2.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### AMLN Diet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "46223\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919_round2_00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921_round2_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423)</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "      <th>Focus Ratio/Region Size</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Detailed Annotation</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>...</th>\n",
       "      <th>Entrez ID</th>\n",
       "      <th>Nearest Unigene</th>\n",
       "      <th>Nearest Refseq</th>\n",
       "      <th>Nearest Ensembl</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Alias</th>\n",
       "      <th>Gene Description</th>\n",
       "      <th>Gene Type</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 Tag Count in 1000 bp (18749728.0 Total, normalization factor = 0.53, effective total = 1e7)</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423 Tag Count in 1000 bp (19231109.0 Total, normalization factor = 0.52, effective total = 1e7)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>52_6108</td>\n",
       "      <td>chr19</td>\n",
       "      <td>6869077</td>\n",
       "      <td>6870077</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>88.0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>RLTR19-int|LTR|ERVK</td>\n",
       "      <td>-11366.0</td>\n",
       "      <td>...</td>\n",
       "      <td>78317.0</td>\n",
       "      <td>Mm.329596</td>\n",
       "      <td>NM_198008</td>\n",
       "      <td>ENSMUSG00000047810</td>\n",
       "      <td>Ccdc88b</td>\n",
       "      <td>2610041P08Rik|Ccdc88</td>\n",
       "      <td>coiled-coil domain containing 88B</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>102.40</td>\n",
       "      <td>111.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>52_27549</td>\n",
       "      <td>chr7</td>\n",
       "      <td>128136635</td>\n",
       "      <td>128137635</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>32.0</td>\n",
       "      <td>intron (NM_021334, intron 14 of 29)</td>\n",
       "      <td>intron (NM_021334, intron 14 of 29)</td>\n",
       "      <td>7567.0</td>\n",
       "      <td>...</td>\n",
       "      <td>16411.0</td>\n",
       "      <td>Mm.22378</td>\n",
       "      <td>NM_021334</td>\n",
       "      <td>ENSMUSG00000030789</td>\n",
       "      <td>Itgax</td>\n",
       "      <td>AI449405|Cd11c|Cr4|N418</td>\n",
       "      <td>integrin alpha X</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>51.73</td>\n",
       "      <td>24.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>52_532</td>\n",
       "      <td>chr11</td>\n",
       "      <td>20200837</td>\n",
       "      <td>20201837</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>178.0</td>\n",
       "      <td>promoter-TSS (NM_008996)</td>\n",
       "      <td>promoter-TSS (NM_008996)</td>\n",
       "      <td>-75.0</td>\n",
       "      <td>...</td>\n",
       "      <td>19324.0</td>\n",
       "      <td>Mm.271944</td>\n",
       "      <td>NM_008996</td>\n",
       "      <td>ENSMUSG00000020149</td>\n",
       "      <td>Rab1a</td>\n",
       "      <td>Gtbp|Rab-1|Rab1|Ypt1|mKIAA3012</td>\n",
       "      <td>RAB1A, member RAS oncogene family</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>450.67</td>\n",
       "      <td>384.79</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>52_21363</td>\n",
       "      <td>chr9</td>\n",
       "      <td>64736183</td>\n",
       "      <td>64737183</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>40.0</td>\n",
       "      <td>intron (NM_017382, intron 1 of 4)</td>\n",
       "      <td>intron (NM_017382, intron 1 of 4)</td>\n",
       "      <td>1073.0</td>\n",
       "      <td>...</td>\n",
       "      <td>53869.0</td>\n",
       "      <td>Mm.1387</td>\n",
       "      <td>NM_017382</td>\n",
       "      <td>ENSMUSG00000004771</td>\n",
       "      <td>Rab11a</td>\n",
       "      <td>-</td>\n",
       "      <td>RAB11A, member RAS oncogene family</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>139.74</td>\n",
       "      <td>110.76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>52_22519</td>\n",
       "      <td>chr4</td>\n",
       "      <td>155561699</td>\n",
       "      <td>155562699</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>38.0</td>\n",
       "      <td>promoter-TSS (NM_138671)</td>\n",
       "      <td>promoter-TSS (NM_138671)</td>\n",
       "      <td>-182.0</td>\n",
       "      <td>...</td>\n",
       "      <td>192185.0</td>\n",
       "      <td>Mm.28347</td>\n",
       "      <td>NM_138671</td>\n",
       "      <td>ENSMUSG00000029063</td>\n",
       "      <td>Nadk</td>\n",
       "      <td>4432404C02Rik|BC004012</td>\n",
       "      <td>NAD kinase</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>135.47</td>\n",
       "      <td>114.40</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919_round2_00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921_round2_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423)  \\\n",
       "0                                            52_6108                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "1                                           52_27549                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "2                                             52_532                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "3                                           52_21363                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "4                                           52_22519                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "\n",
       "     Chr      Start        End Strand  Peak Score  Focus Ratio/Region Size  \\\n",
       "0  chr19    6869077    6870077      +        1000                     88.0   \n",
       "1   chr7  128136635  128137635      +        1000                     32.0   \n",
       "2  chr11   20200837   20201837      +        1000                    178.0   \n",
       "3   chr9   64736183   64737183      +        1000                     40.0   \n",
       "4   chr4  155561699  155562699      +        1000                     38.0   \n",
       "\n",
       "                            Annotation                  Detailed Annotation  \\\n",
       "0                           Intergenic                  RLTR19-int|LTR|ERVK   \n",
       "1  intron (NM_021334, intron 14 of 29)  intron (NM_021334, intron 14 of 29)   \n",
       "2             promoter-TSS (NM_008996)             promoter-TSS (NM_008996)   \n",
       "3    intron (NM_017382, intron 1 of 4)    intron (NM_017382, intron 1 of 4)   \n",
       "4             promoter-TSS (NM_138671)             promoter-TSS (NM_138671)   \n",
       "\n",
       "   Distance to TSS  ... Entrez ID  Nearest Unigene Nearest Refseq  \\\n",
       "0         -11366.0  ...   78317.0        Mm.329596      NM_198008   \n",
       "1           7567.0  ...   16411.0         Mm.22378      NM_021334   \n",
       "2            -75.0  ...   19324.0        Mm.271944      NM_008996   \n",
       "3           1073.0  ...   53869.0          Mm.1387      NM_017382   \n",
       "4           -182.0  ...  192185.0         Mm.28347      NM_138671   \n",
       "\n",
       "      Nearest Ensembl Gene Name                      Gene Alias  \\\n",
       "0  ENSMUSG00000047810   Ccdc88b            2610041P08Rik|Ccdc88   \n",
       "1  ENSMUSG00000030789     Itgax         AI449405|Cd11c|Cr4|N418   \n",
       "2  ENSMUSG00000020149     Rab1a  Gtbp|Rab-1|Rab1|Ypt1|mKIAA3012   \n",
       "3  ENSMUSG00000004771    Rab11a                               -   \n",
       "4  ENSMUSG00000029063      Nadk          4432404C02Rik|BC004012   \n",
       "\n",
       "                     Gene Description       Gene Type  \\\n",
       "0   coiled-coil domain containing 88B  protein-coding   \n",
       "1                    integrin alpha X  protein-coding   \n",
       "2   RAB1A, member RAS oncogene family  protein-coding   \n",
       "3  RAB11A, member RAS oncogene family  protein-coding   \n",
       "4                          NAD kinase  protein-coding   \n",
       "\n",
       "  /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 Tag Count in 1000 bp (18749728.0 Total, normalization factor = 0.53, effective total = 1e7)  \\\n",
       "0                                             102.40                                                                                                                                                 \n",
       "1                                              51.73                                                                                                                                                 \n",
       "2                                             450.67                                                                                                                                                 \n",
       "3                                             139.74                                                                                                                                                 \n",
       "4                                             135.47                                                                                                                                                 \n",
       "\n",
       "   /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423 Tag Count in 1000 bp (19231109.0 Total, normalization factor = 0.52, effective total = 1e7)  \n",
       "0                                             111.28                                                                                                                                                 \n",
       "1                                              24.96                                                                                                                                                 \n",
       "2                                             384.79                                                                                                                                                 \n",
       "3                                             110.76                                                                                                                                                 \n",
       "4                                             114.40                                                                                                                                                 \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strain2 = 'C57_AMLNDiet_round2'\n",
    "path = '/home/hmummey/data/ATAC/annotated_peak_calling/'\n",
    "annot_file2 = path + strain2 + '_annot.txt'\n",
    "annot_df2 = pd.read_csv(annot_file2, sep='\\t')\n",
    "\n",
    "print(len(annot_df2))\n",
    "annot_df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423</th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6.692092</td>\n",
       "      <td>6.810957</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5.720552</td>\n",
       "      <td>4.698218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8.819125</td>\n",
       "      <td>8.591672</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>7.136889</td>\n",
       "      <td>6.804260</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7.092440</td>\n",
       "      <td>6.850499</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423  \\\n",
       "0                                           6.692092      \n",
       "1                                           5.720552      \n",
       "2                                           8.819125      \n",
       "3                                           7.136889      \n",
       "4                                           7.092440      \n",
       "\n",
       "   00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423  \n",
       "0                                           6.810957     \n",
       "1                                           4.698218     \n",
       "2                                           8.591672     \n",
       "3                                           6.804260     \n",
       "4                                           6.850499     "
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "histone_tags12 = annot_df2.iloc[:,-2:]\n",
    "histone_tags12.columns = [i.split(' Tag')[0].split('/')[-1] for i in histone_tags12.columns.values]\n",
    "histone_tags22 = np.log2(histone_tags12+1) #convert to log2 scale\n",
    "histone_tags22.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423</th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>46223.000000</td>\n",
       "      <td>46223.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>4.891646</td>\n",
       "      <td>4.787522</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>2.123845</td>\n",
       "      <td>2.046592</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>3.252476</td>\n",
       "      <td>3.137504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>5.067381</td>\n",
       "      <td>4.912650</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>6.592008</td>\n",
       "      <td>6.413459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>15.693303</td>\n",
       "      <td>15.991561</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423  \\\n",
       "count                                       46223.000000      \n",
       "mean                                            4.891646      \n",
       "std                                             2.123845      \n",
       "min                                             0.000000      \n",
       "25%                                             3.252476      \n",
       "50%                                             5.067381      \n",
       "75%                                             6.592008      \n",
       "max                                            15.693303      \n",
       "\n",
       "       00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423  \n",
       "count                                       46223.000000     \n",
       "mean                                            4.787522     \n",
       "std                                             2.046592     \n",
       "min                                             0.000000     \n",
       "25%                                             3.137504     \n",
       "50%                                             4.912650     \n",
       "75%                                             6.413459     \n",
       "max                                            15.991561     "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "histone_tags22.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Identify active enhancers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1) By setting an average threshold on H3K27ac tags > 16 (average 2 replicates, cutoff 16 on the actual tag counts, or 4 on log2 scale)\n",
    "\n",
    "2) HOMER annotations to be at intronic or intergenic regions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2.5292361802923393, 6.052656031308695, 8.081048307330793, 5.594765277553292, 6.697524937890444]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>C57_control_170915</th>\n",
       "      <th>C57_control_180423</th>\n",
       "      <th>C57_control_avg</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2.875780</td>\n",
       "      <td>2.182692</td>\n",
       "      <td>2.529236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6.100137</td>\n",
       "      <td>6.005175</td>\n",
       "      <td>6.052656</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8.032762</td>\n",
       "      <td>8.129335</td>\n",
       "      <td>8.081048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5.568944</td>\n",
       "      <td>5.620586</td>\n",
       "      <td>5.594765</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6.750473</td>\n",
       "      <td>6.644577</td>\n",
       "      <td>6.697525</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   C57_control_170915  C57_control_180423  C57_control_avg\n",
       "0            2.875780            2.182692         2.529236\n",
       "1            6.100137            6.005175         6.052656\n",
       "2            8.032762            8.129335         8.081048\n",
       "3            5.568944            5.620586         5.594765\n",
       "4            6.750473            6.644577         6.697525"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "histone_tags2.columns = [\"C57_control_170915\", \"C57_control_180423\"]\n",
    "z = [((row.C57_control_170915 + row.C57_control_180423)/2) for index, row in histone_tags2.iterrows() ]\n",
    "print(z[:5])\n",
    "histone_tags2['C57_control_avg'] = z\n",
    "histone_tags2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0    intron (NM_001029873, intron 39 of 41)\n",
      "1        intron (NM_198109, intron 1 of 14)\n",
      "2                  promoter-TSS (NM_007713)\n",
      "3      intron (NM_001162477, intron 1 of 9)\n",
      "4               promoter-TSS (NM_001359477)\n",
      "5     intron (NM_001040111, intron 1 of 33)\n",
      "6               promoter-TSS (NM_001083334)\n",
      "7                                Intergenic\n",
      "8       intron (NM_011075, intron 26 of 27)\n",
      "9                                Intergenic\n",
      "Name: Annotation, dtype: object\n"
     ]
    }
   ],
   "source": [
    "#glancing at annotations\n",
    "print((annot_df.iloc[:,7])[:10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#just gonna loop through all rows (for both dfs) and record rows to save as those with:\n",
    "#histone > 16 (average 2 replicates, cutoff 16 on the actual tag counts, or 4 on log2 scale)\n",
    "#Annotation containing either: intron, Intergenic\n",
    "\n",
    "save_rows = []\n",
    "for row in np.arange(len(annot_df)):\n",
    "    #grab full annotation\n",
    "    full_annot = str(annot_df.iloc[row,7])\n",
    "    if (\"intron\" in full_annot or \"Intergenic\" in full_annot) and (histone_tags2.iloc[row,2] > 4):\n",
    "        save_rows.append(row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22282\n",
      "[1, 3, 5, 7, 10, 13, 14, 15, 16, 17]\n"
     ]
    }
   ],
   "source": [
    "print(len(save_rows))\n",
    "print(save_rows[:10]) #looks good!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22282\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812_round2_00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811_round2_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423)</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "      <th>Focus Ratio/Region Size</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Detailed Annotation</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>...</th>\n",
       "      <th>Entrez ID</th>\n",
       "      <th>Nearest Unigene</th>\n",
       "      <th>Nearest Refseq</th>\n",
       "      <th>Nearest Ensembl</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Alias</th>\n",
       "      <th>Gene Description</th>\n",
       "      <th>Gene Type</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 Tag Count in 1000 bp (9468035.0 Total, normalization factor = 1.06, effective total = 1e7)</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423 Tag Count in 1000 bp (22615879.0 Total, normalization factor = 0.44, effective total = 1e7)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>65_22643</td>\n",
       "      <td>chr5</td>\n",
       "      <td>113991974</td>\n",
       "      <td>113992974</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>43.0</td>\n",
       "      <td>intron (NM_198109, intron 1 of 14)</td>\n",
       "      <td>L1MB8|LINE|L1</td>\n",
       "      <td>1283.0</td>\n",
       "      <td>...</td>\n",
       "      <td>231637.0</td>\n",
       "      <td>Mm.389682</td>\n",
       "      <td>NM_198109</td>\n",
       "      <td>ENSMUSG00000042121</td>\n",
       "      <td>Ssh1</td>\n",
       "      <td>AW551225|Gm1394|Gm1395|SSH-1|SSH-1L</td>\n",
       "      <td>slingshot protein phosphatase 1</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>67.60</td>\n",
       "      <td>63.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>65_28521</td>\n",
       "      <td>chr7</td>\n",
       "      <td>97154713</td>\n",
       "      <td>97155713</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>35.0</td>\n",
       "      <td>intron (NM_001162477, intron 1 of 9)</td>\n",
       "      <td>intron (NM_001162477, intron 1 of 9)</td>\n",
       "      <td>73462.0</td>\n",
       "      <td>...</td>\n",
       "      <td>14389.0</td>\n",
       "      <td>Mm.42033</td>\n",
       "      <td>NM_010248</td>\n",
       "      <td>ENSMUSG00000004508</td>\n",
       "      <td>Gab2</td>\n",
       "      <td>AI463667|D130058I17Rik|p97</td>\n",
       "      <td>growth factor receptor bound protein 2-associa...</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>46.47</td>\n",
       "      <td>48.20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>65_34295</td>\n",
       "      <td>chr7</td>\n",
       "      <td>101360702</td>\n",
       "      <td>101361702</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>29.0</td>\n",
       "      <td>intron (NM_001040111, intron 1 of 33)</td>\n",
       "      <td>intron (NM_001040111, intron 1 of 33)</td>\n",
       "      <td>13133.0</td>\n",
       "      <td>...</td>\n",
       "      <td>69710.0</td>\n",
       "      <td>Mm.277687</td>\n",
       "      <td>NM_027180</td>\n",
       "      <td>ENSMUSG00000032812</td>\n",
       "      <td>Arap1</td>\n",
       "      <td>2410002L19Rik|Centd2|mKIAA0782</td>\n",
       "      <td>ArfGAP with RhoGAP domain, ankyrin repeat and ...</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>191.17</td>\n",
       "      <td>166.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>65_32720</td>\n",
       "      <td>chr15</td>\n",
       "      <td>97949323</td>\n",
       "      <td>97950323</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>30.0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>-14406.0</td>\n",
       "      <td>...</td>\n",
       "      <td>380967.0</td>\n",
       "      <td>Mm.196630</td>\n",
       "      <td>NM_201359</td>\n",
       "      <td>ENSMUSG00000052369</td>\n",
       "      <td>Tmem106c</td>\n",
       "      <td>AI046681|BC046621|D15Ertd405e</td>\n",
       "      <td>transmembrane protein 106C</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>85.55</td>\n",
       "      <td>63.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>65_8821</td>\n",
       "      <td>chr2</td>\n",
       "      <td>164868133</td>\n",
       "      <td>164869133</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>79.0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>MTC|LTR|ERVL-MaLR</td>\n",
       "      <td>-10735.0</td>\n",
       "      <td>...</td>\n",
       "      <td>228866.0</td>\n",
       "      <td>Mm.241134</td>\n",
       "      <td>NM_146129</td>\n",
       "      <td>ENSMUSG00000039849</td>\n",
       "      <td>Pcif1</td>\n",
       "      <td>2310022K11Rik|C20orf67|F730014I05Rik</td>\n",
       "      <td>PDX1 C-terminal inhibiting factor 1</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>66.54</td>\n",
       "      <td>88.88</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812_round2_00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811_round2_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423)  \\\n",
       "1                                            65_22643                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "3                                            65_28521                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "5                                            65_34295                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "7                                            65_32720                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "10                                            65_8821                                                                                                                                                                                                                                                                                                                                                                                            \n",
       "\n",
       "      Chr      Start        End Strand  Peak Score  Focus Ratio/Region Size  \\\n",
       "1    chr5  113991974  113992974      +        1000                     43.0   \n",
       "3    chr7   97154713   97155713      +        1000                     35.0   \n",
       "5    chr7  101360702  101361702      +        1000                     29.0   \n",
       "7   chr15   97949323   97950323      +        1000                     30.0   \n",
       "10   chr2  164868133  164869133      +        1000                     79.0   \n",
       "\n",
       "                               Annotation  \\\n",
       "1      intron (NM_198109, intron 1 of 14)   \n",
       "3    intron (NM_001162477, intron 1 of 9)   \n",
       "5   intron (NM_001040111, intron 1 of 33)   \n",
       "7                              Intergenic   \n",
       "10                             Intergenic   \n",
       "\n",
       "                      Detailed Annotation  Distance to TSS  ... Entrez ID  \\\n",
       "1                           L1MB8|LINE|L1           1283.0  ...  231637.0   \n",
       "3    intron (NM_001162477, intron 1 of 9)          73462.0  ...   14389.0   \n",
       "5   intron (NM_001040111, intron 1 of 33)          13133.0  ...   69710.0   \n",
       "7                              Intergenic         -14406.0  ...  380967.0   \n",
       "10                      MTC|LTR|ERVL-MaLR         -10735.0  ...  228866.0   \n",
       "\n",
       "    Nearest Unigene Nearest Refseq     Nearest Ensembl Gene Name  \\\n",
       "1         Mm.389682      NM_198109  ENSMUSG00000042121      Ssh1   \n",
       "3          Mm.42033      NM_010248  ENSMUSG00000004508      Gab2   \n",
       "5         Mm.277687      NM_027180  ENSMUSG00000032812     Arap1   \n",
       "7         Mm.196630      NM_201359  ENSMUSG00000052369  Tmem106c   \n",
       "10        Mm.241134      NM_146129  ENSMUSG00000039849     Pcif1   \n",
       "\n",
       "                              Gene Alias  \\\n",
       "1    AW551225|Gm1394|Gm1395|SSH-1|SSH-1L   \n",
       "3             AI463667|D130058I17Rik|p97   \n",
       "5         2410002L19Rik|Centd2|mKIAA0782   \n",
       "7          AI046681|BC046621|D15Ertd405e   \n",
       "10  2310022K11Rik|C20orf67|F730014I05Rik   \n",
       "\n",
       "                                     Gene Description       Gene Type  \\\n",
       "1                     slingshot protein phosphatase 1  protein-coding   \n",
       "3   growth factor receptor bound protein 2-associa...  protein-coding   \n",
       "5   ArfGAP with RhoGAP domain, ankyrin repeat and ...  protein-coding   \n",
       "7                          transmembrane protein 106C  protein-coding   \n",
       "10                PDX1 C-terminal inhibiting factor 1  protein-coding   \n",
       "\n",
       "   /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 Tag Count in 1000 bp (9468035.0 Total, normalization factor = 1.06, effective total = 1e7)  \\\n",
       "1                                               67.60                                                                                                                                         \n",
       "3                                               46.47                                                                                                                                         \n",
       "5                                              191.17                                                                                                                                         \n",
       "7                                               85.55                                                                                                                                         \n",
       "10                                              66.54                                                                                                                                         \n",
       "\n",
       "    /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423 Tag Count in 1000 bp (22615879.0 Total, normalization factor = 0.44, effective total = 1e7)  \n",
       "1                                               63.23                                                                                                                                          \n",
       "3                                               48.20                                                                                                                                          \n",
       "5                                              166.70                                                                                                                                          \n",
       "7                                               63.23                                                                                                                                          \n",
       "10                                              88.88                                                                                                                                          \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filt_annot = annot_df.iloc[save_rows,:]\n",
    "print(len(filt_annot))\n",
    "filt_annot.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "#export this just in case\n",
    "out_file = path + strain + \"_filt_annot.txt\"\n",
    "filt_annot.to_csv(out_file, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### AMLN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[6.751524767082686, 5.209385284914761, 8.705398610229484, 6.970574361768902, 6.971469721998103]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>C57_AMLN1</th>\n",
       "      <th>C57_control_AMLN2</th>\n",
       "      <th>C57_AMLN_avg</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6.692092</td>\n",
       "      <td>6.810957</td>\n",
       "      <td>6.751525</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>5.720552</td>\n",
       "      <td>4.698218</td>\n",
       "      <td>5.209385</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>8.819125</td>\n",
       "      <td>8.591672</td>\n",
       "      <td>8.705399</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>7.136889</td>\n",
       "      <td>6.804260</td>\n",
       "      <td>6.970574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>7.092440</td>\n",
       "      <td>6.850499</td>\n",
       "      <td>6.971470</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   C57_AMLN1  C57_control_AMLN2  C57_AMLN_avg\n",
       "0   6.692092           6.810957      6.751525\n",
       "1   5.720552           4.698218      5.209385\n",
       "2   8.819125           8.591672      8.705399\n",
       "3   7.136889           6.804260      6.970574\n",
       "4   7.092440           6.850499      6.971470"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "histone_tags22.columns = [\"C57_AMLN1\", \"C57_control_AMLN2\"]\n",
    "z2 = [((row.C57_AMLN1 + row.C57_control_AMLN2)/2) for index, row in histone_tags22.iterrows() ]\n",
    "print(z2[:5])\n",
    "histone_tags22['C57_AMLN_avg'] = z2\n",
    "histone_tags22.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0                             Intergenic\n",
      "1    intron (NM_021334, intron 14 of 29)\n",
      "2               promoter-TSS (NM_008996)\n",
      "3      intron (NM_017382, intron 1 of 4)\n",
      "4               promoter-TSS (NM_138671)\n",
      "5               promoter-TSS (NM_175542)\n",
      "6     intron (NM_031376, intron 2 of 16)\n",
      "7               promoter-TSS (NM_026050)\n",
      "8      intron (NM_007651, intron 1 of 7)\n",
      "9            promoter-TSS (NM_001048267)\n",
      "Name: Annotation, dtype: object\n"
     ]
    }
   ],
   "source": [
    "#glancing at annotations\n",
    "print((annot_df2.iloc[:,7])[:10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19161\n",
      "[0, 1, 3, 6, 8, 11, 12, 13, 14, 15]\n"
     ]
    }
   ],
   "source": [
    "#just gonna loop through all rows (for both dfs) and record rows to save as those with:\n",
    "#histone > 16 (average 2 replicates, cutoff 16 on the actual tag counts, or 4 on log2 scale)\n",
    "#Annotation containing either: intron, Intergenic\n",
    "\n",
    "save_rows2 = []\n",
    "for row in np.arange(len(annot_df2)):\n",
    "    #grab full annotation\n",
    "    full_annot2 = str(annot_df2.iloc[row,7])\n",
    "    if (\"intron\" in full_annot2 or \"Intergenic\" in full_annot2) and (histone_tags22.iloc[row,2] > 4):\n",
    "        save_rows2.append(row)\n",
    "        \n",
    "print(len(save_rows2))\n",
    "print(save_rows2[:10]) #looks good!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19161\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919_round2_00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921_round2_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423)</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "      <th>Focus Ratio/Region Size</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Detailed Annotation</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>...</th>\n",
       "      <th>Entrez ID</th>\n",
       "      <th>Nearest Unigene</th>\n",
       "      <th>Nearest Refseq</th>\n",
       "      <th>Nearest Ensembl</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Alias</th>\n",
       "      <th>Gene Description</th>\n",
       "      <th>Gene Type</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 Tag Count in 1000 bp (18749728.0 Total, normalization factor = 0.53, effective total = 1e7)</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423 Tag Count in 1000 bp (19231109.0 Total, normalization factor = 0.52, effective total = 1e7)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>52_6108</td>\n",
       "      <td>chr19</td>\n",
       "      <td>6869077</td>\n",
       "      <td>6870077</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>88.0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>RLTR19-int|LTR|ERVK</td>\n",
       "      <td>-11366.0</td>\n",
       "      <td>...</td>\n",
       "      <td>78317.0</td>\n",
       "      <td>Mm.329596</td>\n",
       "      <td>NM_198008</td>\n",
       "      <td>ENSMUSG00000047810</td>\n",
       "      <td>Ccdc88b</td>\n",
       "      <td>2610041P08Rik|Ccdc88</td>\n",
       "      <td>coiled-coil domain containing 88B</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>102.40</td>\n",
       "      <td>111.28</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>52_27549</td>\n",
       "      <td>chr7</td>\n",
       "      <td>128136635</td>\n",
       "      <td>128137635</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>32.0</td>\n",
       "      <td>intron (NM_021334, intron 14 of 29)</td>\n",
       "      <td>intron (NM_021334, intron 14 of 29)</td>\n",
       "      <td>7567.0</td>\n",
       "      <td>...</td>\n",
       "      <td>16411.0</td>\n",
       "      <td>Mm.22378</td>\n",
       "      <td>NM_021334</td>\n",
       "      <td>ENSMUSG00000030789</td>\n",
       "      <td>Itgax</td>\n",
       "      <td>AI449405|Cd11c|Cr4|N418</td>\n",
       "      <td>integrin alpha X</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>51.73</td>\n",
       "      <td>24.96</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>52_21363</td>\n",
       "      <td>chr9</td>\n",
       "      <td>64736183</td>\n",
       "      <td>64737183</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>40.0</td>\n",
       "      <td>intron (NM_017382, intron 1 of 4)</td>\n",
       "      <td>intron (NM_017382, intron 1 of 4)</td>\n",
       "      <td>1073.0</td>\n",
       "      <td>...</td>\n",
       "      <td>53869.0</td>\n",
       "      <td>Mm.1387</td>\n",
       "      <td>NM_017382</td>\n",
       "      <td>ENSMUSG00000004771</td>\n",
       "      <td>Rab11a</td>\n",
       "      <td>-</td>\n",
       "      <td>RAB11A, member RAS oncogene family</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>139.74</td>\n",
       "      <td>110.76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>52_28164</td>\n",
       "      <td>chr19</td>\n",
       "      <td>41345112</td>\n",
       "      <td>41346112</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>31.0</td>\n",
       "      <td>intron (NM_031376, intron 2 of 16)</td>\n",
       "      <td>intron (NM_031376, intron 2 of 16)</td>\n",
       "      <td>39458.0</td>\n",
       "      <td>...</td>\n",
       "      <td>83490.0</td>\n",
       "      <td>Mm.222266</td>\n",
       "      <td>NM_031376</td>\n",
       "      <td>ENSMUSG00000025017</td>\n",
       "      <td>Pik3ap1</td>\n",
       "      <td>1810044J04Rik|BCAP</td>\n",
       "      <td>phosphoinositide-3-kinase adaptor protein 1</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>293.34</td>\n",
       "      <td>274.56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>52_17359</td>\n",
       "      <td>chr3</td>\n",
       "      <td>106784764</td>\n",
       "      <td>106785764</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>47.0</td>\n",
       "      <td>intron (NM_007651, intron 1 of 7)</td>\n",
       "      <td>intron (NM_007651, intron 1 of 7)</td>\n",
       "      <td>4885.0</td>\n",
       "      <td>...</td>\n",
       "      <td>12508.0</td>\n",
       "      <td>Mm.316861</td>\n",
       "      <td>NM_007651</td>\n",
       "      <td>ENSMUSG00000040747</td>\n",
       "      <td>Cd53</td>\n",
       "      <td>AI323659|Ox-44|Tspan25</td>\n",
       "      <td>CD53 antigen</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>141.34</td>\n",
       "      <td>137.80</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919_round2_00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921_round2_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423)  \\\n",
       "0                                            52_6108                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "1                                           52_27549                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "3                                           52_21363                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "6                                           52_28164                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "8                                           52_17359                                                                                                                                                                                                                                                                                                                                                                                                                        \n",
       "\n",
       "     Chr      Start        End Strand  Peak Score  Focus Ratio/Region Size  \\\n",
       "0  chr19    6869077    6870077      +        1000                     88.0   \n",
       "1   chr7  128136635  128137635      +        1000                     32.0   \n",
       "3   chr9   64736183   64737183      +        1000                     40.0   \n",
       "6  chr19   41345112   41346112      +        1000                     31.0   \n",
       "8   chr3  106784764  106785764      +        1000                     47.0   \n",
       "\n",
       "                            Annotation                  Detailed Annotation  \\\n",
       "0                           Intergenic                  RLTR19-int|LTR|ERVK   \n",
       "1  intron (NM_021334, intron 14 of 29)  intron (NM_021334, intron 14 of 29)   \n",
       "3    intron (NM_017382, intron 1 of 4)    intron (NM_017382, intron 1 of 4)   \n",
       "6   intron (NM_031376, intron 2 of 16)   intron (NM_031376, intron 2 of 16)   \n",
       "8    intron (NM_007651, intron 1 of 7)    intron (NM_007651, intron 1 of 7)   \n",
       "\n",
       "   Distance to TSS  ... Entrez ID  Nearest Unigene Nearest Refseq  \\\n",
       "0         -11366.0  ...   78317.0        Mm.329596      NM_198008   \n",
       "1           7567.0  ...   16411.0         Mm.22378      NM_021334   \n",
       "3           1073.0  ...   53869.0          Mm.1387      NM_017382   \n",
       "6          39458.0  ...   83490.0        Mm.222266      NM_031376   \n",
       "8           4885.0  ...   12508.0        Mm.316861      NM_007651   \n",
       "\n",
       "      Nearest Ensembl Gene Name               Gene Alias  \\\n",
       "0  ENSMUSG00000047810   Ccdc88b     2610041P08Rik|Ccdc88   \n",
       "1  ENSMUSG00000030789     Itgax  AI449405|Cd11c|Cr4|N418   \n",
       "3  ENSMUSG00000004771    Rab11a                        -   \n",
       "6  ENSMUSG00000025017   Pik3ap1       1810044J04Rik|BCAP   \n",
       "8  ENSMUSG00000040747      Cd53   AI323659|Ox-44|Tspan25   \n",
       "\n",
       "                              Gene Description       Gene Type  \\\n",
       "0            coiled-coil domain containing 88B  protein-coding   \n",
       "1                             integrin alpha X  protein-coding   \n",
       "3           RAB11A, member RAS oncogene family  protein-coding   \n",
       "6  phosphoinositide-3-kinase adaptor protein 1  protein-coding   \n",
       "8                                 CD53 antigen  protein-coding   \n",
       "\n",
       "  /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 Tag Count in 1000 bp (18749728.0 Total, normalization factor = 0.53, effective total = 1e7)  \\\n",
       "0                                             102.40                                                                                                                                                 \n",
       "1                                              51.73                                                                                                                                                 \n",
       "3                                             139.74                                                                                                                                                 \n",
       "6                                             293.34                                                                                                                                                 \n",
       "8                                             141.34                                                                                                                                                 \n",
       "\n",
       "   /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423 Tag Count in 1000 bp (19231109.0 Total, normalization factor = 0.52, effective total = 1e7)  \n",
       "0                                             111.28                                                                                                                                                 \n",
       "1                                              24.96                                                                                                                                                 \n",
       "3                                             110.76                                                                                                                                                 \n",
       "6                                             274.56                                                                                                                                                 \n",
       "8                                             137.80                                                                                                                                                 \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filt_annot2 = annot_df2.iloc[save_rows2,:]\n",
    "print(len(filt_annot2))\n",
    "filt_annot2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "#export this just in case\n",
    "out_file2 = path + strain2 + \"_filt_annot.txt\"\n",
    "filt_annot.to_csv(out_file2, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Convert peak file to BED"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Bed file format requires:\n",
    "- chrom - name of the chromosome or scaffold. Any valid seq_region_name can be used, and chromosome names can be given with or without the 'chr' prefix.\n",
    "- chromStart - Start position of the feature in standard chromosomal coordinates (i.e. first base is 0).\n",
    "- chromEnd - End position of the feature in standard chromosomal coordinates\n",
    "\n",
    "Optional columns: name, score, strand, thickStart, thickEnd, itemRgb, blockCount, blockSizes, blockStarts\n",
    "\n",
    "from: https://m.ensembl.org/info/website/upload/bed.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22282\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5</td>\n",
       "      <td>113991974</td>\n",
       "      <td>113992974</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr7</td>\n",
       "      <td>97154713</td>\n",
       "      <td>97155713</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr7</td>\n",
       "      <td>101360702</td>\n",
       "      <td>101361702</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr15</td>\n",
       "      <td>97949323</td>\n",
       "      <td>97950323</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr2</td>\n",
       "      <td>164868133</td>\n",
       "      <td>164869133</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr      Start        End Strand  Peak Score\n",
       "1    chr5  113991974  113992974      +        1000\n",
       "3    chr7   97154713   97155713      +        1000\n",
       "5    chr7  101360702  101361702      +        1000\n",
       "7   chr15   97949323   97950323      +        1000\n",
       "10   chr2  164868133  164869133      +        1000"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#just going to use the required cols (and maybe strand and score? why not, I guess)\n",
    "save_cols = [1,2,3,4,5]\n",
    "bed_df = filt_annot.iloc[:,save_cols]\n",
    "print(len(bed_df))\n",
    "bed_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['chr1', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17', 'chr18', 'chr19', 'chr2', 'chr3', 'chr4', 'chr4_GL456216_random', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9', 'chrUn_JH584304', 'chrX', 'chrY']\n"
     ]
    }
   ],
   "source": [
    "chr_list = list(set(bed_df.iloc[:,0]))\n",
    "chr_list.sort()\n",
    "print(chr_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove all rows that are chrY (makes the script to create the background, error) or chrUn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22224\n",
      "[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]\n"
     ]
    }
   ],
   "source": [
    "#also might as well try removing rows with Un chrs (will do this manually and see if helps)\n",
    "keep_rows12 = []\n",
    "for i in range(len(bed_df)):\n",
    "    chr_str = bed_df.iloc[i,0]\n",
    "    if (\"chrY\" in chr_str) or (\"Un\" in chr_str):\n",
    "        pass\n",
    "    else:\n",
    "        keep_rows12.append(i)\n",
    "\n",
    "print(len(keep_rows12))\n",
    "print(keep_rows12[:20])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22224\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr5</td>\n",
       "      <td>113991974</td>\n",
       "      <td>113992974</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr7</td>\n",
       "      <td>97154713</td>\n",
       "      <td>97155713</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr7</td>\n",
       "      <td>101360702</td>\n",
       "      <td>101361702</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr15</td>\n",
       "      <td>97949323</td>\n",
       "      <td>97950323</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr2</td>\n",
       "      <td>164868133</td>\n",
       "      <td>164869133</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr      Start        End Strand  Peak Score\n",
       "1    chr5  113991974  113992974      +        1000\n",
       "3    chr7   97154713   97155713      +        1000\n",
       "5    chr7  101360702  101361702      +        1000\n",
       "7   chr15   97949323   97950323      +        1000\n",
       "10   chr2  164868133  164869133      +        1000"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bed_df_cut12 = bed_df.iloc[keep_rows12,:]\n",
    "print(len(bed_df_cut12))\n",
    "bed_df_cut12.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "out3 = path + strain + \"_filt_annot_fin.bed\"\n",
    "bed_df_cut12.to_csv(out3, index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### AMLN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19161\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6869077</td>\n",
       "      <td>6870077</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr7</td>\n",
       "      <td>128136635</td>\n",
       "      <td>128137635</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr9</td>\n",
       "      <td>64736183</td>\n",
       "      <td>64737183</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41345112</td>\n",
       "      <td>41346112</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3</td>\n",
       "      <td>106784764</td>\n",
       "      <td>106785764</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Strand  Peak Score\n",
       "0  chr19    6869077    6870077      +        1000\n",
       "1   chr7  128136635  128137635      +        1000\n",
       "3   chr9   64736183   64737183      +        1000\n",
       "6  chr19   41345112   41346112      +        1000\n",
       "8   chr3  106784764  106785764      +        1000"
      ]
     },
     "execution_count": 78,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#just going to use the required cols (and maybe strand and score? why not, I guess)\n",
    "\n",
    "save_cols2 = [1,2,3,4,5]\n",
    "bed_df2 = filt_annot2.iloc[:,save_cols2]\n",
    "print(len(bed_df2))\n",
    "bed_df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['chr1', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17', 'chr18', 'chr19', 'chr2', 'chr3', 'chr4', 'chr4_GL456216_random', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9', 'chrUn_JH584304', 'chrX', 'chrY']\n"
     ]
    }
   ],
   "source": [
    "chr_list = list(set(bed_df2.iloc[:,0]))\n",
    "chr_list.sort()\n",
    "print(chr_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove all rows that are chrY (makes the script to create the background, error) or chrUn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19114\n",
      "[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]\n"
     ]
    }
   ],
   "source": [
    "#also might as well try removing rows with Un chrs (will do this manually and see if helps)\n",
    "keep_rows22 = []\n",
    "for i in range(len(bed_df2)):\n",
    "    chr_str = bed_df2.iloc[i,0]\n",
    "    if (\"chrY\" in chr_str) or (\"Un\" in chr_str):\n",
    "        pass\n",
    "    else:\n",
    "        keep_rows22.append(i)\n",
    "        \n",
    "print(len(keep_rows22))\n",
    "print(keep_rows22[:20])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "19114\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr19</td>\n",
       "      <td>6869077</td>\n",
       "      <td>6870077</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr7</td>\n",
       "      <td>128136635</td>\n",
       "      <td>128137635</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr9</td>\n",
       "      <td>64736183</td>\n",
       "      <td>64737183</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr19</td>\n",
       "      <td>41345112</td>\n",
       "      <td>41346112</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr3</td>\n",
       "      <td>106784764</td>\n",
       "      <td>106785764</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End Strand  Peak Score\n",
       "0  chr19    6869077    6870077      +        1000\n",
       "1   chr7  128136635  128137635      +        1000\n",
       "3   chr9   64736183   64737183      +        1000\n",
       "6  chr19   41345112   41346112      +        1000\n",
       "8   chr3  106784764  106785764      +        1000"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bed_df_cut22 = bed_df2.iloc[keep_rows22,:]\n",
    "print(len(bed_df_cut22))\n",
    "bed_df_cut22.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "out4 = path + strain2 + \"_filt_annot_fin.bed\"\n",
    "bed_df_cut22.to_csv(out4, index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Generate random backgrounds"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Bed file is the positives we've just chosen (only matching GC content). Make sure to remove the top row (col names) in terminal with sed '1d' before running the script!"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/hmummey/data/ATAC/annotated_peak_calling/C57_control_round2_filt_annot_fin_bg\n",
      "reading genome mm10\n",
      "fasta file does not exist: chr20\n",
      "fasta file does not exist: chr21\n",
      "fasta file does not exist: chr22\n",
      "done reading genome\n",
      "0 0\n",
      "target GC: 0.37210436347276776 background GC: 0.3745772895570298 target length: 1000 numTargetPositions 2223 backgroundPositions 2223\n",
      "0 0\n",
      "target GC: 0.406293429342916 background GC: 0.4064176147884336 target length: 1000 numTargetPositions 2222 backgroundPositions 2222\n",
      "0 0\n",
      "target GC: 0.4242187218721681 background GC: 0.4224825579460853 target length: 1000 numTargetPositions 2222 backgroundPositions 2222\n",
      "0 0\n",
      "target GC: 0.4386300630062809 background GC: 0.435475865268825 target length: 1000 numTargetPositions 2222 backgroundPositions 2222\n",
      "0 0\n",
      "target GC: 0.4519702970296826 background GC: 0.4483882454179282 target length: 1000 numTargetPositions 2222 backgroundPositions 2222\n",
      "0 0\n",
      "target GC: 0.46530198019799884 background GC: 0.46021305427243947 target length: 1000 numTargetPositions 2222 backgroundPositions 2222\n",
      "0 0\n",
      "target GC: 0.4795972097209505 background GC: 0.4734500423069067 target length: 1000 numTargetPositions 2222 backgroundPositions 2222\n",
      "0 0\n",
      "target GC: 0.49652610261023866 background GC: 0.48922139966242173 target length: 1000 numTargetPositions 2222 backgroundPositions 2222\n",
      "0 0\n",
      "target GC: 0.5195774077407507 background GC: 0.5113239595687611 target length: 1000 numTargetPositions 2222 backgroundPositions 2222\n",
      "0 0\n",
      "target GC: 0.5826228622862024 background GC: 0.5719712330873189 target length: 1000 numTargetPositions 2222 backgroundPositions 2222\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "bed_file=\"/home/hmummey/data/ATAC/annotated_peak_calling/C57_control_round2_filt_annot_fin.bed\" #+ peaks, no chrY, no chrUn\n",
    "output=$(echo $bed_file | sed 's/.bed/_bg/g')\n",
    "echo $output\n",
    "python /home/zes017/Spacing/Codes/generate_background_coordinates.py $bed_file $output"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### AMLN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/hmummey/data/ATAC/annotated_peak_calling/C57_AMLNDiet_round2_filt_annot_fin_bg\n",
      "reading genome mm10\n",
      "fasta file does not exist: chr20\n",
      "fasta file does not exist: chr21\n",
      "fasta file does not exist: chr22\n",
      "done reading genome\n",
      "0 0\n",
      "target GC: 0.3800549163179718 background GC: 0.3820771278930075 target length: 1000 numTargetPositions 1912 backgroundPositions 1912\n",
      "0 0\n",
      "target GC: 0.4144524058577189 background GC: 0.41463975355186583 target length: 1000 numTargetPositions 1912 backgroundPositions 1912\n",
      "0 0\n",
      "target GC: 0.4324071166928084 background GC: 0.4303937820421112 target length: 1000 numTargetPositions 1911 backgroundPositions 1911\n",
      "0 0\n",
      "target GC: 0.4472857142856909 background GC: 0.44361969793678624 target length: 1000 numTargetPositions 1911 backgroundPositions 1911\n",
      "0 0\n",
      "target GC: 0.460633176347438 background GC: 0.4564509274085174 target length: 1000 numTargetPositions 1911 backgroundPositions 1911\n",
      "0 0\n",
      "target GC: 0.47450863422289513 background GC: 0.4688759696608745 target length: 1000 numTargetPositions 1911 backgroundPositions 1911\n",
      "0 0\n",
      "target GC: 0.4893521716378603 background GC: 0.48204647262729516 target length: 1000 numTargetPositions 1911 backgroundPositions 1911\n",
      "0 0\n",
      "target GC: 0.5069246467817632 background GC: 0.4997942925729164 target length: 1000 numTargetPositions 1911 backgroundPositions 1911\n",
      "0 0\n",
      "target GC: 0.5305013082155662 background GC: 0.5228701178465427 target length: 1000 numTargetPositions 1911 backgroundPositions 1911\n",
      "0 0\n",
      "target GC: 0.597787022501277 background GC: 0.589786456348435 target length: 1000 numTargetPositions 1911 backgroundPositions 1911\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "bed_file=\"/home/hmummey/data/ATAC/annotated_peak_calling/C57_AMLNDiet_round2_filt_annot_fin.bed\" #+ peaks, no chrY, no chrUn\n",
    "output=$(echo $bed_file | sed 's/.bed/_bg/g')\n",
    "echo $output\n",
    "python /home/zes017/Spacing/Codes/generate_background_coordinates.py $bed_file $output"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "zs-deeplift",
   "language": "python",
   "name": "zs-deeplift"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
