{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Summary:\n",
    "This notebook will be used to process the raw ATAC data and call peaks (first unfiltered with Homer, then reproducible with IDR). I'll then use the H3K27ac data to call positive and negative peaks from both control and NASH treated data. Compared to previous iterations, I'll be raising our H3K27Ac threshold here to 32 (rather than 16).\n",
    "\n",
    "Output files:\n",
    "- strain_treat_annot.txt: original peak annotation file \n",
    "- strain_treat_filt_annot.txt: filtered peaks with H3K27ac > 16 and \"intron\" or \"Intergenic\"\n",
    "- strain_treat_filt_annot_cut.bed: bed file of ^ with chrY and chrUn peaks removed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "%matplotlib inline\n",
    "\n",
    "import csv\n",
    "import os\n",
    "\n",
    "from scipy.stats.stats import pearsonr,spearmanr\n",
    "\n",
    "from os import listdir\n",
    "import fnmatch"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "New ATAC peaks from Hunter (circa end of Jan; I'm pretty sure I used these when making the round 2 peaks set? But maybe not bc then I'd expect it to NOT be completely identical to the round 1 set??):\n",
    "- /gpfs/data01/glasslab/home/h1bennet/strains/data/ATAC/control/00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119/\n",
    "- /gpfs/data01/glasslab/home/h1bennet/strains/data/ATAC/control/00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916/\n",
    "- /gpfs/data01/glasslab/home/h1bennet/strains/data/ATAC/AMLN_30week/00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425/\n",
    "- /gpfs/data01/glasslab/home/h1bennet/strains/data/ATAC/AMLN_30week/00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Make Genome Browser Hub"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Hunter changed these tag directories a bit so I remade these. BUT they still don't work, and I have no idea why not..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119 00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916 00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812 00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811 01_balbc_Kupffer_ATAC_control_young_balb10A_TDT_JSS_20161013 01_balbc_Kupffer_ATAC_control_young_balb10B_TDT_JSS_20161013 01_balbc_Kupffer_ATAC_control_young_balb10C_TDT_JSS_20161013 02_aj_Kupffer_ATAC_control_young_aj10A_TDT_JSS_20161013 02_aj_Kupffer_ATAC_control_young_aj10B_TDT_JSS_20161013 02_aj_Kupffer_ATAC_control_young_aj10C_TDT_JSS_20161013\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Colors that will be used (change with -color or -gradient options):\n",
      "\tIndex\tColor\tNeg. color\tTag Directory\n",
      "\t1\t255,150,150\t(255,180,180)\t00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119\n",
      "\t2\t150,150,255\t(180,180,255)\t00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916\n",
      "\t3\t150,255,150\t(180,255,180)\t00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812\n",
      "\t4\t255,200,150\t(255,210,180)\t00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811\n",
      "\t5\t150,255,220\t(180,255,240)\t01_balbc_Kupffer_ATAC_control_young_balb10A_TDT_JSS_20161013\n",
      "\t6\t200,150,255\t(210,180,255)\t01_balbc_Kupffer_ATAC_control_young_balb10B_TDT_JSS_20161013\n",
      "\t7\t200,200,150\t(210,210,170)\t01_balbc_Kupffer_ATAC_control_young_balb10C_TDT_JSS_20161013\n",
      "\t8\t150,200,200\t(170,210,210)\t02_aj_Kupffer_ATAC_control_young_aj10A_TDT_JSS_20161013\n",
      "\t9\t200,150,200\t(210,170,210)\t02_aj_Kupffer_ATAC_control_young_aj10B_TDT_JSS_20161013\n",
      "\t10\t50,251,25\t(68,107,94)\t02_aj_Kupffer_ATAC_control_young_aj10C_TDT_JSS_20161013\n",
      "\n",
      "\n",
      "\tOnce finished, you will want to upload the following hub URL:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-control3/hub.txt\n",
      "\n",
      "\tIf loading to the Wash U Epigenome Browser, use:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-control3/washU.hub.txt\n",
      "\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 62\n",
      "!!! Could not open autofile 00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119/00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119/00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 46\n",
      "!!! Could not open autofile 00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916/00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916/00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 131\n",
      "!!! Could not open autofile 00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812/00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812/00_NCoRWT_Kupffer_ATAC_control_young_LN12aM_JSS_150812.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 141\n",
      "!!! Could not open autofile 00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811/00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811/00_NCoRWT_Kupffer_ATAC_control_young_NNaF_JSS_150811.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 50\n",
      "!!! Could not open autofile 01_balbc_Kupffer_ATAC_control_young_balb10A_TDT_JSS_20161013/01_balbc_Kupffer_ATAC_control_young_balb10A_TDT_JSS_20161013.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘01_balbc_Kupffer_ATAC_control_young_balb10A_TDT_JSS_20161013/01_balbc_Kupffer_ATAC_control_young_balb10A_TDT_JSS_20161013.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 71\n",
      "!!! Could not open autofile 01_balbc_Kupffer_ATAC_control_young_balb10B_TDT_JSS_20161013/01_balbc_Kupffer_ATAC_control_young_balb10B_TDT_JSS_20161013.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘01_balbc_Kupffer_ATAC_control_young_balb10B_TDT_JSS_20161013/01_balbc_Kupffer_ATAC_control_young_balb10B_TDT_JSS_20161013.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 51\n",
      "!!! Could not open autofile 01_balbc_Kupffer_ATAC_control_young_balb10C_TDT_JSS_20161013/01_balbc_Kupffer_ATAC_control_young_balb10C_TDT_JSS_20161013.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘01_balbc_Kupffer_ATAC_control_young_balb10C_TDT_JSS_20161013/01_balbc_Kupffer_ATAC_control_young_balb10C_TDT_JSS_20161013.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 57\n",
      "!!! Could not open autofile 02_aj_Kupffer_ATAC_control_young_aj10A_TDT_JSS_20161013/02_aj_Kupffer_ATAC_control_young_aj10A_TDT_JSS_20161013.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘02_aj_Kupffer_ATAC_control_young_aj10A_TDT_JSS_20161013/02_aj_Kupffer_ATAC_control_young_aj10A_TDT_JSS_20161013.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 56\n",
      "!!! Could not open autofile 02_aj_Kupffer_ATAC_control_young_aj10B_TDT_JSS_20161013/02_aj_Kupffer_ATAC_control_young_aj10B_TDT_JSS_20161013.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘02_aj_Kupffer_ATAC_control_young_aj10B_TDT_JSS_20161013/02_aj_Kupffer_ATAC_control_young_aj10B_TDT_JSS_20161013.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 54\n",
      "!!! Could not open autofile 02_aj_Kupffer_ATAC_control_young_aj10C_TDT_JSS_20161013/02_aj_Kupffer_ATAC_control_young_aj10C_TDT_JSS_20161013.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘02_aj_Kupffer_ATAC_control_young_aj10C_TDT_JSS_20161013/02_aj_Kupffer_ATAC_control_young_aj10C_TDT_JSS_20161013.ucsc.bigWig’: No such file or directory\n",
      "\n",
      "\tAll finished: you will want to upload the following hub URL:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-control3/hub.txt\n",
      "\n",
      "\tIf loading to the Wash U Epigenome Browser, use:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-control3/washU.hub.txt\n",
      "\n",
      "\n",
      "\tTo edit track settings, edit files in:\n",
      "\t\t/homer_data/www/html/hubs//C57Kupffer-ATAC-control3\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/h1bennet/strains/data/ATAC/control/\n",
    "tag_dir=$(ls -d *)\n",
    "echo $tag_dir\n",
    "makeMultiWigHub.pl C57Kupffer-ATAC-control3 mm10 -url http://homer.ucsd.edu/hubs/ -webdir /homer_data/www/html/hubs/ -d $tag_dir -force #&>/dev/null &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425 00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919 00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919 00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921 01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3C_JSS_TDT_160926 01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3D_JSS_TDT_160928 02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3A_JSS_TDT_160926 02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3B_JSS_TDT_160926\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Colors that will be used (change with -color or -gradient options):\n",
      "\tIndex\tColor\tNeg. color\tTag Directory\n",
      "\t1\t255,150,150\t(255,180,180)\t00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425\n",
      "\t2\t150,150,255\t(180,180,255)\t00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919\n",
      "\t3\t150,255,150\t(180,255,180)\t00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919\n",
      "\t4\t255,200,150\t(255,210,180)\t00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921\n",
      "\t5\t150,255,220\t(180,255,240)\t01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3C_JSS_TDT_160926\n",
      "\t6\t200,150,255\t(210,180,255)\t01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3D_JSS_TDT_160928\n",
      "\t7\t200,200,150\t(210,210,170)\t02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3A_JSS_TDT_160926\n",
      "\t8\t150,200,200\t(170,210,210)\t02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3B_JSS_TDT_160926\n",
      "\n",
      "\n",
      "\tOnce finished, you will want to upload the following hub URL:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-AMLN3/hub.txt\n",
      "\n",
      "\tIf loading to the Wash U Epigenome Browser, use:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-AMLN3/washU.hub.txt\n",
      "\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 56\n",
      "!!! Could not open autofile 00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425/00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425/00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 58\n",
      "!!! Could not open autofile 00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919/00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919/00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 59\n",
      "!!! Could not open autofile 00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919/00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919/00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN136C_JSS_TDT_160919.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 58\n",
      "!!! Could not open autofile 00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921/00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921/00_NCoRWT_Kupffer_ATAC_AMLNDiet_30week_LN141A_JSS_TDT_160921.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 57\n",
      "!!! Could not open autofile 01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3C_JSS_TDT_160926/01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3C_JSS_TDT_160926.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3C_JSS_TDT_160926/01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3C_JSS_TDT_160926.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 56\n",
      "!!! Could not open autofile 01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3D_JSS_TDT_160928/01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3D_JSS_TDT_160928.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3D_JSS_TDT_160928/01_balbc_Kupffer_ATAC_AMLNDiet_30week_Balb3D_JSS_TDT_160928.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 56\n",
      "!!! Could not open autofile 02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3A_JSS_TDT_160926/02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3A_JSS_TDT_160926.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3A_JSS_TDT_160926/02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3A_JSS_TDT_160926.ucsc.bigWig’: No such file or directory\n",
      "print() on closed filehandle HUBLIST at /bioinformatics/homer/bin/makeMultiWigHub.pl line 388, <IN> line 28.\n",
      "\n",
      "\tVisualization fragment length = 56\n",
      "!!! Could not open autofile 02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3B_JSS_TDT_160926/02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3B_JSS_TDT_160926.ucsc.bigWig for writing!!!\n",
      "mv: cannot stat ‘02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3B_JSS_TDT_160926/02_aj_Kupffer_ATAC_AMLNDiet_30week_AJ3B_JSS_TDT_160926.ucsc.bigWig’: No such file or directory\n",
      "\n",
      "\tAll finished: you will want to upload the following hub URL:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-AMLN3/hub.txt\n",
      "\n",
      "\tIf loading to the Wash U Epigenome Browser, use:\n",
      "\t\thttp://homer.ucsd.edu/hubs//C57Kupffer-ATAC-AMLN3/washU.hub.txt\n",
      "\n",
      "\n",
      "\tTo edit track settings, edit files in:\n",
      "\t\t/homer_data/www/html/hubs//C57Kupffer-ATAC-AMLN3\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/h1bennet/strains/data/ATAC/AMLN_30week/\n",
    "tag_dir=$(ls -d *)\n",
    "echo $tag_dir\n",
    "makeMultiWigHub.pl C57Kupffer-ATAC-AMLN3 mm10 -url http://homer.ucsd.edu/hubs/ -webdir /homer_data/www/html/hubs/ -d $tag_dir -force #&>/dev/null &"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Call Peaks"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 1: Unfiltered peaks from HOMER"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Control"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "File names:\n",
    "- 00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119/\n",
    "- 00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119_round3\n",
      "00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916_round3\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/hmummey/data/ATAC/raw_peaks/Control\n",
    "for data in ./*C57BL6*; do\n",
    "    strain=\"$(cut -d '_' -f2 <<<\"$data\")\"\n",
    "    treatment=\"$(cut -d '_' -f5 <<<\"$data\")\"    \n",
    "    name=\"$(cut -d '/' -f2 <<<\"$data\")_round3\"\n",
    "    echo $name\n",
    "    findPeaks $data -style factor -L 0 -C 0 -fdr 0.9 -o /home/hmummey/data/ATAC/init_peak_calling/${name}.txt -size 200 &>/dev/null &\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119',\n",
       "       '00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916'],\n",
       "      dtype='<U57')"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tag_diretory_array = np.array(fnmatch.filter(listdir('/home/hmummey/data/ATAC/raw_peaks/Control'), '*C57BL6*'))\n",
    "tag_diretory_array.sort()\n",
    "tag_diretory_array"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### AMLN"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "File names:\n",
    "- 00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425/\n",
    "- 00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425_round3\n",
      "00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919_round3\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "cd /home/hmummey/data/ATAC/raw_peaks/AMLN_30week\n",
    "for data in ./*AMLNDiet_2*; do\n",
    "    strain=\"$(cut -d '_' -f2 <<<\"$data\")\"\n",
    "    treatment=\"$(cut -d '_' -f5 <<<\"$data\")\"    \n",
    "    name=\"$(cut -d '/' -f2 <<<\"$data\")_round3\"\n",
    "    echo $name\n",
    "    findPeaks $data -style factor -L 0 -C 0 -fdr 0.9 -o /home/hmummey/data/ATAC/init_peak_calling/${name}.txt -size 200 &>/dev/null &\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425',\n",
       "       '00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919'],\n",
       "      dtype='<U60')"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tag_diretory_array2 = np.array(fnmatch.filter(listdir('/home/hmummey/data/ATAC/raw_peaks/AMLN_30week'), '*AMLNDiet_2*'))\n",
    "tag_diretory_array2.sort()\n",
    "tag_diretory_array2"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Creating a file with all chip_names (control and AMLN)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['C57BL6_control', 'NCoRWT_AMLNDiet_2'], dtype='<U17')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chip_names = []\n",
    "for i in tag_diretory_array:\n",
    "    strain = i.split('_')[1]\n",
    "    treat = i.split('_')[4]\n",
    "    chip_names.append('_'.join([strain, treat]))\n",
    "    \n",
    "for i in tag_diretory_array2:\n",
    "    strain = i.split('_')[1]\n",
    "    treat = i.split('_')[4]\n",
    "    weeks = i.split('_')[5][0]\n",
    "    chip_names.append('_'.join([strain, treat, weeks]))\n",
    "    \n",
    "chip_names = np.unique(np.array(chip_names))\n",
    "chip_names"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "chip_file = '/home/hmummey/data/ATAC/chip_names_list_round3.txt' #exactly the same as prev file\n",
    "with open(chip_file, 'w') as f:\n",
    "    for c in chip_names:\n",
    "        f.write(c + '\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Step 2: Reproducible peaks from IDR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C57BL6_control\n",
      "./00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119_round3.txt ./00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916_round3.txt\n",
      "\n",
      "Performing IDR analysis on the following samples: ./00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119_round3.txt, ./00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916_round3.txt\n",
      "Output files will be written to: ../IDR/C57BL6_control_round3\n",
      "Using the following IDR threshold: 0.05\n",
      "Peaks will be ranked using: findPeaks Score\n",
      "Other available scoreColumns: ['Normalized Tag Count' 'focus ratio' 'findPeaks Score' 'Score']\n",
      "idr --samples ../IDR/C57BL6_control_round3/00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119_round3.narrowPeak ../IDR/C57BL6_control_round3/00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916_round3.narrowPeak --output-file ../IDR/C57BL6_control_round3/00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119_round3_00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916_round3_idr.out --plot --idr-threshold 0.05 &\n",
      "NCoRWT_AMLNDiet_2\n",
      "./00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425_round3.txt ./00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919_round3.txt\n",
      "\n",
      "Performing IDR analysis on the following samples: ./00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425_round3.txt, ./00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919_round3.txt\n",
      "Output files will be written to: ../IDR/NCoRWT_AMLNDiet_2_round3\n",
      "Using the following IDR threshold: 0.05\n",
      "Peaks will be ranked using: findPeaks Score\n",
      "Other available scoreColumns: ['Normalized Tag Count' 'focus ratio' 'findPeaks Score' 'Score']\n",
      "idr --samples ../IDR/NCoRWT_AMLNDiet_2_round3/00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425_round3.narrowPeak ../IDR/NCoRWT_AMLNDiet_2_round3/00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919_round3.narrowPeak --output-file ../IDR/NCoRWT_AMLNDiet_2_round3/00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425_round3_00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919_round3_idr.out --plot --idr-threshold 0.05 &\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [2.26 1.42 0.92 0.65]\n",
      "Number of reported peaks - 54774/94248 (58.1%)\n",
      "\n",
      "Number of peaks passing IDR cutoff of 0.05 - 54774/94248 (58.1%)\n",
      "\n",
      "/home/zes017/run_idr_homerPeaks.py:178: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  narrowPeakFrame['pValue'] = -1\n",
      "Initial parameter values: [0.10 1.00 0.20 0.50]\n",
      "Final parameter values: [1.90 1.34 0.87 0.62]\n",
      "Number of reported peaks - 32825/65689 (50.0%)\n",
      "\n",
      "Number of peaks passing IDR cutoff of 0.05 - 32825/65689 (50.0%)\n",
      "\n",
      "/home/zes017/run_idr_homerPeaks.py:178: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  narrowPeakFrame['pValue'] = -1\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "cd /home/hmummey/data/ATAC/init_peak_calling\n",
    "while IFS='' read -r IP\n",
    "do\n",
    "    strain=\"$(cut -d '_' -f1 <<<\"$IP\")\"\n",
    "    treatment=\"$(cut -d '_' -f2 <<<\"$IP\")\"\n",
    "    round=\"round3\"\n",
    "    #name=\"$(cut -d '/' -f2 <<<\"$data\" | cut -d '.' -f1)\"\n",
    "\n",
    "    echo $IP\n",
    "    echo ./*${strain}*${treatment}*${round}*\n",
    "    #echo *${strain}*${treatment}*\n",
    "    echo \n",
    "\n",
    "    #mkdir ../IDR/${IP}_${round}\n",
    "    \n",
    "    python3 /home/zes017/run_idr_homerPeaks.py -threshold 0.05 ./*${strain}*${treatment}*${round}* ../IDR/${IP}_${round} #&>/dev/null &\n",
    "done < '/home/hmummey/data/ATAC/chip_names_list_round3.txt'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RESUME HERE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Count Tags of Histone Marks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/hmummey/data/ATAC/annotated_peak_calling/round3/00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119_round3_00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916_round3_idr.tsv\n",
      "C57\n",
      "control\n",
      "cont\n",
      "/home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423\n",
      "\n",
      "/home/hmummey/data/ATAC/annotated_peak_calling/round3/00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425_round3_00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919_round3_idr.tsv\n",
      "C57\n",
      "AMLNDiet\n",
      "AMLN\n",
      "/home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\tPeak file = /home/hmummey/data/ATAC/annotated_peak_calling/round3/00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119_round3_00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916_round3_idr.tsv\n",
      "\tGenome = mm10\n",
      "\tOrganism = mouse\n",
      "\tPeak Region set to 1000\n",
      "\tWill normalize tag counts to 1e7 per experiment\n",
      "\tTag Directories:\n",
      "\t\t/home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915\n",
      "\t\t/home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 0\n",
      "\t\tpeakfile formatted lines: 54773\n",
      "\t\tDuplicated Peak IDs: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 54773\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tResizing peaks...\n",
      "\tReading Positions...\n",
      "\t-----------------------\n",
      "\tFinding Closest TSS...\n",
      "\tAnnotating:........................\n",
      "\t\tAnnotation\tNumber of peaks\tTotal size (bp)\tLog2 Ratio (obs/exp)\tLogP enrichment (+values depleted)\n",
      "\t\t3UTR\t692.0\t20049228\t0.779\t-88.948\n",
      "\t\tmiRNA\t0.0\t24864\t-0.585\t0.500\n",
      "\t\tncRNA\t169.0\t3372538\t1.317\t-56.146\n",
      "\t\tTTS\t946.0\t27011103\t0.800\t-126.820\n",
      "\t\tpseudo\t9.0\t528823\t-0.241\t0.965\n",
      "\t\tExon\t1270.0\t34014877\t0.892\t-206.494\n",
      "\t\tIntron\t21986.0\t942718221\t0.213\t-363.650\n",
      "\t\tIntergenic\t18041.0\t1662058715\t-0.890\t8833.428\n",
      "\t\tPromoter\t11171.0\t29761709\t4.222\t-23227.901\n",
      "\t\t5UTR\t472.0\t2270504\t3.369\t-681.499\n",
      "\t\tsnoRNA\t1.0\t19\t11.353\t-7.870\n",
      "\t\trRNA\t4.0\t5631\t5.142\t-11.980\n",
      "\tNOTE: If this part takes more than 2 minutes, there is a good chance\n",
      "\t\tyour machine ran out of memory: consider hitting ctrl+C and rerunning\n",
      "\t\tthe command with \"-noann\"\n",
      "\tTo capture annotation stats in a file, use \"-annStats <filename>\" next time\n",
      "\tAnnotating:.............................\n",
      "\t\tAnnotation\tNumber of peaks\tTotal size (bp)\tLog2 Ratio (obs/exp)\tLogP enrichment (+values depleted)\n",
      "\t\t3UTR\t692.0\t20049228\t0.779\t-89.032\n",
      "\t\tOther\t71.0\t7955052\t-1.172\t33.862\n",
      "\t\tRC?\t0.0\t10979\t-0.288\t0.221\n",
      "\t\tRNA\t1.0\t114021\t-1.197\t1.101\n",
      "\t\tmiRNA\t0.0\t24864\t-0.585\t0.500\n",
      "\t\tncRNA\t169.0\t3372538\t1.317\t-56.175\n",
      "\t\tTTS\t946.0\t27011103\t0.800\t-126.937\n",
      "\t\tLINE\t1587.0\t543786362\t-2.785\t7224.507\n",
      "\t\tLINE?\t0.0\t8168\t-0.219\t0.164\n",
      "\t\tsrpRNA\t1.0\t43307\t0.199\t-0.542\n",
      "\t\tSINE\t1971.0\t196568429\t-1.004\t652.877\n",
      "\t\tRC\t1.0\t65629\t-0.401\t0.478\n",
      "\t\ttRNA\t15.0\t266845\t1.483\t-7.667\n",
      "\t\tDNA?\t0.0\t142206\t-1.949\t2.860\n",
      "\t\tpseudo\t9.0\t528823\t-0.241\t0.964\n",
      "\t\tDNA\t332.0\t28691693\t-0.798\t65.067\n",
      "\t\tExon\t1268.0\t34014877\t0.890\t-205.405\n",
      "\t\tIntron\t17207.0\t598538808\t0.515\t-1308.465\n",
      "\t\tIntergenic\t13248.0\t828484831\t-0.331\t529.712\n",
      "\t\tPromoter\t11170.0\t29761709\t4.222\t-23227.576\n",
      "\t\t5UTR\t472.0\t2270504\t3.370\t-681.622\n",
      "\t\tsnoRNA\t1.0\t19\t11.354\t-7.870\n",
      "\t\tLTR?\t3.0\t192563\t-0.369\t0.779\n",
      "\t\tscRNA\t4.0\t602998\t-1.600\t4.970\n",
      "\t\tCpG-Island\t1448.0\t3194561\t4.494\t-3149.065\n",
      "\t\tLow_complexity\t136.0\t19404608\t-1.521\t114.461\n",
      "\t\tLTR\t3780.0\t313085789\t-0.736\t656.390\n",
      "\t\tSimple_repeat\t155.0\t57561961\t-2.901\t703.712\n",
      "\t\tsnRNA\t5.0\t237524\t0.066\t-0.655\n",
      "\t\tUnknown\t12.0\t2407459\t-2.013\t21.586\n",
      "\t\tSINE?\t0.0\t29758\t-0.677\t0.599\n",
      "\t\tSatellite\t48.0\t4603878\t-0.948\t15.226\n",
      "\t\trRNA\t21.0\t166488\t2.649\t-23.188\n",
      "\tCounting Tags in Peaks from each directory...\n",
      "\tRatio for /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 : 1.05618536475626\n",
      "\tRatio for /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423 : 0.442167204732569\n",
      "\tOrganism: mouse\n",
      "\tLoading Gene Informaiton...\n",
      "\tOutputing Annotation File...\n",
      "\tDone annotating peaks file\n",
      "\n",
      "\n",
      "\tPeak file = /home/hmummey/data/ATAC/annotated_peak_calling/round3/00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425_round3_00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919_round3_idr.tsv\n",
      "\tGenome = mm10\n",
      "\tOrganism = mouse\n",
      "\tPeak Region set to 1000\n",
      "\tWill normalize tag counts to 1e7 per experiment\n",
      "\tTag Directories:\n",
      "\t\t/home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423\n",
      "\t\t/home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423\n",
      "\tPeak/BED file conversion summary:\n",
      "\t\tBED/Header formatted lines: 0\n",
      "\t\tpeakfile formatted lines: 32824\n",
      "\t\tDuplicated Peak IDs: 0\n",
      "\n",
      "\tPeak File Statistics:\n",
      "\t\tTotal Peaks: 32824\n",
      "\t\tRedundant Peak IDs: 0\n",
      "\t\tPeaks lacking information: 0 (need at least 5 columns per peak)\n",
      "\t\tPeaks with misformatted coordinates: 0 (should be integer)\n",
      "\t\tPeaks with misformatted strand: 0 (should be either +/- or 0/1)\n",
      "\n",
      "\tPeak file looks good!\n",
      "\n",
      "\tResizing peaks...\n",
      "\tReading Positions...\n",
      "\t-----------------------\n",
      "\tFinding Closest TSS...\n",
      "\tAnnotating:........................\n",
      "\t\tAnnotation\tNumber of peaks\tTotal size (bp)\tLog2 Ratio (obs/exp)\tLogP enrichment (+values depleted)\n",
      "\t\t3UTR\t281.0\t20049228\t0.217\t-4.941\n",
      "\t\tmiRNA\t0.0\t24864\t-0.378\t0.300\n",
      "\t\tncRNA\t87.0\t3372538\t1.097\t-22.405\n",
      "\t\tTTS\t438.0\t27011103\t0.427\t-20.277\n",
      "\t\tpseudo\t5.0\t528823\t-0.351\t0.949\n",
      "\t\tExon\t500.0\t34014877\t0.286\t-11.678\n",
      "\t\tIntron\t11857.0\t942718221\t0.061\t-18.700\n",
      "\t\tIntergenic\t9835.0\t1662058715\t-1.027\t6497.048\n",
      "\t\tPromoter\t9531.0\t29761709\t4.731\t-23527.660\n",
      "\t\t5UTR\t281.0\t2270504\t3.360\t-405.343\n",
      "\t\tsnoRNA\t0.0\t19\t-0.000\t0.000\n",
      "\t\trRNA\t4.0\t5631\t5.881\t-13.992\n",
      "\tNOTE: If this part takes more than 2 minutes, there is a good chance\n",
      "\t\tyour machine ran out of memory: consider hitting ctrl+C and rerunning\n",
      "\t\tthe command with \"-noann\"\n",
      "\tTo capture annotation stats in a file, use \"-annStats <filename>\" next time\n",
      "\tAnnotating:...........................\n",
      "\t\tAnnotation\tNumber of peaks\tTotal size (bp)\tLog2 Ratio (obs/exp)\tLogP enrichment (+values depleted)\n",
      "\t\t3UTR\t281.0\t20049228\t0.218\t-4.956\n",
      "\t\tOther\t35.0\t7955052\t-1.454\t27.924\n",
      "\t\tRC?\t0.0\t10979\t-0.179\t0.132\n",
      "\t\tRNA\t3.0\t114021\t1.126\t-1.830\n",
      "\t\tmiRNA\t0.0\t24864\t-0.378\t0.300\n",
      "\t\tncRNA\t87.0\t3372538\t1.098\t-22.421\n",
      "\t\tTTS\t438.0\t27011103\t0.428\t-20.317\n",
      "\t\tLINE\t836.0\t543776315\t-2.971\t4582.700\n",
      "\t\tLINE?\t0.0\t8168\t-0.135\t0.098\n",
      "\t\tsrpRNA\t0.0\t43307\t-0.606\t0.522\n",
      "\t\tSINE\t888.0\t196566416\t-1.416\t649.177\n",
      "\t\tRC\t0.0\t65629\t-0.841\t0.791\n",
      "\t\ttRNA\t7.0\t266576\t1.123\t-3.092\n",
      "\t\tDNA?\t1.0\t142206\t-0.777\t0.716\n",
      "\t\tpseudo\t5.0\t528823\t-0.350\t0.948\n",
      "\t\tDNA\t190.0\t28691693\t-0.864\t45.168\n",
      "\t\tExon\t500.0\t34014877\t0.286\t-11.711\n",
      "\t\tIntron\t9376.0\t598538808\t0.378\t-393.552\n",
      "\t\tIntergenic\t7297.0\t828469732\t-0.453\t556.426\n",
      "\t\tPromoter\t9530.0\t29761709\t4.731\t-23526.961\n",
      "\t\t5UTR\t281.0\t2270504\t3.360\t-405.429\n",
      "\t\tsnoRNA\t0.0\t19\t-0.000\t0.000\n",
      "\t\tLTR?\t3.0\t192563\t0.370\t-0.893\n",
      "\t\tscRNA\t1.0\t602998\t-2.862\t5.157\n",
      "\t\tCpG-Island\t996.0\t3193893\t4.693\t-2301.062\n",
      "\t\tLow_complexity\t54.0\t19402818\t-2.115\t103.872\n",
      "\t\tLTR\t1889.0\t313078376\t-0.998\t641.400\n",
      "\t\tSimple_repeat\t74.0\t57560484\t-3.229\t463.092\n",
      "\t\tsnRNA\t1.0\t237524\t-1.518\t1.512\n",
      "\t\tUnknown\t6.0\t2407459\t-2.274\t15.177\n",
      "\t\tSINE?\t0.0\t29758\t-0.442\t0.359\n",
      "\t\tSatellite\t35.0\t4603878\t-0.665\t6.137\n",
      "\t\trRNA\t10.0\t166488\t2.317\t-9.949\n",
      "\tCounting Tags in Peaks from each directory...\n",
      "\tRatio for /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 : 0.53334107033446\n",
      "\tRatio for /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423 : 0.519990812802319\n",
      "\tOrganism: mouse\n",
      "\tLoading Gene Informaiton...\n",
      "\tOutputing Annotation File...\n",
      "\tDone annotating peaks file\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for peak_file in /home/hmummey/data/ATAC/annotated_peak_calling/round3/*_idr.tsv; do\n",
    "    echo $peak_file\n",
    "\n",
    "    #strain=\"$(echo $peak_file | cut -d '/' -f7 | cut -d '_' -f2)\"\n",
    "    strain=\"C57\"\n",
    "    treat=\"$(echo $peak_file | cut -d '/' -f8 | cut -d '_' -f5)\"\n",
    "    \n",
    "    #treat (AMLNDiet) doesn't match his dir name: /home/h1bennet/strains/data/H3K27Ac/AMLN_30week\n",
    "    #but it does match the file names: 00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423\n",
    "    #so modifying this a bit (cut down and then use * later)\n",
    "    treat2=\"$(echo $treat | cut -b 1-4)\"\n",
    "\n",
    "    echo $strain\n",
    "    echo $treat\n",
    "    echo $treat2\n",
    "    \n",
    "    files=''\n",
    "    for h in 'H3K27Ac'; do\n",
    "        files+=\"$(echo /home/h1bennet/strains/data/$h/$treat2*/*${strain}*${h}*)\"\n",
    "\n",
    "        files+=' '\n",
    "    done\n",
    "    echo $files\n",
    "    echo\n",
    "    annotatePeaks.pl $peak_file mm10 -size -500,500 -norm 1e7 -d $files > /home/hmummey/data/ATAC/annotated_peak_calling/round3/${strain}_${treat}_round3_annot.txt #2>/dev/null &\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "strain = 'C57_control_round3'\n",
    "path = '/home/hmummey/data/ATAC/annotated_peak_calling/round3/'\n",
    "annot_file = path + strain + '_annot.txt'\n",
    "annot_df = pd.read_csv(annot_file, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "54773\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/round3/00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119_round3_00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916_round3_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423)</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "      <th>Focus Ratio/Region Size</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Detailed Annotation</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>...</th>\n",
       "      <th>Entrez ID</th>\n",
       "      <th>Nearest Unigene</th>\n",
       "      <th>Nearest Refseq</th>\n",
       "      <th>Nearest Ensembl</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Alias</th>\n",
       "      <th>Gene Description</th>\n",
       "      <th>Gene Type</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 Tag Count in 1000 bp (9468035.0 Total, normalization factor = 1.06, effective total = 1e7)</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423 Tag Count in 1000 bp (22615879.0 Total, normalization factor = 0.44, effective total = 1e7)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>457_20603</td>\n",
       "      <td>chr2</td>\n",
       "      <td>145952312</td>\n",
       "      <td>145953312</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>63.0</td>\n",
       "      <td>intron (NM_175280, intron 6 of 6)</td>\n",
       "      <td>intron (NM_175280, intron 6 of 6)</td>\n",
       "      <td>18028.0</td>\n",
       "      <td>...</td>\n",
       "      <td>78774.0</td>\n",
       "      <td>Mm.136347</td>\n",
       "      <td>NM_175280</td>\n",
       "      <td>ENSMUSG00000037143</td>\n",
       "      <td>Cfap61</td>\n",
       "      <td>4930529M08Rik|BB178539|BC024760</td>\n",
       "      <td>cilia and flagella associated protein 61</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>15.84</td>\n",
       "      <td>7.07</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>457_9613</td>\n",
       "      <td>chr4</td>\n",
       "      <td>123920875</td>\n",
       "      <td>123921875</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>104.0</td>\n",
       "      <td>intron (NM_017475, intron 3 of 6)</td>\n",
       "      <td>intron (NM_017475, intron 3 of 6)</td>\n",
       "      <td>3942.0</td>\n",
       "      <td>...</td>\n",
       "      <td>54170.0</td>\n",
       "      <td>Mm.220922</td>\n",
       "      <td>NM_017475</td>\n",
       "      <td>ENSMUSG00000028646</td>\n",
       "      <td>Rragc</td>\n",
       "      <td>AU041672|Gtr2|RAGC|TIB929|YGR163W</td>\n",
       "      <td>Ras-related GTP binding C</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>102.45</td>\n",
       "      <td>188.36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>457_1301</td>\n",
       "      <td>chr5</td>\n",
       "      <td>110839354</td>\n",
       "      <td>110840354</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>204.0</td>\n",
       "      <td>promoter-TSS (NM_016681)</td>\n",
       "      <td>promoter-TSS (NM_016681)</td>\n",
       "      <td>-77.0</td>\n",
       "      <td>...</td>\n",
       "      <td>100900.0</td>\n",
       "      <td>Mm.200231</td>\n",
       "      <td>NM_153571</td>\n",
       "      <td>ENSMUSG00000043510</td>\n",
       "      <td>Hscb</td>\n",
       "      <td>AI325508|AW049829|Hsc20</td>\n",
       "      <td>HscB iron-sulfur cluster co-chaperone</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>240.81</td>\n",
       "      <td>193.23</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>457_6548</td>\n",
       "      <td>chr14</td>\n",
       "      <td>105434993</td>\n",
       "      <td>105435993</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>125.0</td>\n",
       "      <td>intron (NR_045859, intron 2 of 2)</td>\n",
       "      <td>MTD|LTR|ERVL-MaLR</td>\n",
       "      <td>7934.0</td>\n",
       "      <td>...</td>\n",
       "      <td>71362.0</td>\n",
       "      <td>Mm.410781</td>\n",
       "      <td>NR_045859</td>\n",
       "      <td>ENSMUSG00000115548</td>\n",
       "      <td>5430440P10Rik</td>\n",
       "      <td>-</td>\n",
       "      <td>RIKEN cDNA 5430440P10 gene</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>35.91</td>\n",
       "      <td>54.39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>457_4608</td>\n",
       "      <td>chr8</td>\n",
       "      <td>107029706</td>\n",
       "      <td>107030706</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>144.0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>-1120.0</td>\n",
       "      <td>...</td>\n",
       "      <td>116733.0</td>\n",
       "      <td>Mm.236004</td>\n",
       "      <td>NM_126165</td>\n",
       "      <td>ENSMUSG00000031913</td>\n",
       "      <td>Vps4a</td>\n",
       "      <td>4930589C15Rik|AI325971|AW553189</td>\n",
       "      <td>vacuolar protein sorting 4A</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>54.92</td>\n",
       "      <td>38.91</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/round3/00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119_round3_00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916_round3_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423)  \\\n",
       "0                                          457_20603                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "1                                           457_9613                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "2                                           457_1301                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "3                                           457_6548                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "4                                           457_4608                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "\n",
       "     Chr      Start        End Strand  Peak Score  Focus Ratio/Region Size  \\\n",
       "0   chr2  145952312  145953312      +        1000                     63.0   \n",
       "1   chr4  123920875  123921875      +        1000                    104.0   \n",
       "2   chr5  110839354  110840354      +        1000                    204.0   \n",
       "3  chr14  105434993  105435993      +        1000                    125.0   \n",
       "4   chr8  107029706  107030706      +        1000                    144.0   \n",
       "\n",
       "                          Annotation                Detailed Annotation  \\\n",
       "0  intron (NM_175280, intron 6 of 6)  intron (NM_175280, intron 6 of 6)   \n",
       "1  intron (NM_017475, intron 3 of 6)  intron (NM_017475, intron 3 of 6)   \n",
       "2           promoter-TSS (NM_016681)           promoter-TSS (NM_016681)   \n",
       "3  intron (NR_045859, intron 2 of 2)                  MTD|LTR|ERVL-MaLR   \n",
       "4                         Intergenic                         Intergenic   \n",
       "\n",
       "   Distance to TSS  ... Entrez ID  Nearest Unigene Nearest Refseq  \\\n",
       "0          18028.0  ...   78774.0        Mm.136347      NM_175280   \n",
       "1           3942.0  ...   54170.0        Mm.220922      NM_017475   \n",
       "2            -77.0  ...  100900.0        Mm.200231      NM_153571   \n",
       "3           7934.0  ...   71362.0        Mm.410781      NR_045859   \n",
       "4          -1120.0  ...  116733.0        Mm.236004      NM_126165   \n",
       "\n",
       "      Nearest Ensembl      Gene Name                         Gene Alias  \\\n",
       "0  ENSMUSG00000037143         Cfap61    4930529M08Rik|BB178539|BC024760   \n",
       "1  ENSMUSG00000028646          Rragc  AU041672|Gtr2|RAGC|TIB929|YGR163W   \n",
       "2  ENSMUSG00000043510           Hscb            AI325508|AW049829|Hsc20   \n",
       "3  ENSMUSG00000115548  5430440P10Rik                                  -   \n",
       "4  ENSMUSG00000031913          Vps4a    4930589C15Rik|AI325971|AW553189   \n",
       "\n",
       "                           Gene Description       Gene Type  \\\n",
       "0  cilia and flagella associated protein 61  protein-coding   \n",
       "1                 Ras-related GTP binding C  protein-coding   \n",
       "2     HscB iron-sulfur cluster co-chaperone  protein-coding   \n",
       "3                RIKEN cDNA 5430440P10 gene           ncRNA   \n",
       "4               vacuolar protein sorting 4A  protein-coding   \n",
       "\n",
       "  /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 Tag Count in 1000 bp (9468035.0 Total, normalization factor = 1.06, effective total = 1e7)  \\\n",
       "0                                              15.84                                                                                                                                         \n",
       "1                                             102.45                                                                                                                                         \n",
       "2                                             240.81                                                                                                                                         \n",
       "3                                              35.91                                                                                                                                         \n",
       "4                                              54.92                                                                                                                                         \n",
       "\n",
       "   /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423 Tag Count in 1000 bp (22615879.0 Total, normalization factor = 0.44, effective total = 1e7)  \n",
       "0                                               7.07                                                                                                                                          \n",
       "1                                             188.36                                                                                                                                          \n",
       "2                                             193.23                                                                                                                                          \n",
       "3                                              54.39                                                                                                                                          \n",
       "4                                              38.91                                                                                                                                          \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(annot_df))\n",
    "annot_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_control_young_C571A_170915</th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_control_young_C572A_180423</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4.073820</td>\n",
       "      <td>3.012569</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6.692790</td>\n",
       "      <td>7.564988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7.917730</td>\n",
       "      <td>7.601622</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5.205940</td>\n",
       "      <td>5.791554</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5.805292</td>\n",
       "      <td>5.318678</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   00_C57_Kupffer_H3K27Ac_control_young_C571A_170915  \\\n",
       "0                                           4.073820   \n",
       "1                                           6.692790   \n",
       "2                                           7.917730   \n",
       "3                                           5.205940   \n",
       "4                                           5.805292   \n",
       "\n",
       "   00_C57_Kupffer_H3K27Ac_control_young_C572A_180423  \n",
       "0                                           3.012569  \n",
       "1                                           7.564988  \n",
       "2                                           7.601622  \n",
       "3                                           5.791554  \n",
       "4                                           5.318678  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "histone_tags1 = annot_df.iloc[:,-2:]\n",
    "histone_tags1.columns = [i.split(' Tag')[0].split('/')[-1] for i in histone_tags1.columns.values]\n",
    "histone_tags2 = np.log2(histone_tags1+1) #convert to log2 scale\n",
    "histone_tags2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_control_young_C571A_170915</th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_control_young_C572A_180423</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>54773.000000</td>\n",
       "      <td>54773.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>4.989888</td>\n",
       "      <td>4.791526</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>1.985577</td>\n",
       "      <td>2.159701</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>3.393691</td>\n",
       "      <td>3.090853</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>5.121015</td>\n",
       "      <td>4.936873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>6.601548</td>\n",
       "      <td>6.559186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>15.400366</td>\n",
       "      <td>14.988302</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       00_C57_Kupffer_H3K27Ac_control_young_C571A_170915  \\\n",
       "count                                       54773.000000   \n",
       "mean                                            4.989888   \n",
       "std                                             1.985577   \n",
       "min                                             0.000000   \n",
       "25%                                             3.393691   \n",
       "50%                                             5.121015   \n",
       "75%                                             6.601548   \n",
       "max                                            15.400366   \n",
       "\n",
       "       00_C57_Kupffer_H3K27Ac_control_young_C572A_180423  \n",
       "count                                       54773.000000  \n",
       "mean                                            4.791526  \n",
       "std                                             2.159701  \n",
       "min                                             0.000000  \n",
       "25%                                             3.090853  \n",
       "50%                                             4.936873  \n",
       "75%                                             6.559186  \n",
       "max                                            14.988302  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "histone_tags2.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### AMLN Diet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32824\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/round3/00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425_round3_00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919_round3_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423)</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "      <th>Focus Ratio/Region Size</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Detailed Annotation</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>...</th>\n",
       "      <th>Entrez ID</th>\n",
       "      <th>Nearest Unigene</th>\n",
       "      <th>Nearest Refseq</th>\n",
       "      <th>Nearest Ensembl</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Alias</th>\n",
       "      <th>Gene Description</th>\n",
       "      <th>Gene Type</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 Tag Count in 1000 bp (18749728.0 Total, normalization factor = 0.53, effective total = 1e7)</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423 Tag Count in 1000 bp (19231109.0 Total, normalization factor = 0.52, effective total = 1e7)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>868_10955</td>\n",
       "      <td>chr2</td>\n",
       "      <td>91537228</td>\n",
       "      <td>91538228</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>62.0</td>\n",
       "      <td>intron (NM_001165989, intron 1 of 43)</td>\n",
       "      <td>intron (NM_001165989, intron 1 of 43)</td>\n",
       "      <td>-8594.0</td>\n",
       "      <td>...</td>\n",
       "      <td>75786.0</td>\n",
       "      <td>Mm.168478</td>\n",
       "      <td>NM_029437</td>\n",
       "      <td>ENSMUSG00000040549</td>\n",
       "      <td>Ckap5</td>\n",
       "      <td>3110043H24Rik|4930432B04Rik|D730027C18Rik|T256...</td>\n",
       "      <td>cytoskeleton associated protein 5</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>18.13</td>\n",
       "      <td>25.48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>868_8833</td>\n",
       "      <td>chr13</td>\n",
       "      <td>48967669</td>\n",
       "      <td>48968669</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>69.0</td>\n",
       "      <td>promoter-TSS (NR_015601)</td>\n",
       "      <td>promoter-TSS (NR_015601)</td>\n",
       "      <td>57.0</td>\n",
       "      <td>...</td>\n",
       "      <td>68128.0</td>\n",
       "      <td>Mm.426304</td>\n",
       "      <td>NR_015601</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Fam120aos</td>\n",
       "      <td>C030044B11Rik</td>\n",
       "      <td>family with sequence similarity 120A, opposite...</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>155.74</td>\n",
       "      <td>108.16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>868_14942</td>\n",
       "      <td>chr5</td>\n",
       "      <td>147860025</td>\n",
       "      <td>147861025</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>52.0</td>\n",
       "      <td>promoter-TSS (NM_025624)</td>\n",
       "      <td>promoter-TSS (NM_025624)</td>\n",
       "      <td>-103.0</td>\n",
       "      <td>...</td>\n",
       "      <td>66537.0</td>\n",
       "      <td>Mm.332855</td>\n",
       "      <td>NM_025624</td>\n",
       "      <td>ENSMUSG00000029649</td>\n",
       "      <td>Pomp</td>\n",
       "      <td>2510048O06Rik|Ump1</td>\n",
       "      <td>proteasome maturation protein</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>192.54</td>\n",
       "      <td>162.24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>868_16901</td>\n",
       "      <td>chr5</td>\n",
       "      <td>150522108</td>\n",
       "      <td>150523108</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>48.0</td>\n",
       "      <td>promoter-TSS (NM_009765)</td>\n",
       "      <td>promoter-TSS (NM_009765)</td>\n",
       "      <td>-13.0</td>\n",
       "      <td>...</td>\n",
       "      <td>12190.0</td>\n",
       "      <td>Mm.236256</td>\n",
       "      <td>NM_009765</td>\n",
       "      <td>ENSMUSG00000041147</td>\n",
       "      <td>Brca2</td>\n",
       "      <td>Fancd1|RAB163</td>\n",
       "      <td>breast cancer 2, early onset</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>83.20</td>\n",
       "      <td>49.92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>868_13869</td>\n",
       "      <td>chr13</td>\n",
       "      <td>64401616</td>\n",
       "      <td>64402616</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>54.0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>-30437.0</td>\n",
       "      <td>...</td>\n",
       "      <td>105278.0</td>\n",
       "      <td>Mm.74982</td>\n",
       "      <td>NM_053180</td>\n",
       "      <td>ENSMUSG00000021483</td>\n",
       "      <td>Cdk20</td>\n",
       "      <td>4932702G04Rik|AU019450|AW558027|CDCH|Ccrk|PNQL...</td>\n",
       "      <td>cyclin-dependent kinase 20</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>42.13</td>\n",
       "      <td>40.04</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/round3/00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425_round3_00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919_round3_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423)  \\\n",
       "0                                          868_10955                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "1                                           868_8833                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "2                                          868_14942                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "3                                          868_16901                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "4                                          868_13869                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "\n",
       "     Chr      Start        End Strand  Peak Score  Focus Ratio/Region Size  \\\n",
       "0   chr2   91537228   91538228      +        1000                     62.0   \n",
       "1  chr13   48967669   48968669      +        1000                     69.0   \n",
       "2   chr5  147860025  147861025      +        1000                     52.0   \n",
       "3   chr5  150522108  150523108      +        1000                     48.0   \n",
       "4  chr13   64401616   64402616      +        1000                     54.0   \n",
       "\n",
       "                              Annotation  \\\n",
       "0  intron (NM_001165989, intron 1 of 43)   \n",
       "1               promoter-TSS (NR_015601)   \n",
       "2               promoter-TSS (NM_025624)   \n",
       "3               promoter-TSS (NM_009765)   \n",
       "4                             Intergenic   \n",
       "\n",
       "                     Detailed Annotation  Distance to TSS  ... Entrez ID  \\\n",
       "0  intron (NM_001165989, intron 1 of 43)          -8594.0  ...   75786.0   \n",
       "1               promoter-TSS (NR_015601)             57.0  ...   68128.0   \n",
       "2               promoter-TSS (NM_025624)           -103.0  ...   66537.0   \n",
       "3               promoter-TSS (NM_009765)            -13.0  ...   12190.0   \n",
       "4                             Intergenic         -30437.0  ...  105278.0   \n",
       "\n",
       "   Nearest Unigene Nearest Refseq     Nearest Ensembl  Gene Name  \\\n",
       "0        Mm.168478      NM_029437  ENSMUSG00000040549      Ckap5   \n",
       "1        Mm.426304      NR_015601                 NaN  Fam120aos   \n",
       "2        Mm.332855      NM_025624  ENSMUSG00000029649       Pomp   \n",
       "3        Mm.236256      NM_009765  ENSMUSG00000041147      Brca2   \n",
       "4         Mm.74982      NM_053180  ENSMUSG00000021483      Cdk20   \n",
       "\n",
       "                                          Gene Alias  \\\n",
       "0  3110043H24Rik|4930432B04Rik|D730027C18Rik|T256...   \n",
       "1                                      C030044B11Rik   \n",
       "2                                 2510048O06Rik|Ump1   \n",
       "3                                      Fancd1|RAB163   \n",
       "4  4932702G04Rik|AU019450|AW558027|CDCH|Ccrk|PNQL...   \n",
       "\n",
       "                                    Gene Description       Gene Type  \\\n",
       "0                  cytoskeleton associated protein 5  protein-coding   \n",
       "1  family with sequence similarity 120A, opposite...           ncRNA   \n",
       "2                      proteasome maturation protein  protein-coding   \n",
       "3                       breast cancer 2, early onset  protein-coding   \n",
       "4                         cyclin-dependent kinase 20  protein-coding   \n",
       "\n",
       "  /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 Tag Count in 1000 bp (18749728.0 Total, normalization factor = 0.53, effective total = 1e7)  \\\n",
       "0                                              18.13                                                                                                                                                 \n",
       "1                                             155.74                                                                                                                                                 \n",
       "2                                             192.54                                                                                                                                                 \n",
       "3                                              83.20                                                                                                                                                 \n",
       "4                                              42.13                                                                                                                                                 \n",
       "\n",
       "   /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423 Tag Count in 1000 bp (19231109.0 Total, normalization factor = 0.52, effective total = 1e7)  \n",
       "0                                              25.48                                                                                                                                                 \n",
       "1                                             108.16                                                                                                                                                 \n",
       "2                                             162.24                                                                                                                                                 \n",
       "3                                              49.92                                                                                                                                                 \n",
       "4                                              40.04                                                                                                                                                 \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "strain2 = 'C57_AMLNDiet_round3'\n",
    "path = '/home/hmummey/data/ATAC/annotated_peak_calling/round3/'\n",
    "annot_file2 = path + strain2 + '_annot.txt'\n",
    "annot_df2 = pd.read_csv(annot_file2, sep='\\t')\n",
    "\n",
    "print(len(annot_df2))\n",
    "annot_df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423</th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4.257765</td>\n",
       "      <td>4.726831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7.292230</td>\n",
       "      <td>6.770300</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7.596488</td>\n",
       "      <td>7.350851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6.395748</td>\n",
       "      <td>5.670161</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5.430620</td>\n",
       "      <td>5.358959</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423  \\\n",
       "0                                           4.257765      \n",
       "1                                           7.292230      \n",
       "2                                           7.596488      \n",
       "3                                           6.395748      \n",
       "4                                           5.430620      \n",
       "\n",
       "   00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423  \n",
       "0                                           4.726831     \n",
       "1                                           6.770300     \n",
       "2                                           7.350851     \n",
       "3                                           5.670161     \n",
       "4                                           5.358959     "
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "histone_tags12 = annot_df2.iloc[:,-2:]\n",
    "histone_tags12.columns = [i.split(' Tag')[0].split('/')[-1] for i in histone_tags12.columns.values]\n",
    "histone_tags22 = np.log2(histone_tags12+1) #convert to log2 scale\n",
    "histone_tags22.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423</th>\n",
       "      <th>00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>32824.000000</td>\n",
       "      <td>32824.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>5.459177</td>\n",
       "      <td>5.331023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>1.922137</td>\n",
       "      <td>1.861231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>4.131754</td>\n",
       "      <td>4.007196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>5.691255</td>\n",
       "      <td>5.531069</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>6.950352</td>\n",
       "      <td>6.756490</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>15.693564</td>\n",
       "      <td>15.991699</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423  \\\n",
       "count                                       32824.000000      \n",
       "mean                                            5.459177      \n",
       "std                                             1.922137      \n",
       "min                                             0.000000      \n",
       "25%                                             4.131754      \n",
       "50%                                             5.691255      \n",
       "75%                                             6.950352      \n",
       "max                                            15.693564      \n",
       "\n",
       "       00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423  \n",
       "count                                       32824.000000     \n",
       "mean                                            5.331023     \n",
       "std                                             1.861231     \n",
       "min                                             0.000000     \n",
       "25%                                             4.007196     \n",
       "50%                                             5.531069     \n",
       "75%                                             6.756490     \n",
       "max                                            15.991699     "
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "histone_tags22.describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Identify active enhancers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "1) By setting an average threshold on H3K27ac tags > 32!!! (average 2 replicates, cutoff 32 on the actual tag counts, or 5 on log2 scale)\n",
    "\n",
    "2) HOMER annotations to be at intronic or intergenic regions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[3.543194453397364, 7.128888817966269, 7.759676169332867, 5.498746732929741, 5.561985414668629]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>C57_control_170915</th>\n",
       "      <th>C57_control_180423</th>\n",
       "      <th>C57_control_avg</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4.073820</td>\n",
       "      <td>3.012569</td>\n",
       "      <td>3.543194</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6.692790</td>\n",
       "      <td>7.564988</td>\n",
       "      <td>7.128889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7.917730</td>\n",
       "      <td>7.601622</td>\n",
       "      <td>7.759676</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>5.205940</td>\n",
       "      <td>5.791554</td>\n",
       "      <td>5.498747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5.805292</td>\n",
       "      <td>5.318678</td>\n",
       "      <td>5.561985</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   C57_control_170915  C57_control_180423  C57_control_avg\n",
       "0            4.073820            3.012569         3.543194\n",
       "1            6.692790            7.564988         7.128889\n",
       "2            7.917730            7.601622         7.759676\n",
       "3            5.205940            5.791554         5.498747\n",
       "4            5.805292            5.318678         5.561985"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "histone_tags2.columns = [\"C57_control_170915\", \"C57_control_180423\"]\n",
    "z = [((row.C57_control_170915 + row.C57_control_180423)/2) for index, row in histone_tags2.iterrows() ]\n",
    "print(z[:5])\n",
    "histone_tags2['C57_control_avg'] = z\n",
    "histone_tags2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0     intron (NM_175280, intron 6 of 6)\n",
      "1     intron (NM_017475, intron 3 of 6)\n",
      "2              promoter-TSS (NM_016681)\n",
      "3     intron (NR_045859, intron 2 of 2)\n",
      "4                            Intergenic\n",
      "5     intron (NM_008187, intron 1 of 5)\n",
      "6              promoter-TSS (NM_007509)\n",
      "7    intron (NM_008976, intron 3 of 18)\n",
      "8                            Intergenic\n",
      "9              promoter-TSS (NR_155747)\n",
      "Name: Annotation, dtype: object\n"
     ]
    }
   ],
   "source": [
    "#glancing at annotations\n",
    "print((annot_df.iloc[:,7])[:10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#just gonna loop through all rows (for both dfs) and record rows to save as those with:\n",
    "#histone > 16 (average 2 replicates, cutoff 16 on the actual tag counts, or 4 on log2 scale)\n",
    "#Annotation containing either: intron, Intergenic\n",
    "\n",
    "save_rows = []\n",
    "for row in np.arange(len(annot_df)):\n",
    "    #grab full annotation\n",
    "    full_annot = str(annot_df.iloc[row,7])\n",
    "    if (\"intron\" in full_annot or \"Intergenic\" in full_annot) and (histone_tags2.iloc[row,2] > 5):\n",
    "        save_rows.append(row)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17539\n",
      "[1, 3, 4, 5, 10, 11, 14, 18, 19, 20]\n"
     ]
    }
   ],
   "source": [
    "print(len(save_rows))\n",
    "print(save_rows[:10]) #looks good!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17539\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/round3/00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119_round3_00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916_round3_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423)</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "      <th>Focus Ratio/Region Size</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Detailed Annotation</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>...</th>\n",
       "      <th>Entrez ID</th>\n",
       "      <th>Nearest Unigene</th>\n",
       "      <th>Nearest Refseq</th>\n",
       "      <th>Nearest Ensembl</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Alias</th>\n",
       "      <th>Gene Description</th>\n",
       "      <th>Gene Type</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 Tag Count in 1000 bp (9468035.0 Total, normalization factor = 1.06, effective total = 1e7)</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423 Tag Count in 1000 bp (22615879.0 Total, normalization factor = 0.44, effective total = 1e7)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>457_9613</td>\n",
       "      <td>chr4</td>\n",
       "      <td>123920875</td>\n",
       "      <td>123921875</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>104.0</td>\n",
       "      <td>intron (NM_017475, intron 3 of 6)</td>\n",
       "      <td>intron (NM_017475, intron 3 of 6)</td>\n",
       "      <td>3942.0</td>\n",
       "      <td>...</td>\n",
       "      <td>54170.0</td>\n",
       "      <td>Mm.220922</td>\n",
       "      <td>NM_017475</td>\n",
       "      <td>ENSMUSG00000028646</td>\n",
       "      <td>Rragc</td>\n",
       "      <td>AU041672|Gtr2|RAGC|TIB929|YGR163W</td>\n",
       "      <td>Ras-related GTP binding C</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>102.45</td>\n",
       "      <td>188.36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>457_6548</td>\n",
       "      <td>chr14</td>\n",
       "      <td>105434993</td>\n",
       "      <td>105435993</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>125.0</td>\n",
       "      <td>intron (NR_045859, intron 2 of 2)</td>\n",
       "      <td>MTD|LTR|ERVL-MaLR</td>\n",
       "      <td>7934.0</td>\n",
       "      <td>...</td>\n",
       "      <td>71362.0</td>\n",
       "      <td>Mm.410781</td>\n",
       "      <td>NR_045859</td>\n",
       "      <td>ENSMUSG00000115548</td>\n",
       "      <td>5430440P10Rik</td>\n",
       "      <td>-</td>\n",
       "      <td>RIKEN cDNA 5430440P10 gene</td>\n",
       "      <td>ncRNA</td>\n",
       "      <td>35.91</td>\n",
       "      <td>54.39</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>457_4608</td>\n",
       "      <td>chr8</td>\n",
       "      <td>107029706</td>\n",
       "      <td>107030706</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>144.0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>-1120.0</td>\n",
       "      <td>...</td>\n",
       "      <td>116733.0</td>\n",
       "      <td>Mm.236004</td>\n",
       "      <td>NM_126165</td>\n",
       "      <td>ENSMUSG00000031913</td>\n",
       "      <td>Vps4a</td>\n",
       "      <td>4930589C15Rik|AI325971|AW553189</td>\n",
       "      <td>vacuolar protein sorting 4A</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>54.92</td>\n",
       "      <td>38.91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>457_19168</td>\n",
       "      <td>chr8</td>\n",
       "      <td>95427860</td>\n",
       "      <td>95428860</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>67.0</td>\n",
       "      <td>intron (NM_008187, intron 1 of 5)</td>\n",
       "      <td>intron (NM_008187, intron 1 of 5)</td>\n",
       "      <td>6509.0</td>\n",
       "      <td>...</td>\n",
       "      <td>14894.0</td>\n",
       "      <td>Mm.2080</td>\n",
       "      <td>NM_008187</td>\n",
       "      <td>ENSMUSG00000031796</td>\n",
       "      <td>Cfap20</td>\n",
       "      <td>2600014O15Rik|AL024127|Gtl3|T10-2A2</td>\n",
       "      <td>cilia and flagella associated protein 20</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>36.97</td>\n",
       "      <td>36.70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>457_1618</td>\n",
       "      <td>chr8</td>\n",
       "      <td>27260710</td>\n",
       "      <td>27261710</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>195.0</td>\n",
       "      <td>intron (NM_007918, intron 1 of 2)</td>\n",
       "      <td>intron (NM_007918, intron 1 of 2)</td>\n",
       "      <td>883.0</td>\n",
       "      <td>...</td>\n",
       "      <td>13685.0</td>\n",
       "      <td>Mm.6700</td>\n",
       "      <td>NM_007918</td>\n",
       "      <td>ENSMUSG00000031490</td>\n",
       "      <td>Eif4ebp1</td>\n",
       "      <td>4e-bp1|AA959816|PHAS-I</td>\n",
       "      <td>eukaryotic translation initiation factor 4E bi...</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>102.45</td>\n",
       "      <td>97.72</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/round3/00_C57BL6_Kupffer_ATAC_control_young_C5701_JSS_TDT_160119_round3_00_C57BL6_Kupffer_ATAC_control_young_C57E_JSS_TDT_170916_round3_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423)  \\\n",
       "1                                            457_9613                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "3                                            457_6548                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "4                                            457_4608                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "5                                           457_19168                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "10                                           457_1618                                                                                                                                                                                                                                                                                                                                                                                                          \n",
       "\n",
       "      Chr      Start        End Strand  Peak Score  Focus Ratio/Region Size  \\\n",
       "1    chr4  123920875  123921875      +        1000                    104.0   \n",
       "3   chr14  105434993  105435993      +        1000                    125.0   \n",
       "4    chr8  107029706  107030706      +        1000                    144.0   \n",
       "5    chr8   95427860   95428860      +        1000                     67.0   \n",
       "10   chr8   27260710   27261710      +        1000                    195.0   \n",
       "\n",
       "                           Annotation                Detailed Annotation  \\\n",
       "1   intron (NM_017475, intron 3 of 6)  intron (NM_017475, intron 3 of 6)   \n",
       "3   intron (NR_045859, intron 2 of 2)                  MTD|LTR|ERVL-MaLR   \n",
       "4                          Intergenic                         Intergenic   \n",
       "5   intron (NM_008187, intron 1 of 5)  intron (NM_008187, intron 1 of 5)   \n",
       "10  intron (NM_007918, intron 1 of 2)  intron (NM_007918, intron 1 of 2)   \n",
       "\n",
       "    Distance to TSS  ... Entrez ID  Nearest Unigene Nearest Refseq  \\\n",
       "1            3942.0  ...   54170.0        Mm.220922      NM_017475   \n",
       "3            7934.0  ...   71362.0        Mm.410781      NR_045859   \n",
       "4           -1120.0  ...  116733.0        Mm.236004      NM_126165   \n",
       "5            6509.0  ...   14894.0          Mm.2080      NM_008187   \n",
       "10            883.0  ...   13685.0          Mm.6700      NM_007918   \n",
       "\n",
       "       Nearest Ensembl      Gene Name                           Gene Alias  \\\n",
       "1   ENSMUSG00000028646          Rragc    AU041672|Gtr2|RAGC|TIB929|YGR163W   \n",
       "3   ENSMUSG00000115548  5430440P10Rik                                    -   \n",
       "4   ENSMUSG00000031913          Vps4a      4930589C15Rik|AI325971|AW553189   \n",
       "5   ENSMUSG00000031796         Cfap20  2600014O15Rik|AL024127|Gtl3|T10-2A2   \n",
       "10  ENSMUSG00000031490       Eif4ebp1               4e-bp1|AA959816|PHAS-I   \n",
       "\n",
       "                                     Gene Description       Gene Type  \\\n",
       "1                           Ras-related GTP binding C  protein-coding   \n",
       "3                          RIKEN cDNA 5430440P10 gene           ncRNA   \n",
       "4                         vacuolar protein sorting 4A  protein-coding   \n",
       "5            cilia and flagella associated protein 20  protein-coding   \n",
       "10  eukaryotic translation initiation factor 4E bi...  protein-coding   \n",
       "\n",
       "   /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C571A_170915 Tag Count in 1000 bp (9468035.0 Total, normalization factor = 1.06, effective total = 1e7)  \\\n",
       "1                                              102.45                                                                                                                                         \n",
       "3                                               35.91                                                                                                                                         \n",
       "4                                               54.92                                                                                                                                         \n",
       "5                                               36.97                                                                                                                                         \n",
       "10                                             102.45                                                                                                                                         \n",
       "\n",
       "    /home/h1bennet/strains/data/H3K27Ac/control/00_C57_Kupffer_H3K27Ac_control_young_C572A_180423 Tag Count in 1000 bp (22615879.0 Total, normalization factor = 0.44, effective total = 1e7)  \n",
       "1                                              188.36                                                                                                                                          \n",
       "3                                               54.39                                                                                                                                          \n",
       "4                                               38.91                                                                                                                                          \n",
       "5                                               36.70                                                                                                                                          \n",
       "10                                              97.72                                                                                                                                          \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filt_annot = annot_df.iloc[save_rows,:]\n",
    "print(len(filt_annot))\n",
    "filt_annot.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "#export this just in case\n",
    "out_file = path + strain + \"_filt_annot.txt\"\n",
    "filt_annot.to_csv(out_file, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### AMLN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[4.49229809280194, 7.031265040702184, 7.473669381170035, 6.032954421152834, 5.394789318396052]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>C57_AMLN1</th>\n",
       "      <th>C57_control_AMLN2</th>\n",
       "      <th>C57_AMLN_avg</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>4.257765</td>\n",
       "      <td>4.726831</td>\n",
       "      <td>4.492298</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>7.292230</td>\n",
       "      <td>6.770300</td>\n",
       "      <td>7.031265</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>7.596488</td>\n",
       "      <td>7.350851</td>\n",
       "      <td>7.473669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6.395748</td>\n",
       "      <td>5.670161</td>\n",
       "      <td>6.032954</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5.430620</td>\n",
       "      <td>5.358959</td>\n",
       "      <td>5.394789</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   C57_AMLN1  C57_control_AMLN2  C57_AMLN_avg\n",
       "0   4.257765           4.726831      4.492298\n",
       "1   7.292230           6.770300      7.031265\n",
       "2   7.596488           7.350851      7.473669\n",
       "3   6.395748           5.670161      6.032954\n",
       "4   5.430620           5.358959      5.394789"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "histone_tags22.columns = [\"C57_AMLN1\", \"C57_control_AMLN2\"]\n",
    "z2 = [((row.C57_AMLN1 + row.C57_control_AMLN2)/2) for index, row in histone_tags22.iterrows() ]\n",
    "print(z2[:5])\n",
    "histone_tags22['C57_AMLN_avg'] = z2\n",
    "histone_tags22.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0     intron (NM_001165989, intron 1 of 43)\n",
      "1                  promoter-TSS (NR_015601)\n",
      "2                  promoter-TSS (NM_025624)\n",
      "3                  promoter-TSS (NM_009765)\n",
      "4                                Intergenic\n",
      "5    intron (NM_001033167, intron 10 of 10)\n",
      "6                  promoter-TSS (NM_029711)\n",
      "7      intron (NM_001310481, intron 4 of 5)\n",
      "8         intron (NM_021297, intron 1 of 2)\n",
      "9                                Intergenic\n",
      "Name: Annotation, dtype: object\n"
     ]
    }
   ],
   "source": [
    "#glancing at annotations\n",
    "print((annot_df2.iloc[:,7])[:10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11600\n",
      "[4, 5, 7, 8, 9, 13, 14, 18, 20, 21]\n"
     ]
    }
   ],
   "source": [
    "#just gonna loop through all rows (for both dfs) and record rows to save as those with:\n",
    "#histone > 16 (average 2 replicates, cutoff 16 on the actual tag counts, or 4 on log2 scale)\n",
    "#Annotation containing either: intron, Intergenic\n",
    "\n",
    "save_rows2 = []\n",
    "for row in np.arange(len(annot_df2)):\n",
    "    #grab full annotation\n",
    "    full_annot2 = str(annot_df2.iloc[row,7])\n",
    "    if (\"intron\" in full_annot2 or \"Intergenic\" in full_annot2) and (histone_tags22.iloc[row,2] > 5):\n",
    "        save_rows2.append(row)\n",
    "        \n",
    "print(len(save_rows2))\n",
    "print(save_rows2[:10]) #looks good!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11600\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/round3/00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425_round3_00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919_round3_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423)</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "      <th>Focus Ratio/Region Size</th>\n",
       "      <th>Annotation</th>\n",
       "      <th>Detailed Annotation</th>\n",
       "      <th>Distance to TSS</th>\n",
       "      <th>...</th>\n",
       "      <th>Entrez ID</th>\n",
       "      <th>Nearest Unigene</th>\n",
       "      <th>Nearest Refseq</th>\n",
       "      <th>Nearest Ensembl</th>\n",
       "      <th>Gene Name</th>\n",
       "      <th>Gene Alias</th>\n",
       "      <th>Gene Description</th>\n",
       "      <th>Gene Type</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 Tag Count in 1000 bp (18749728.0 Total, normalization factor = 0.53, effective total = 1e7)</th>\n",
       "      <th>/home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423 Tag Count in 1000 bp (19231109.0 Total, normalization factor = 0.52, effective total = 1e7)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>868_13869</td>\n",
       "      <td>chr13</td>\n",
       "      <td>64401616</td>\n",
       "      <td>64402616</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>54.0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>-30437.0</td>\n",
       "      <td>...</td>\n",
       "      <td>105278.0</td>\n",
       "      <td>Mm.74982</td>\n",
       "      <td>NM_053180</td>\n",
       "      <td>ENSMUSG00000021483</td>\n",
       "      <td>Cdk20</td>\n",
       "      <td>4932702G04Rik|AU019450|AW558027|CDCH|Ccrk|PNQL...</td>\n",
       "      <td>cyclin-dependent kinase 20</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>42.13</td>\n",
       "      <td>40.04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>868_20649</td>\n",
       "      <td>chr13</td>\n",
       "      <td>34187911</td>\n",
       "      <td>34188911</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>41.0</td>\n",
       "      <td>intron (NM_001033167, intron 10 of 10)</td>\n",
       "      <td>intron (NM_001033167, intron 10 of 10)</td>\n",
       "      <td>25447.0</td>\n",
       "      <td>...</td>\n",
       "      <td>69666.0</td>\n",
       "      <td>Mm.24788</td>\n",
       "      <td>NM_001101430</td>\n",
       "      <td>ENSMUSG00000071451</td>\n",
       "      <td>Psmg4</td>\n",
       "      <td>PAC-4|mPAC4</td>\n",
       "      <td>proteasome (prosome, macropain) assembly chape...</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>79.47</td>\n",
       "      <td>80.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>868_18847</td>\n",
       "      <td>chr15</td>\n",
       "      <td>50761283</td>\n",
       "      <td>50762283</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>44.0</td>\n",
       "      <td>intron (NM_001310481, intron 4 of 5)</td>\n",
       "      <td>intron (NM_001310481, intron 4 of 5)</td>\n",
       "      <td>127266.0</td>\n",
       "      <td>...</td>\n",
       "      <td>83925.0</td>\n",
       "      <td>Mm.30466</td>\n",
       "      <td>NM_032000</td>\n",
       "      <td>ENSMUSG00000038679</td>\n",
       "      <td>Trps1</td>\n",
       "      <td>AI115454|AI447310|D15Ertd586e</td>\n",
       "      <td>transcriptional repressor GATA binding 1</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>33.60</td>\n",
       "      <td>40.56</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>868_15363</td>\n",
       "      <td>chr4</td>\n",
       "      <td>66828378</td>\n",
       "      <td>66829378</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>51.0</td>\n",
       "      <td>intron (NM_021297, intron 1 of 2)</td>\n",
       "      <td>intron (NM_021297, intron 1 of 2)</td>\n",
       "      <td>1327.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21898.0</td>\n",
       "      <td>Mm.38049</td>\n",
       "      <td>NM_021297</td>\n",
       "      <td>ENSMUSG00000039005</td>\n",
       "      <td>Tlr4</td>\n",
       "      <td>Lps|Ly87|Ran/M1|Rasl2-8</td>\n",
       "      <td>toll-like receptor 4</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>270.94</td>\n",
       "      <td>241.80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>868_3909</td>\n",
       "      <td>chr14</td>\n",
       "      <td>30592930</td>\n",
       "      <td>30593930</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "      <td>97.0</td>\n",
       "      <td>Intergenic</td>\n",
       "      <td>MIR|SINE|MIR</td>\n",
       "      <td>16880.0</td>\n",
       "      <td>...</td>\n",
       "      <td>18753.0</td>\n",
       "      <td>Mm.2314</td>\n",
       "      <td>NM_011103</td>\n",
       "      <td>ENSMUSG00000021948</td>\n",
       "      <td>Prkcd</td>\n",
       "      <td>AI385711|D14Ertd420e|PKC[d]|PKCdelta|Pkcd</td>\n",
       "      <td>protein kinase C, delta</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>355.74</td>\n",
       "      <td>367.11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  PeakID (cmd=annotatePeaks.pl /home/hmummey/data/ATAC/annotated_peak_calling/round3/00_NCoRWT_Kupffer_ATAC_AMLNDiet_20week_LN203D_JSS_TDT_180425_round3_00_NCoRWT_Kupffer_ATAC_AMLNDiet_21week_LN170C_JSS_TDT_160919_round3_idr.tsv mm10 -size -500,500 -norm 1e7 -d /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423)  \\\n",
       "4                                          868_13869                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "5                                          868_20649                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "7                                          868_18847                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "8                                          868_15363                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "9                                           868_3909                                                                                                                                                                                                                                                                                                                                                                                                                               \n",
       "\n",
       "     Chr     Start       End Strand  Peak Score  Focus Ratio/Region Size  \\\n",
       "4  chr13  64401616  64402616      +        1000                     54.0   \n",
       "5  chr13  34187911  34188911      +        1000                     41.0   \n",
       "7  chr15  50761283  50762283      +        1000                     44.0   \n",
       "8   chr4  66828378  66829378      +        1000                     51.0   \n",
       "9  chr14  30592930  30593930      +        1000                     97.0   \n",
       "\n",
       "                               Annotation  \\\n",
       "4                              Intergenic   \n",
       "5  intron (NM_001033167, intron 10 of 10)   \n",
       "7    intron (NM_001310481, intron 4 of 5)   \n",
       "8       intron (NM_021297, intron 1 of 2)   \n",
       "9                              Intergenic   \n",
       "\n",
       "                      Detailed Annotation  Distance to TSS  ... Entrez ID  \\\n",
       "4                              Intergenic         -30437.0  ...  105278.0   \n",
       "5  intron (NM_001033167, intron 10 of 10)          25447.0  ...   69666.0   \n",
       "7    intron (NM_001310481, intron 4 of 5)         127266.0  ...   83925.0   \n",
       "8       intron (NM_021297, intron 1 of 2)           1327.0  ...   21898.0   \n",
       "9                            MIR|SINE|MIR          16880.0  ...   18753.0   \n",
       "\n",
       "   Nearest Unigene Nearest Refseq     Nearest Ensembl Gene Name  \\\n",
       "4         Mm.74982      NM_053180  ENSMUSG00000021483     Cdk20   \n",
       "5         Mm.24788   NM_001101430  ENSMUSG00000071451     Psmg4   \n",
       "7         Mm.30466      NM_032000  ENSMUSG00000038679     Trps1   \n",
       "8         Mm.38049      NM_021297  ENSMUSG00000039005      Tlr4   \n",
       "9          Mm.2314      NM_011103  ENSMUSG00000021948     Prkcd   \n",
       "\n",
       "                                          Gene Alias  \\\n",
       "4  4932702G04Rik|AU019450|AW558027|CDCH|Ccrk|PNQL...   \n",
       "5                                        PAC-4|mPAC4   \n",
       "7                      AI115454|AI447310|D15Ertd586e   \n",
       "8                            Lps|Ly87|Ran/M1|Rasl2-8   \n",
       "9          AI385711|D14Ertd420e|PKC[d]|PKCdelta|Pkcd   \n",
       "\n",
       "                                    Gene Description       Gene Type  \\\n",
       "4                         cyclin-dependent kinase 20  protein-coding   \n",
       "5  proteasome (prosome, macropain) assembly chape...  protein-coding   \n",
       "7           transcriptional repressor GATA binding 1  protein-coding   \n",
       "8                               toll-like receptor 4  protein-coding   \n",
       "9                            protein kinase C, delta  protein-coding   \n",
       "\n",
       "  /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579A_180423 Tag Count in 1000 bp (18749728.0 Total, normalization factor = 0.53, effective total = 1e7)  \\\n",
       "4                                              42.13                                                                                                                                                 \n",
       "5                                              79.47                                                                                                                                                 \n",
       "7                                              33.60                                                                                                                                                 \n",
       "8                                             270.94                                                                                                                                                 \n",
       "9                                             355.74                                                                                                                                                 \n",
       "\n",
       "   /home/h1bennet/strains/data/H3K27Ac/AMLN_30week/00_C57_Kupffer_H3K27Ac_AMLNDiet_20weeks_C579C_180423 Tag Count in 1000 bp (19231109.0 Total, normalization factor = 0.52, effective total = 1e7)  \n",
       "4                                              40.04                                                                                                                                                 \n",
       "5                                              80.08                                                                                                                                                 \n",
       "7                                              40.56                                                                                                                                                 \n",
       "8                                             241.80                                                                                                                                                 \n",
       "9                                             367.11                                                                                                                                                 \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "filt_annot2 = annot_df2.iloc[save_rows2,:]\n",
    "print(len(filt_annot2))\n",
    "filt_annot2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "#export this just in case\n",
    "out_file2 = path + strain2 + \"_filt_annot.txt\"\n",
    "filt_annot.to_csv(out_file2, index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Convert peak file to BED"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Bed file format requires:\n",
    "- chrom - name of the chromosome or scaffold. Any valid seq_region_name can be used, and chromosome names can be given with or without the 'chr' prefix.\n",
    "- chromStart - Start position of the feature in standard chromosomal coordinates (i.e. first base is 0).\n",
    "- chromEnd - End position of the feature in standard chromosomal coordinates\n",
    "\n",
    "Optional columns: name, score, strand, thickStart, thickEnd, itemRgb, blockCount, blockSizes, blockStarts\n",
    "\n",
    "from: https://m.ensembl.org/info/website/upload/bed.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17539\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4</td>\n",
       "      <td>123920875</td>\n",
       "      <td>123921875</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr14</td>\n",
       "      <td>105434993</td>\n",
       "      <td>105435993</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr8</td>\n",
       "      <td>107029706</td>\n",
       "      <td>107030706</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr8</td>\n",
       "      <td>95427860</td>\n",
       "      <td>95428860</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr8</td>\n",
       "      <td>27260710</td>\n",
       "      <td>27261710</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr      Start        End Strand  Peak Score\n",
       "1    chr4  123920875  123921875      +        1000\n",
       "3   chr14  105434993  105435993      +        1000\n",
       "4    chr8  107029706  107030706      +        1000\n",
       "5    chr8   95427860   95428860      +        1000\n",
       "10   chr8   27260710   27261710      +        1000"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#just going to use the required cols (and maybe strand and score? why not, I guess)\n",
    "save_cols = [1,2,3,4,5]\n",
    "bed_df = filt_annot.iloc[:,save_cols]\n",
    "print(len(bed_df))\n",
    "bed_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['chr1', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17', 'chr18', 'chr19', 'chr2', 'chr3', 'chr4', 'chr4_GL456216_random', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9', 'chrUn_JH584304', 'chrX', 'chrY']\n"
     ]
    }
   ],
   "source": [
    "chr_list = list(set(bed_df.iloc[:,0]))\n",
    "chr_list.sort()\n",
    "print(chr_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove all rows that are chrY (makes the script to create the background, error) or chrUn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17500\n",
      "[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]\n"
     ]
    }
   ],
   "source": [
    "#also might as well try removing rows with Un chrs (will do this manually and see if helps)\n",
    "keep_rows12 = []\n",
    "for i in range(len(bed_df)):\n",
    "    chr_str = bed_df.iloc[i,0]\n",
    "    if (\"chrY\" in chr_str) or (\"Un\" in chr_str):\n",
    "        pass\n",
    "    else:\n",
    "        keep_rows12.append(i)\n",
    "\n",
    "print(len(keep_rows12))\n",
    "print(keep_rows12[:20])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17500\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr4</td>\n",
       "      <td>123920875</td>\n",
       "      <td>123921875</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr14</td>\n",
       "      <td>105434993</td>\n",
       "      <td>105435993</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr8</td>\n",
       "      <td>107029706</td>\n",
       "      <td>107030706</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr8</td>\n",
       "      <td>95427860</td>\n",
       "      <td>95428860</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>chr8</td>\n",
       "      <td>27260710</td>\n",
       "      <td>27261710</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Chr      Start        End Strand  Peak Score\n",
       "1    chr4  123920875  123921875      +        1000\n",
       "3   chr14  105434993  105435993      +        1000\n",
       "4    chr8  107029706  107030706      +        1000\n",
       "5    chr8   95427860   95428860      +        1000\n",
       "10   chr8   27260710   27261710      +        1000"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bed_df_cut12 = bed_df.iloc[keep_rows12,:]\n",
    "print(len(bed_df_cut12))\n",
    "bed_df_cut12.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "out3 = path + strain + \"_filt_annot_fin.bed\"\n",
    "bed_df_cut12.to_csv(out3, index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### AMLN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11600\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr13</td>\n",
       "      <td>64401616</td>\n",
       "      <td>64402616</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr13</td>\n",
       "      <td>34187911</td>\n",
       "      <td>34188911</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr15</td>\n",
       "      <td>50761283</td>\n",
       "      <td>50762283</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr4</td>\n",
       "      <td>66828378</td>\n",
       "      <td>66829378</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr14</td>\n",
       "      <td>30592930</td>\n",
       "      <td>30593930</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr     Start       End Strand  Peak Score\n",
       "4  chr13  64401616  64402616      +        1000\n",
       "5  chr13  34187911  34188911      +        1000\n",
       "7  chr15  50761283  50762283      +        1000\n",
       "8   chr4  66828378  66829378      +        1000\n",
       "9  chr14  30592930  30593930      +        1000"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#just going to use the required cols (and maybe strand and score? why not, I guess)\n",
    "\n",
    "save_cols2 = [1,2,3,4,5]\n",
    "bed_df2 = filt_annot2.iloc[:,save_cols2]\n",
    "print(len(bed_df2))\n",
    "bed_df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['chr1', 'chr10', 'chr11', 'chr12', 'chr13', 'chr14', 'chr15', 'chr16', 'chr17', 'chr18', 'chr19', 'chr2', 'chr3', 'chr4', 'chr4_GL456216_random', 'chr5', 'chr6', 'chr7', 'chr8', 'chr9', 'chrUn_JH584304', 'chrX', 'chrY']\n"
     ]
    }
   ],
   "source": [
    "chr_list = list(set(bed_df2.iloc[:,0]))\n",
    "chr_list.sort()\n",
    "print(chr_list)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove all rows that are chrY (makes the script to create the background, error) or chrUn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11568\n",
      "[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]\n"
     ]
    }
   ],
   "source": [
    "#also might as well try removing rows with Un chrs (will do this manually and see if helps)\n",
    "keep_rows22 = []\n",
    "for i in range(len(bed_df2)):\n",
    "    chr_str = bed_df2.iloc[i,0]\n",
    "    if (\"chrY\" in chr_str) or (\"Un\" in chr_str):\n",
    "        pass\n",
    "    else:\n",
    "        keep_rows22.append(i)\n",
    "        \n",
    "print(len(keep_rows22))\n",
    "print(keep_rows22[:20])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11568\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Peak Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr13</td>\n",
       "      <td>64401616</td>\n",
       "      <td>64402616</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>chr13</td>\n",
       "      <td>34187911</td>\n",
       "      <td>34188911</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>chr15</td>\n",
       "      <td>50761283</td>\n",
       "      <td>50762283</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>chr4</td>\n",
       "      <td>66828378</td>\n",
       "      <td>66829378</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr14</td>\n",
       "      <td>30592930</td>\n",
       "      <td>30593930</td>\n",
       "      <td>+</td>\n",
       "      <td>1000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr     Start       End Strand  Peak Score\n",
       "4  chr13  64401616  64402616      +        1000\n",
       "5  chr13  34187911  34188911      +        1000\n",
       "7  chr15  50761283  50762283      +        1000\n",
       "8   chr4  66828378  66829378      +        1000\n",
       "9  chr14  30592930  30593930      +        1000"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bed_df_cut22 = bed_df2.iloc[keep_rows22,:]\n",
    "print(len(bed_df_cut22))\n",
    "bed_df_cut22.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "out4 = path + strain2 + \"_filt_annot_fin.bed\"\n",
    "bed_df_cut22.to_csv(out4, index=False, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Generate backgrounds sets"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Random matched-GC content sets"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Bed file is the positives we've just chosen (only matching GC content). Make sure to remove the top row (col names) in terminal with sed '1d' before running the script! Col names are the same for all the C57 output files so not a huge loss to just delete them and then overwrite the file."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Control"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/hmummey/data/ATAC/annotated_peak_calling/round3/C57_control_round3_filt_annot_fin_bg\n",
      "reading genome mm10\n",
      "fasta file does not exist: chr20\n",
      "fasta file does not exist: chr21\n",
      "fasta file does not exist: chr22\n",
      "done reading genome\n",
      "0 0\n",
      "target GC: 0.37876971428569267 background GC: 0.38047438276007534 target length: 1000 numTargetPositions 1750 backgroundPositions 1750\n",
      "0 0\n",
      "target GC: 0.41272399999997644 background GC: 0.41207192807190457 target length: 1000 numTargetPositions 1750 backgroundPositions 1750\n",
      "0 0\n",
      "target GC: 0.43063999999997543 background GC: 0.42904524047378745 target length: 1000 numTargetPositions 1750 backgroundPositions 1750\n",
      "0 0\n",
      "target GC: 0.4453834285714032 background GC: 0.4416081061795096 target length: 1000 numTargetPositions 1750 backgroundPositions 1750\n",
      "0 0\n",
      "target GC: 0.45895542857140237 background GC: 0.4549216497787667 target length: 1000 numTargetPositions 1750 backgroundPositions 1750\n",
      "0 0\n",
      "target GC: 0.47263085714283015 background GC: 0.46738347366916133 target length: 1000 numTargetPositions 1750 backgroundPositions 1750\n",
      "0 0\n",
      "target GC: 0.48751771428568647 background GC: 0.48130669330666587 target length: 1000 numTargetPositions 1750 backgroundPositions 1750\n",
      "0 0\n",
      "target GC: 0.5054697142856854 background GC: 0.4979163693449124 target length: 1000 numTargetPositions 1750 backgroundPositions 1750\n",
      "0 0\n",
      "target GC: 0.5302858776443379 background GC: 0.5221970710821184 target length: 1000 numTargetPositions 1749 backgroundPositions 1749\n",
      "0 0\n",
      "target GC: 0.6005683247569698 background GC: 0.5925458189608795 target length: 1000 numTargetPositions 1749 backgroundPositions 1749\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "bed_file=\"/home/hmummey/data/ATAC/annotated_peak_calling/round3/C57_control_round3_filt_annot_fin.bed\" #+ peaks, no chrY, no chrUn\n",
    "output=$(echo $bed_file | sed 's/.bed/_bg/g')\n",
    "echo $output\n",
    "python /home/zes017/Spacing/Codes/generate_background_coordinates.py $bed_file $output"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### AMLN --> skipping this step for now"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#%%bash\n",
    "#bed_file=\"/home/hmummey/data/ATAC/annotated_peak_calling/C57_AMLNDiet_round2_filt_annot_fin.bed\" #+ peaks, no chrY, no chrUn\n",
    "#output=$(echo $bed_file | sed 's/.bed/_bg/g')\n",
    "#echo $output\n",
    "#python /home/zes017/Spacing/Codes/generate_background_coordinates.py $bed_file $output"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate Differential Peak Sets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C57_AMLNDiet_round3_filt_annot_fin.bed\n",
      "C57_control_round3_filt_annot_fin.bed\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tMax distance to merge: direct overlap required (-d given)\n",
      "\tMerging peaks... \n",
      "\tComparing C57_AMLNDiet_round3_filt_annot_fin.bed (11568 total) and C57_AMLNDiet_round3_filt_annot_fin.bed (11568 total)\n",
      "\tComparing C57_AMLNDiet_round3_filt_annot_fin.bed (11568 total) and C57_control_round3_filt_annot_fin.bed (17500 total)\n",
      "\tComparing C57_control_round3_filt_annot_fin.bed (17500 total) and C57_AMLNDiet_round3_filt_annot_fin.bed (11568 total)\n",
      "\tComparing C57_control_round3_filt_annot_fin.bed (17500 total) and C57_control_round3_filt_annot_fin.bed (17500 total)\n",
      "\n",
      "C57_AMLNDiet_round3_filt_annot_fin.bed\tC57_control_round3_filt_annot_fin.bed\tTotal\tName\n",
      "\tX\t6221\tC57_control_round3_filt_annot_fin.bed\n",
      "X\t\t1820\tC57_AMLNDiet_round3_filt_annot_fin.bed\n",
      "X\tX\t8343\tC57_AMLNDiet_round3_filt_annot_fin.bed|C57_control_round3_filt_annot_fin.bed\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "#comparison of MY FINAL ANNOTATED PEAK SETS\n",
    "#based off code in the notebook: 210124_Peak_comparison.ipynb\n",
    "\n",
    "cd /home/hmummey/data/ATAC/annotated_peak_calling/round3/peak_comparison\n",
    "\n",
    "file1=\"C57_AMLNDiet_round3_filt_annot_fin.bed\"\n",
    "file2=\"C57_control_round3_filt_annot_fin.bed\"\n",
    "\n",
    "echo $file1\n",
    "echo $file2\n",
    "\n",
    "mergePeaks -d given $file1 $file2 -prefix merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Generate Matched Size Bg (control) set for AMLN"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#probably going to read in previous sets (with lower Ac to have larger training set)\n",
    "#referring to peak comparison document to get differential peak files\n",
    "#from the previous NCoRWT control vs. AMLN Diet\n",
    "\n",
    "direct = '/home/hmummey/data/ATAC/annotated_peak_calling/peak_comparison/'\n",
    "file1 = 'merge5_C57_control_round2_filt_annot_fin.bed'\n",
    "file2 = 'merge5_C57_AMLNDiet_round2_filt_annot_fin.bed'\n",
    "file3 = 'merge5_C57_AMLNDiet_round2_filt_annot_fin.bed_C57_control_round2_filt_annot_fin.bed'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/hmummey/data/ATAC/annotated_peak_calling/peak_comparison/merge5_C57_control_round2_filt_annot_fin.bed\n",
      "6745\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#name (cmd = mergePeaks -d given C57_AMLNDiet_round2_filt_annot_fin.bed C57_control_round2_filt_annot_fin.bed -prefix merge5)</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>C57_AMLNDiet_round2_filt_annot_fin.bed</th>\n",
       "      <th>C57_control_round2_filt_annot_fin.bed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6740</th>\n",
       "      <td>Merged-chr11-5757997-1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>5757498</td>\n",
       "      <td>5758497</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_control_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>default-4041</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6741</th>\n",
       "      <td>Merged-chr3-136078820-1</td>\n",
       "      <td>chr3</td>\n",
       "      <td>136078321</td>\n",
       "      <td>136079320</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_control_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>default-4043</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6742</th>\n",
       "      <td>Merged-chr1-21070186-1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>21069687</td>\n",
       "      <td>21070686</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_control_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>default-4058</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6743</th>\n",
       "      <td>Merged-chr13-111508828-1</td>\n",
       "      <td>chr13</td>\n",
       "      <td>111508329</td>\n",
       "      <td>111509328</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_control_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>default-5662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6744</th>\n",
       "      <td>Merged-chr13-104152560-1</td>\n",
       "      <td>chr13</td>\n",
       "      <td>104152061</td>\n",
       "      <td>104153060</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_control_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>default-5668</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     #name (cmd = mergePeaks -d given C57_AMLNDiet_round2_filt_annot_fin.bed C57_control_round2_filt_annot_fin.bed -prefix merge5)  \\\n",
       "6740                             Merged-chr11-5757997-1                                                                              \n",
       "6741                            Merged-chr3-136078820-1                                                                              \n",
       "6742                             Merged-chr1-21070186-1                                                                              \n",
       "6743                           Merged-chr13-111508828-1                                                                              \n",
       "6744                           Merged-chr13-104152560-1                                                                              \n",
       "\n",
       "        chr      start        end strand  Stat  \\\n",
       "6740  chr11    5757498    5758497      +   0.0   \n",
       "6741   chr3  136078321  136079320      +   0.0   \n",
       "6742   chr1   21069687   21070686      +   0.0   \n",
       "6743  chr13  111508329  111509328      +   0.0   \n",
       "6744  chr13  104152061  104153060      +   0.0   \n",
       "\n",
       "                               Parent files  Total subpeaks  \\\n",
       "6740  C57_control_round2_filt_annot_fin.bed               1   \n",
       "6741  C57_control_round2_filt_annot_fin.bed               1   \n",
       "6742  C57_control_round2_filt_annot_fin.bed               1   \n",
       "6743  C57_control_round2_filt_annot_fin.bed               1   \n",
       "6744  C57_control_round2_filt_annot_fin.bed               1   \n",
       "\n",
       "      C57_AMLNDiet_round2_filt_annot_fin.bed  \\\n",
       "6740                                     NaN   \n",
       "6741                                     NaN   \n",
       "6742                                     NaN   \n",
       "6743                                     NaN   \n",
       "6744                                     NaN   \n",
       "\n",
       "     C57_control_round2_filt_annot_fin.bed  \n",
       "6740                          default-4041  \n",
       "6741                          default-4043  \n",
       "6742                          default-4058  \n",
       "6743                          default-5662  \n",
       "6744                          default-5668  "
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(direct+file1)\n",
    "control_diff = pd.read_csv((direct+file1),sep='\\t')\n",
    "print(len(control_diff))\n",
    "control_diff.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3966\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#name (cmd = mergePeaks -d given C57_AMLNDiet_round2_filt_annot_fin.bed C57_control_round2_filt_annot_fin.bed -prefix merge5)</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>C57_AMLNDiet_round2_filt_annot_fin.bed</th>\n",
       "      <th>C57_control_round2_filt_annot_fin.bed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3961</th>\n",
       "      <td>Merged-chr11-74522634-1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>74522135</td>\n",
       "      <td>74523134</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_AMLNDiet_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>default-5654</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3962</th>\n",
       "      <td>Merged-chr4-125127426-1</td>\n",
       "      <td>chr4</td>\n",
       "      <td>125126927</td>\n",
       "      <td>125127926</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_AMLNDiet_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>default-5656</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3963</th>\n",
       "      <td>Merged-chr6-146271352-1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>146270853</td>\n",
       "      <td>146271852</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_AMLNDiet_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>default-5658</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3964</th>\n",
       "      <td>Merged-chr11-55131169-1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>55130670</td>\n",
       "      <td>55131669</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_AMLNDiet_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>default-4056</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3965</th>\n",
       "      <td>Merged-chr1-74308870-1</td>\n",
       "      <td>chr1</td>\n",
       "      <td>74308371</td>\n",
       "      <td>74309370</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_AMLNDiet_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>default-5660</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     #name (cmd = mergePeaks -d given C57_AMLNDiet_round2_filt_annot_fin.bed C57_control_round2_filt_annot_fin.bed -prefix merge5)  \\\n",
       "3961                            Merged-chr11-74522634-1                                                                              \n",
       "3962                            Merged-chr4-125127426-1                                                                              \n",
       "3963                            Merged-chr6-146271352-1                                                                              \n",
       "3964                            Merged-chr11-55131169-1                                                                              \n",
       "3965                             Merged-chr1-74308870-1                                                                              \n",
       "\n",
       "        chr      start        end strand  Stat  \\\n",
       "3961  chr11   74522135   74523134      +   0.0   \n",
       "3962   chr4  125126927  125127926      +   0.0   \n",
       "3963   chr6  146270853  146271852      +   0.0   \n",
       "3964  chr11   55130670   55131669      +   0.0   \n",
       "3965   chr1   74308371   74309370      +   0.0   \n",
       "\n",
       "                                Parent files  Total subpeaks  \\\n",
       "3961  C57_AMLNDiet_round2_filt_annot_fin.bed               1   \n",
       "3962  C57_AMLNDiet_round2_filt_annot_fin.bed               1   \n",
       "3963  C57_AMLNDiet_round2_filt_annot_fin.bed               1   \n",
       "3964  C57_AMLNDiet_round2_filt_annot_fin.bed               1   \n",
       "3965  C57_AMLNDiet_round2_filt_annot_fin.bed               1   \n",
       "\n",
       "     C57_AMLNDiet_round2_filt_annot_fin.bed  \\\n",
       "3961                           default-5654   \n",
       "3962                           default-5656   \n",
       "3963                           default-5658   \n",
       "3964                           default-4056   \n",
       "3965                           default-5660   \n",
       "\n",
       "      C57_control_round2_filt_annot_fin.bed  \n",
       "3961                                    NaN  \n",
       "3962                                    NaN  \n",
       "3963                                    NaN  \n",
       "3964                                    NaN  \n",
       "3965                                    NaN  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "amln_diff = pd.read_csv((direct+file2),sep='\\t')\n",
    "print(len(amln_diff))\n",
    "amln_diff.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[2912, 1870, 4352, 2798, 4552, 755, 2185, 3978, 2242, 2154]\n",
      "[0, 3, 4, 5, 6, 8, 9, 10, 12, 15]\n"
     ]
    }
   ],
   "source": [
    "#randomly sample 3,966 peaks from control first !!!!!!\n",
    "import random\n",
    "\n",
    "all_cnt_rows = list(range(len(control_diff)))\n",
    "save_rows = random.sample(all_cnt_rows, len(amln_diff))\n",
    "print(save_rows[:10])\n",
    "save_rows.sort()\n",
    "print(save_rows[:10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3966\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#name (cmd = mergePeaks -d given C57_AMLNDiet_round2_filt_annot_fin.bed C57_control_round2_filt_annot_fin.bed -prefix merge5)</th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>C57_AMLNDiet_round2_filt_annot_fin.bed</th>\n",
       "      <th>C57_control_round2_filt_annot_fin.bed</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Merged-chr4-16169024-1</td>\n",
       "      <td>chr4</td>\n",
       "      <td>16168525</td>\n",
       "      <td>16169524</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_control_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>default-4062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Merged-chr13-55479188-1</td>\n",
       "      <td>chr13</td>\n",
       "      <td>55478689</td>\n",
       "      <td>55479688</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_control_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>default-4068</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Merged-chr6-146172458-1</td>\n",
       "      <td>chr6</td>\n",
       "      <td>146171959</td>\n",
       "      <td>146172958</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_control_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>default-5671</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Merged-chr17-53485827-1</td>\n",
       "      <td>chr17</td>\n",
       "      <td>53485328</td>\n",
       "      <td>53486327</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_control_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>default-5675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Merged-chr3-149085001-1</td>\n",
       "      <td>chr3</td>\n",
       "      <td>149084502</td>\n",
       "      <td>149085501</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>C57_control_round2_filt_annot_fin.bed</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>default-5678</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  #name (cmd = mergePeaks -d given C57_AMLNDiet_round2_filt_annot_fin.bed C57_control_round2_filt_annot_fin.bed -prefix merge5)  \\\n",
       "0                             Merged-chr4-16169024-1                                                                              \n",
       "3                            Merged-chr13-55479188-1                                                                              \n",
       "4                            Merged-chr6-146172458-1                                                                              \n",
       "5                            Merged-chr17-53485827-1                                                                              \n",
       "6                            Merged-chr3-149085001-1                                                                              \n",
       "\n",
       "     chr      start        end strand  Stat  \\\n",
       "0   chr4   16168525   16169524      +   0.0   \n",
       "3  chr13   55478689   55479688      +   0.0   \n",
       "4   chr6  146171959  146172958      +   0.0   \n",
       "5  chr17   53485328   53486327      +   0.0   \n",
       "6   chr3  149084502  149085501      +   0.0   \n",
       "\n",
       "                            Parent files  Total subpeaks  \\\n",
       "0  C57_control_round2_filt_annot_fin.bed               1   \n",
       "3  C57_control_round2_filt_annot_fin.bed               1   \n",
       "4  C57_control_round2_filt_annot_fin.bed               1   \n",
       "5  C57_control_round2_filt_annot_fin.bed               1   \n",
       "6  C57_control_round2_filt_annot_fin.bed               1   \n",
       "\n",
       "   C57_AMLNDiet_round2_filt_annot_fin.bed  \\\n",
       "0                                     NaN   \n",
       "3                                     NaN   \n",
       "4                                     NaN   \n",
       "5                                     NaN   \n",
       "6                                     NaN   \n",
       "\n",
       "  C57_control_round2_filt_annot_fin.bed  \n",
       "0                          default-4062  \n",
       "3                          default-4068  \n",
       "4                          default-5671  \n",
       "5                          default-5675  \n",
       "6                          default-5678  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cut_control_diff = control_diff.iloc[save_rows,:]\n",
    "print(len(cut_control_diff))\n",
    "cut_control_diff.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "#for training the models we need these as less complicated bed files in this format:\n",
    "#chr, start, end, (anything else)\n",
    "#so going to cut these down and then save in round3 folder\n",
    "\n",
    "fin_file1 = '/home/hmummey/data/ATAC/annotated_peak_calling/round3/C57_control_diff_round2_filt_annot_fin.bed'\n",
    "fin_file2 = '/home/hmummey/data/ATAC/annotated_peak_calling/round3/C57_AMLNDiet_diff_round2_filt_annot_fin.bed'\n",
    "\n",
    "cut_control_diff.to_csv(fin_file1,index=False,header=False,columns=['chr','start','end'],sep='\\t')\n",
    "amln_diff.to_csv(fin_file2,index=False,header=False,columns=['chr','start','end'],sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "zs-deeplift",
   "language": "python",
   "name": "zs-deeplift"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
