{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Summary:\n",
    "This is based on my second version model interpretation notebook, modified from one Zeyang shared. It reads in the positive peak sequences (based on methods from the previous notebook) and then uses DeepLift to calculate nt importance scores for all of them. Then it has some preliminary visualization techniques. Here we are using it to repeat previous interpretation methods on my new set of models.\n",
    "\n",
    "#### Changes from the original version:\n",
    "- Combining peaks from control and AMLN sets into one peak set... use mergePeaks and then take the union; if they overlap, take the 300bp center\n",
    "- Remove peaks not predicted to be positive (based on model.predict and a 0.05 threshold) --> consider if we want to standardize these by the 2 model's predictions???\n",
    "- Going to considering predictions from Zeyang's IL4 model so generating importance scores from this (BUT NOT cutting down peaks based on them)\n",
    "\n",
    "#### New Considerations - CHANGES FROM PREV V3 NOTEBOOK:\n",
    "- Which peaks sets to generate importance scores from??? --> start with the same one as used previously in v2(all peaks found in BOTH datasets, then narrow down based on if the models predict them + or not)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import print_function\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "from itertools import compress\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from collections import Counter\n",
    "import sys\n",
    "import random\n",
    "import os\n",
    "import fnmatch\n",
    "import sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "from itertools import compress"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "dataDirectory = '/home/h1bennet/strains/data/ATAC/control_cohort2/'\n",
    "workingDirectory = '/home/h1bennet//strains_machinelearning/results/00_New_ATAC_H3K27Ac_Model/'\n",
    "if not os.path.isdir(workingDirectory):\n",
    "    os.mkdir(workingDirectory)\n",
    "os.chdir(workingDirectory)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['bed_files',\n",
       " 'model_weights',\n",
       " 'poised_active_merged_peaks.txt',\n",
       " 'bg_files',\n",
       " 'region_scores']"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.listdir()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge active and poised peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\tMax distance to merge: direct overlap required (-d given)\n",
      "\tMerging peaks... \n",
      "\tComparing /home/h1bennet/strains/results/06_Strains_Control_Cohort2_ATAC/poised_enhancers/C57Bl6J_poised_enhancer_peaks.txt (37146 total) and /home/h1bennet/strains/results/06_Strains_Control_Cohort2_ATAC/poised_enhancers/C57Bl6J_poised_enhancer_peaks.txt (37146 total)\n",
      "\tComparing /home/h1bennet/strains/results/06_Strains_Control_Cohort2_ATAC/poised_enhancers/C57Bl6J_poised_enhancer_peaks.txt (37146 total) and /home/h1bennet/strains/results/06b_Strains_Control_Combined_H3K27Ac/active_enhancers/C57Bl6J_active_enhancer_peaks.txt (36151 total)\n",
      "\tComparing /home/h1bennet/strains/results/06b_Strains_Control_Combined_H3K27Ac/active_enhancers/C57Bl6J_active_enhancer_peaks.txt (36151 total) and /home/h1bennet/strains/results/06_Strains_Control_Cohort2_ATAC/poised_enhancers/C57Bl6J_poised_enhancer_peaks.txt (37146 total)\n",
      "\tComparing /home/h1bennet/strains/results/06b_Strains_Control_Combined_H3K27Ac/active_enhancers/C57Bl6J_active_enhancer_peaks.txt (36151 total) and /home/h1bennet/strains/results/06b_Strains_Control_Combined_H3K27Ac/active_enhancers/C57Bl6J_active_enhancer_peaks.txt (36151 total)\n",
      "\n",
      "/home/h1bennet/strains/results/06_Strains_Control_Cohort2_ATAC/poised_enhancers/C57Bl6J_poised_enhancer_peaks.txt\t/home/h1bennet/strains/results/06b_Strains_Control_Combined_H3K27Ac/active_enhancers/C57Bl6J_active_enhancer_peaks.txt\tTotal\tName\n",
      "\tX\t11178\t/home/h1bennet/strains/results/06b_Strains_Control_Combined_H3K27Ac/active_enhancers/C57Bl6J_active_enhancer_peaks.txt\n",
      "X\t\t12173\t/home/h1bennet/strains/results/06_Strains_Control_Cohort2_ATAC/poised_enhancers/C57Bl6J_poised_enhancer_peaks.txt\n",
      "X\tX\t24973\t/home/h1bennet/strains/results/06_Strains_Control_Cohort2_ATAC/poised_enhancers/C57Bl6J_poised_enhancer_peaks.txt|/home/h1bennet/strains/results/06b_Strains_Control_Combined_H3K27Ac/active_enhancers/C57Bl6J_active_enhancer_peaks.txt\n"
     ]
    }
   ],
   "source": [
    "# %%bash\n",
    "\n",
    "# mergePeaks -d given \\\n",
    "# /home/h1bennet/strains/results/06_Strains_Control_Cohort2_ATAC/poised_enhancers/C57Bl6J_poised_enhancer_peaks.txt \\\n",
    "# /home/h1bennet/strains/results/06b_Strains_Control_Combined_H3K27Ac/active_enhancers/C57Bl6J_active_enhancer_peaks.txt \\\n",
    "# > ./poised_active_merged_peaks.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read in the combined peak file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "48324\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peakid</th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Parent files</th>\n",
       "      <th>Total subpeaks</th>\n",
       "      <th>/home/h1bennet/strains/results/06_strains_control_cohort2_atac/poised_enhancers/c57bl6j_poised_enhancer_peaks.txt</th>\n",
       "      <th>/home/h1bennet/strains/results/06b_strains_control_combined_h3k27ac/active_enhancers/c57bl6j_active_enhancer_peaks.txt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>Merged-chr5-147894194-1</td>\n",
       "      <td>chr5</td>\n",
       "      <td>147894094</td>\n",
       "      <td>147894294</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>/home/h1bennet/strains/results/06b_Strains_Con...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-chr5-147894185-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>Merged-chr15-53399324-1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>53399224</td>\n",
       "      <td>53399424</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>/home/h1bennet/strains/results/06b_Strains_Con...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-chr15-53399288-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>Merged-chr10-95323630-1</td>\n",
       "      <td>chr10</td>\n",
       "      <td>95323530</td>\n",
       "      <td>95323730</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>/home/h1bennet/strains/results/06b_Strains_Con...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-chr10-95323630-1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>Merged-chr12-111743521-1</td>\n",
       "      <td>chr12</td>\n",
       "      <td>111743421</td>\n",
       "      <td>111743621</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>/home/h1bennet/strains/results/06b_Strains_Con...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-chr12-111743523-2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>Merged-chr11-100583077-1</td>\n",
       "      <td>chr11</td>\n",
       "      <td>100582977</td>\n",
       "      <td>100583177</td>\n",
       "      <td>+</td>\n",
       "      <td>0.0</td>\n",
       "      <td>/home/h1bennet/strains/results/06b_Strains_Con...</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Merged-chr11-100583077-1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     Peakid    Chr      Start        End Strand  Stat  \\\n",
       "0   Merged-chr5-147894194-1   chr5  147894094  147894294      +   0.0   \n",
       "1   Merged-chr15-53399324-1  chr15   53399224   53399424      +   0.0   \n",
       "2   Merged-chr10-95323630-1  chr10   95323530   95323730      +   0.0   \n",
       "3  Merged-chr12-111743521-1  chr12  111743421  111743621      +   0.0   \n",
       "4  Merged-chr11-100583077-1  chr11  100582977  100583177      +   0.0   \n",
       "\n",
       "                                        Parent files  Total subpeaks  \\\n",
       "0  /home/h1bennet/strains/results/06b_Strains_Con...               1   \n",
       "1  /home/h1bennet/strains/results/06b_Strains_Con...               1   \n",
       "2  /home/h1bennet/strains/results/06b_Strains_Con...               1   \n",
       "3  /home/h1bennet/strains/results/06b_Strains_Con...               1   \n",
       "4  /home/h1bennet/strains/results/06b_Strains_Con...               1   \n",
       "\n",
       "  /home/h1bennet/strains/results/06_strains_control_cohort2_atac/poised_enhancers/c57bl6j_poised_enhancer_peaks.txt  \\\n",
       "0                                                NaN                                                                  \n",
       "1                                                NaN                                                                  \n",
       "2                                                NaN                                                                  \n",
       "3                                                NaN                                                                  \n",
       "4                                                NaN                                                                  \n",
       "\n",
       "  /home/h1bennet/strains/results/06b_strains_control_combined_h3k27ac/active_enhancers/c57bl6j_active_enhancer_peaks.txt  \n",
       "0                            Merged-chr5-147894185-2                                                                      \n",
       "1                            Merged-chr15-53399288-3                                                                      \n",
       "2                            Merged-chr10-95323630-1                                                                      \n",
       "3                           Merged-chr12-111743523-2                                                                      \n",
       "4                           Merged-chr11-100583077-1                                                                      "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#reading in the mergePeaks file (should be the union!)\n",
    "merged_peaks = pd.read_table('./poised_active_merged_peaks.txt', index_col=0)\n",
    "merged_peaks.index = merged_peaks.index.rename('peakID')\n",
    "merged_peaks = merged_peaks.reset_index()\n",
    "merged_peaks.columns = merged_peaks.columns.str.capitalize()\n",
    "print(len(merged_peaks))\n",
    "merged_peaks.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Convert to a .bed file"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Bed file format requires:\n",
    "- chrom - name of the chromosome or scaffold. Any valid seq_region_name can be used, and chromosome names can be given with or without the 'chr' prefix.\n",
    "- chromStart - Start position of the feature in standard chromosomal coordinates (i.e. first base is 0).\n",
    "- chromEnd - End position of the feature in standard chromosomal coordinates\n",
    "\n",
    "Optional columns: name, score, strand, thickStart, thickEnd, itemRgb, blockCount, blockSizes, blockStarts\n",
    "\n",
    "from: https://m.ensembl.org/info/website/upload/bed.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "#just going to use the required cols (and maybe strand and score? why not, I guess)\n",
    "save_cols = ['Chr', 'Start', 'End', 'Peakid', 'Stat', 'Strand']\n",
    "merged_bed = merged_peaks.loc[:,save_cols]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Peakid</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr5</td>\n",
       "      <td>147894094</td>\n",
       "      <td>147894294</td>\n",
       "      <td>Merged-chr5-147894194-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>53399224</td>\n",
       "      <td>53399424</td>\n",
       "      <td>Merged-chr15-53399324-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>95323530</td>\n",
       "      <td>95323730</td>\n",
       "      <td>Merged-chr10-95323630-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr12</td>\n",
       "      <td>111743421</td>\n",
       "      <td>111743621</td>\n",
       "      <td>Merged-chr12-111743521-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr11</td>\n",
       "      <td>100582977</td>\n",
       "      <td>100583177</td>\n",
       "      <td>Merged-chr11-100583077-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End                    Peakid  Stat Strand\n",
       "0   chr5  147894094  147894294   Merged-chr5-147894194-1   0.0      +\n",
       "1  chr15   53399224   53399424   Merged-chr15-53399324-1   0.0      +\n",
       "2  chr10   95323530   95323730   Merged-chr10-95323630-1   0.0      +\n",
       "3  chr12  111743421  111743621  Merged-chr12-111743521-1   0.0      +\n",
       "4  chr11  100582977  100583177  Merged-chr11-100583077-1   0.0      +"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_bed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "48324\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Peakid</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Strand</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr5</td>\n",
       "      <td>147894094</td>\n",
       "      <td>147894294</td>\n",
       "      <td>Merged-chr5-147894194-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>53399224</td>\n",
       "      <td>53399424</td>\n",
       "      <td>Merged-chr15-53399324-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>95323530</td>\n",
       "      <td>95323730</td>\n",
       "      <td>Merged-chr10-95323630-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr12</td>\n",
       "      <td>111743421</td>\n",
       "      <td>111743621</td>\n",
       "      <td>Merged-chr12-111743521-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr11</td>\n",
       "      <td>100582977</td>\n",
       "      <td>100583177</td>\n",
       "      <td>Merged-chr11-100583077-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End                    Peakid  Stat Strand\n",
       "0   chr5  147894094  147894294   Merged-chr5-147894194-1   0.0      +\n",
       "1  chr15   53399224   53399424   Merged-chr15-53399324-1   0.0      +\n",
       "2  chr10   95323530   95323730   Merged-chr10-95323630-1   0.0      +\n",
       "3  chr12  111743421  111743621  Merged-chr12-111743521-1   0.0      +\n",
       "4  chr11  100582977  100583177  Merged-chr11-100583077-1   0.0      +"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(merged_bed))\n",
    "merged_bed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# save this bc\n",
    "save_name = \"./bed_files/poised_active_merged_peaks.bed\"\n",
    "merged_bed.to_csv(save_name, index=False, sep=\"\\t\", header=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Read sequences\n",
    "Adapted from the previous notebook since we don't have fasta files for the + peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "from __future__ import print_function\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from collections import Counter\n",
    "\n",
    "import sys\n",
    "import random\n",
    "\n",
    "import os\n",
    "import fnmatch\n",
    "\n",
    "import sys"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "#both of these can be imported to python in the terminal in zs-deeplift env\n",
    "\n",
    "from Bio import SeqIO\n",
    "from Bio.Seq import Seq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Load in the mm10 genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_genome(ref_path):\n",
    "    ref_dict = {}\n",
    "    for seq in SeqIO.parse(ref_path, \"fasta\"):\n",
    "        chromID = seq.id\n",
    "        chromSeq = (str(seq.seq)).upper()\n",
    "        ref_dict[chromID] = chromSeq\n",
    "    return ref_dict"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "ref_path = \"/home/zes017/genomes/mm10/mm10.fa\"\n",
    "genomes = load_genome(ref_path)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Prepare data from a BED/peak file\n",
    "Almost the correct output format! Just need one more term in the returned list for each element in this format: 'chr5_140648065_140648415_C57'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "def data_prep(path, genomes, label, strain, size=500, rep=1, shift=None):\n",
    "    print (\"Scaled to %d\" % (size))\n",
    "    if shift is None:\n",
    "        shifts = size//rep*np.arange(-rep//2+1,rep//2+1)\n",
    "    elif type(shift) is int:\n",
    "        shifts = np.abs(shift)*np.arange(-rep//2+1,rep//2+1)\n",
    "    elif type(shift) is list:\n",
    "        if len(shift) != rep:\n",
    "            raise TypeError(\"Number of shift unequal to replicates\")\n",
    "        shifts = np.array(shift)\n",
    "    if rep == 1 and shift is not None:\n",
    "        random_shift = True\n",
    "        print(\"Random shifting within %s\" % (str(-np.abs(shift))+'~'+str(np.abs(shift))))\n",
    "    else:\n",
    "        random_shift = False\n",
    "        print(\"Generate replicates by shifting %s bp\" % (','.join(shifts.astype(str))))\n",
    "    \n",
    "    data_list = []\n",
    "    for line in open(path):\n",
    "        elems = line.split()\n",
    "        chromID = elems[0]\n",
    "        start, end = int(elems[1]), int(elems[2])\n",
    "        length = end-start\n",
    "        mid = (start+end)//2\n",
    "        for s in shifts:\n",
    "            if random_shift:\n",
    "                new_mid = mid + np.random.choice(np.arange(-np.abs(shift), np.abs(shift)+1))\n",
    "            else:\n",
    "                new_mid = mid + s\n",
    "            start = new_mid - size//2\n",
    "            end = new_mid + size//2\n",
    "            seq = genomes[chromID][start:end]\n",
    "            if len(seq) != size:\n",
    "                continue\n",
    "                \n",
    "            #created combined name term, ex: chr5_140648065_140648415_C57\n",
    "            k = chromID + \"_\" + str(start) + \"_\" + str(end) + \"_\" + strain\n",
    "\n",
    "            data_point = (seq, label, chromID, start, end, k)\n",
    "            data_list.append(data_point)\n",
    "    return data_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaled to 300\n",
      "Generate replicates by shifting 0 bp\n",
      "48324\n"
     ]
    }
   ],
   "source": [
    "#control --> delete first col!\n",
    "data_path = \"./bed_files/poised_active_merged_peaks.bed\"\n",
    "orig_data = data_prep(data_path, genomes, 1, \"C57\", size=300, rep=1, shift=None)\n",
    "print(len(orig_data))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('TCTATTGCAGAGCGGGAAGCGGGAGGAGAGAAACGGGGCGAGTGTGGGGGGGGGGGATAGAAATCATCAGTTGATTTTTTATCCCAGAGGCTAAATTCTCAAAAGTGGGGAAGTTGGAACAATGTGTCATAAACAAACATTAGCAAGCAAAAAGAGTTCCAGATTTCTGAGAGAGTGCCCTTGCCGGGAGAGGGTACCCTAGTTCTCTCCGAGGAGCAGGGACTAAGTCAACCTCCGCCCCCGACTCCGCTTAGCTAATGCTCACAGTGCTGAGCAGAACAGCACCGGCCGGTTCCCATA', 1, 'chr5', 147894044, 147894344, 'chr5_147894044_147894344_C57')\n"
     ]
    }
   ],
   "source": [
    "print(orig_data[0])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Convert to one hot encode sequences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "import random\n",
    "from Bio import Seq\n",
    "\n",
    "def dataset2onehot(dataset, shuffle=True):\n",
    "    nucleotides = [\"A\", \"T\", \"C\", \"G\"]\n",
    "    def seq2onehot(seq):\n",
    "        onehot_list = []\n",
    "        for nuc in seq:\n",
    "            if nuc == \"N\":\n",
    "                onehot = [0.25 for _ in range(len(nucleotides))]\n",
    "                onehot_list.append(onehot)\n",
    "            else:\n",
    "                onehot = [0 for _ in range(len(nucleotides))]\n",
    "                onehot[nucleotides.index(nuc)] = 1\n",
    "                onehot_list.append(onehot)\n",
    "        return onehot_list\n",
    "    \n",
    "    def rc(seq):\n",
    "        return str((Seq.Seq(seq)).reverse_complement())\n",
    "    \n",
    "    onehot_dataset = []\n",
    "    for (seq, label, chromID, start, end, ID) in dataset:\n",
    "        onehot_dataset.append((seq2onehot(seq), label, (chromID, start, end, \"+\")))\n",
    "        onehot_dataset.append((seq2onehot(rc(seq)), label, (chromID, start, end, \"-\")))\n",
    "    \n",
    "    if shuffle:\n",
    "        random.shuffle(onehot_dataset)\n",
    "    \n",
    "    x_list, y_list, info_list = [], [], [] \n",
    "    for (x, y, info) in onehot_dataset:\n",
    "        x_list.append(x)\n",
    "        y_list.append(y)\n",
    "        info_list.append(info)\n",
    "    return np.array(x_list), np.array(y_list), info_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "x_vis, y_vis, info_vis = dataset2onehot(orig_data, False) #combined!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(96648, 300, 4)\n",
      "(96648,)\n"
     ]
    }
   ],
   "source": [
    "print(x_vis.shape)\n",
    "print(y_vis.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "48324\n"
     ]
    }
   ],
   "source": [
    "print(len(orig_data))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filter data for positive peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using TensorFlow backend.\n"
     ]
    }
   ],
   "source": [
    "#need to import the model first!\n",
    "import keras\n",
    "from keras.models import model_from_json"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Poised model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "#homeostatic model first\n",
    "keras_model_weights = workingDirectory+\"/model_weights/modelWeights_300bp_C57BL6J_poised_epo20_RandomShift0bp.h5\"\n",
    "keras_model_json = workingDirectory+\"/model_weights/modelWeights_300bp_C57BL6J_poised_epo20_RandomShift0bp.json\"\n",
    "\n",
    "keras_model = model_from_json(open(keras_model_json).read())\n",
    "keras_model.load_weights(keras_model_weights)\n",
    "keras_model.compile(loss=keras.losses.binary_crossentropy,\n",
    "        optimizer=keras.optimizers.Adam(lr=1e-2),\n",
    "        metrics=['accuracy'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:From /home/h1bennet/anaconda3/envs/zs-deeplift/lib/python3.7/site-packages/keras/backend/tensorflow_backend.py:431: The name tf.is_variable_initialized is deprecated. Please use tf.compat.v1.is_variable_initialized instead.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "peak_preds = keras_model.predict(x_vis).flatten()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "96648\n",
      "71270\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(array([ 4370.,  6005.,  5089.,  5043.,  4871.,  4960.,  6145.,  8077.,\n",
       "        14524., 37564.]),\n",
       " array([4.5299530e-06, 1.0000408e-01, 2.0000362e-01, 3.0000317e-01,\n",
       "        4.0000272e-01, 5.0000226e-01, 6.0000181e-01, 7.0000136e-01,\n",
       "        8.0000091e-01, 9.0000045e-01, 1.0000000e+00], dtype=float32),\n",
       " <a list of 10 Patch objects>)"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYMAAAD4CAYAAAAO9oqkAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAUNElEQVR4nO3df6xf9X3f8ecrNiRsSWoT30TMdmbWuVocpBpyB54ibSlExvBHTCUywdTiIjR3DKZ2i6pA9wcpBIlsSpCQCK0jXEzVxDDaDguZeRYhyjKFHzeFGAxFvgUGN7bwpQZKhEoGe++P78fTd+Z7fY/vT1/f50M6+p7zPp9zvp+Pf9yXz+ec79epKiRJi9uH5rsDkqT5ZxhIkgwDSZJhIEnCMJAkAUvnuwNTtWLFilqzZs18d0OSFpSf/OQnr1fV0LH1BRsGa9asYWRkZL67IUkLSpL/NajuNJEkyTCQJBkGkiQMA0kShoEkCcNAkoRhIEnCMJAkYRhIkljAn0CWpPl021Ovz8v73nDuilk5r1cGkiTDQJJkGEiSMAwkSRgGkiQMA0kShoEkCcNAkoRhIEnCMJAk0SEMknwkyRNJfppkf5I/aPV7kryU5Om2rG/1JLkjyWiSfUnO6zvXliQH2rKlr/65JM+0Y+5IktkYrCRpsC7fTfQucGFV/TzJacCPkjzc9v1eVT1wTPtLgLVtuQC4C7ggyZnATcAwUMBPkuyqqjdam63AY8BuYBPwMJKkOTHplUH1/LxtntaWOs4hm4F723GPAcuSnAVcDOytqiMtAPYCm9q+j1fVj6uqgHuBy6YxJknSCep0zyDJkiRPA4fp/UB/vO26tU0F3Z7kw622Eni17/CxVjtefWxAfVA/tiYZSTIyPj7epeuSpA46hUFVvV9V64FVwPlJzgFuBP4J8E+BM4GvtuaD5vtrCvVB/dhWVcNVNTw0NNSl65KkDk7oaaKqehP4AbCpqg61qaB3gT8Gzm/NxoDVfYetAg5OUl81oC5JmiNdniYaSrKsrZ8BfBH4qzbXT3vy5zLg2XbILuCq9lTRBuCtqjoE7AE2JlmeZDmwEdjT9r2dZEM711XAgzM7TEnS8XR5mugsYEeSJfTC4/6qeijJ95MM0ZvmeRr4N639buBSYBR4B7gaoKqOJLkFeLK1u7mqjrT1a4F7gDPoPUXkk0SSNIcmDYOq2gecO6B+4QTtC7hugn3bge0D6iPAOZP1RZI0O/wEsiTJMJAkGQaSJAwDSRKGgSQJw0CShGEgScIwkCRhGEiSMAwkSRgGkiQMA0kShoEkCcNAkoRhIEnCMJAkYRhIkjAMJEl0CIMkH0nyRJKfJtmf5A9a/ewkjyc5kOS+JKe3+ofb9mjbv6bvXDe2+gtJLu6rb2q10SQ3zPwwJUnH0+XK4F3gwqr6VWA9sCnJBuAbwO1VtRZ4A7imtb8GeKOq/jFwe2tHknXAFcBngU3At5MsSbIEuBO4BFgHXNnaSpLmyKRhUD0/b5untaWAC4EHWn0HcFlb39y2afsvSpJW31lV71bVS8AocH5bRqvqxar6BbCztZUkzZFO9wzav+CfBg4De4G/Bt6sqvdakzFgZVtfCbwK0Pa/BXyiv37MMRPVB/Vja5KRJCPj4+Ndui5J6qBTGFTV+1W1HlhF71/ynxnUrL1mgn0nWh/Uj21VNVxVw0NDQ5N3XJLUyQk9TVRVbwI/ADYAy5IsbbtWAQfb+hiwGqDt/yXgSH/9mGMmqkuS5kiXp4mGkixr62cAXwSeBx4FLm/NtgAPtvVdbZu2//tVVa1+RXva6GxgLfAE8CSwtj2ddDq9m8y7ZmJwkqRulk7ehLOAHe2pnw8B91fVQ0meA3Ym+TrwFHB3a3838CdJRuldEVwBUFX7k9wPPAe8B1xXVe8DJLke2AMsAbZX1f4ZG6EkaVKThkFV7QPOHVB/kd79g2Prfwd8eYJz3QrcOqC+G9jdob+SpFngJ5AlSYaBJMkwkCRhGEiSMAwkSRgGkiQMA0kShoEkCcNAkoRhIEnCMJAkYRhIkjAMJEkYBpIkDANJEoaBJAnDQJKEYSBJwjCQJNEhDJKsTvJokueT7E/yO63+tSQ/S/J0Wy7tO+bGJKNJXkhycV99U6uNJrmhr352kseTHEhyX5LTZ3qgkqSJdbkyeA/4SlV9BtgAXJdkXdt3e1Wtb8tugLbvCuCzwCbg20mWJFkC3AlcAqwDruw7zzfaudYCbwDXzND4JEkdTBoGVXWoqv6yrb8NPA+sPM4hm4GdVfVuVb0EjALnt2W0ql6sql8AO4HNSQJcCDzQjt8BXDbVAUmSTtwJ3TNIsgY4F3i8la5Psi/J9iTLW20l8GrfYWOtNlH9E8CbVfXeMfVB7781yUiSkfHx8RPpuiTpODqHQZKPAn8G/G5V/S1wF/DLwHrgEPDNo00HHF5TqH+wWLWtqoaranhoaKhr1yVJk1japVGS0+gFwZ9W1Z8DVNVrffu/AzzUNseA1X2HrwIOtvVB9deBZUmWtquD/vaSpDnQ5WmiAHcDz1fVt/rqZ/U1+3Xg2ba+C7giyYeTnA2sBZ4AngTWtieHTqd3k3lXVRXwKHB5O34L8OD0hiVJOhFdrgw+D/wm8EySp1vt9+k9DbSe3pTOy8BvA1TV/iT3A8/RexLpuqp6HyDJ9cAeYAmwvar2t/N9FdiZ5OvAU/TCR5I0RyYNg6r6EYPn9Xcf55hbgVsH1HcPOq6qXqT3tJEkaR74CWRJkmEgSTIMJEkYBpIkDANJEoaBJAnDQJKEYSBJwjCQJGEYSJIwDCRJGAaSJAwDSRKGgSQJw0CShGEgScIwkCRhGEiS6BAGSVYneTTJ80n2J/mdVj8zyd4kB9rr8lZPkjuSjCbZl+S8vnNtae0PJNnSV/9ckmfaMXckGfTfbEqSZkmXK4P3gK9U1WeADcB1SdYBNwCPVNVa4JG2DXAJsLYtW4G7oBcewE3ABfT+v+ObjgZIa7O177hN0x+aJKmrScOgqg5V1V+29beB54GVwGZgR2u2A7isrW8G7q2ex4BlSc4CLgb2VtWRqnoD2Atsavs+XlU/rqoC7u07lyRpDpzQPYMka4BzgceBT1XVIegFBvDJ1mwl8GrfYWOtdrz62IC6JGmOdA6DJB8F/gz43ar62+M1HVCrKdQH9WFrkpEkI+Pj45N1WZLUUacwSHIavSD406r681Z+rU3x0F4Pt/oYsLrv8FXAwUnqqwbUP6CqtlXVcFUNDw0Ndem6JKmDLk8TBbgbeL6qvtW3axdw9ImgLcCDffWr2lNFG4C32jTSHmBjkuXtxvFGYE/b93aSDe29ruo7lyRpDizt0ObzwG8CzyR5utV+H7gNuD/JNcArwJfbvt3ApcAo8A5wNUBVHUlyC/Bka3dzVR1p69cC9wBnAA+3RZI0RyYNg6r6EYPn9QEuGtC+gOsmONd2YPuA+ghwzmR9kSTNDj+BLEkyDCRJhoEkCcNAkoRhIEnCMJAkYRhIkjAMJEkYBpIkDANJEoaBJAnDQJKEYSBJwjCQJGEYSJIwDCRJGAaSJAwDSRKGgSSJDmGQZHuSw0me7at9LcnPkjzdlkv79t2YZDTJC0ku7qtvarXRJDf01c9O8niSA0nuS3L6TA5QkjS5LlcG9wCbBtRvr6r1bdkNkGQdcAXw2XbMt5MsSbIEuBO4BFgHXNnaAnyjnWst8AZwzXQGJEk6cZOGQVX9EDjS8XybgZ1V9W5VvQSMAue3ZbSqXqyqXwA7gc1JAlwIPNCO3wFcdoJjkCRN03TuGVyfZF+bRlreaiuBV/vajLXaRPVPAG9W1XvH1AdKsjXJSJKR8fHxaXRdktRvqmFwF/DLwHrgEPDNVs+AtjWF+kBVta2qhqtqeGho6MR6LEma0NKpHFRVrx1dT/Id4KG2OQas7mu6CjjY1gfVXweWJVnarg7620uS5siUwiDJWVV1qG3+OnD0SaNdwHeTfAv4B8Ba4Al6VwBrk5wN/IzeTeZ/VVWV5FHgcnr3EbYAD051MJIWl9ueen2+u3DKmDQMknwP+AKwIskYcBPwhSTr6U3pvAz8NkBV7U9yP/Ac8B5wXVW9385zPbAHWAJsr6r97S2+CuxM8nXgKeDuGRudJKmTScOgqq4cUJ7wB3ZV3QrcOqC+G9g9oP4ivaeNJEnzxE8gS5IMA0mSYSBJwjCQJGEYSJIwDCRJGAaSJAwDSRKGgSQJw0CShGEgScIwkCRhGEiSMAwkSRgGkiQMA0kShoEkCcNAkoRhIEmiQxgk2Z7kcJJn+2pnJtmb5EB7Xd7qSXJHktEk+5Kc13fMltb+QJItffXPJXmmHXNHksz0ICVJx9flyuAeYNMxtRuAR6pqLfBI2wa4BFjblq3AXdALD+Am4ALgfOCmowHS2mztO+7Y95IkzbJJw6CqfggcOaa8GdjR1ncAl/XV762ex4BlSc4CLgb2VtWRqnoD2Atsavs+XlU/rqoC7u07lyRpjkz1nsGnquoQQHv9ZKuvBF7tazfWaserjw2oD5Rka5KRJCPj4+NT7Lok6VgzfQN50Hx/TaE+UFVtq6rhqhoeGhqaYhclSceaahi81qZ4aK+HW30MWN3XbhVwcJL6qgF1SdIcmmoY7AKOPhG0BXiwr35Ve6poA/BWm0baA2xMsrzdON4I7Gn73k6yoT1FdFXfuSRJc2TpZA2SfA/4ArAiyRi9p4JuA+5Pcg3wCvDl1nw3cCkwCrwDXA1QVUeS3AI82drdXFVHb0pfS++JpTOAh9siSZpDk4ZBVV05wa6LBrQt4LoJzrMd2D6gPgKcM1k/JEmzx08gS5IMA0mSYSBJwjCQJNHhBrIkTea2p16f7y5omrwykCQZBpIkw0CShGEgScIwkCRhGEiSMAwkSRgGkiQMA0kShoEkCb+OQjql+LUQmirDYA7N51/UG85dMW/vLenk5zSRJGl6YZDk5STPJHk6yUirnZlkb5ID7XV5qyfJHUlGk+xLcl7feba09geSbJnekCRJJ2omrgx+rarWV9Vw274BeKSq1gKPtG2AS4C1bdkK3AW98ABuAi4AzgduOhogkqS5MRvTRJuBHW19B3BZX/3e6nkMWJbkLOBiYG9VHamqN4C9wKZZ6JckaQLTvYFcwH9PUsAfVdU24FNVdQigqg4l+WRruxJ4te/YsVabqP4BSbbSu6rg05/+9DS7vrgsxqdM5uum+WL8tdbCN90w+HxVHWw/8Pcm+avjtM2AWh2n/sFiL2y2AQwPDw9sIx3lD2Wpu2lNE1XVwfZ6GPgLenP+r7XpH9rr4dZ8DFjdd/gq4OBx6pKkOTLlMEjy95N87Og6sBF4FtgFHH0iaAvwYFvfBVzVniraALzVppP2ABuTLG83jje2miRpjkxnmuhTwF8kOXqe71bVf0vyJHB/kmuAV4Avt/a7gUuBUeAd4GqAqjqS5Bbgydbu5qo6Mo1+TcrpA0n6/005DKrqReBXB9T/BrhoQL2A6yY413Zg+1T7IkmaHj+BLEkyDCRJhoEkCcNAkoRhIEnCMJAkYRhIkjAMJEkYBpIkDANJEoaBJAnDQJKEYSBJwjCQJGEYSJIwDCRJGAaSJAwDSRKGgSSJkygMkmxK8kKS0SQ3zHd/JGkxOSnCIMkS4E7gEmAdcGWSdfPbK0laPE6KMADOB0ar6sWq+gWwE9g8z32SpEVj6Xx3oFkJvNq3PQZccGyjJFuBrW3z50lemOL7rQBen+KxC5VjXhwW25gX23i5cfpj/oeDiidLGGRArT5QqNoGbJv2myUjVTU83fMsJI55cVhsY15s44XZG/PJMk00Bqzu214FHJynvkjSonOyhMGTwNokZyc5HbgC2DXPfZKkReOkmCaqqveSXA/sAZYA26tq/yy+5bSnmhYgx7w4LLYxL7bxwiyNOVUfmJqXJC0yJ8s0kSRpHhkGkqRTOwwm+4qLJB9Ocl/b/3iSNXPfy5nTYbz/IclzSfYleSTJwOeNF5KuX2OS5PIklWTBP4bYZcxJ/mX7vd6f5Ltz3ceZ1uHP9qeTPJrkqfbn+9L56OdMSbI9yeEkz06wP0nuaL8e+5KcN+03rapTcqF3I/qvgX8EnA78FFh3TJt/C/xhW78CuG+++z3L4/014O+19WsX8ni7jrm1+xjwQ+AxYHi++z0Hv89rgaeA5W37k/Pd7zkY8zbg2ra+Dnh5vvs9zTH/c+A84NkJ9l8KPEzvM1obgMen+56n8pVBl6+42AzsaOsPABclGfQBuIVg0vFW1aNV9U7bfIze5zkWsq5fY3IL8J+Av5vLzs2SLmP+18CdVfUGQFUdnuM+zrQuYy7g4239l1jgn1Oqqh8CR47TZDNwb/U8BixLctZ03vNUDoNBX3GxcqI2VfUe8BbwiTnp3czrMt5+19D7l8VCNumYk5wLrK6qh+ayY7Ooy+/zrwC/kuR/JnksyaY5693s6DLmrwG/kWQM2A38u7np2rw50b/vkzopPmcwS7p8xUWnr8FYIDqPJclvAMPAv5jVHs2+4445yYeA24HfmqsOzYEuv89L6U0VfYHe1d//SHJOVb05y32bLV3GfCVwT1V9M8k/A/6kjfn/zH735sWM/+w6la8MunzFxf9rk2QpvcvL412ancw6faVHki8C/xH4UlW9O0d9my2TjfljwDnAD5K8TG9uddcCv4nc9c/1g1X1v6vqJeAFeuGwUHUZ8zXA/QBV9WPgI/S+0O1UNeNf4XMqh0GXr7jYBWxp65cD3692d2YBmnS8bcrkj+gFwUKfR4ZJxlxVb1XViqpaU1Vr6N0n+VJVjcxPd2dElz/X/5XewwIkWUFv2ujFOe3lzOoy5leAiwCSfIZeGIzPaS/n1i7gqvZU0Qbgrao6NJ0TnrLTRDXBV1wkuRkYqapdwN30LidH6V0RXDF/PZ6ejuP9z8BHgf/S7pO/UlVfmrdOT1PHMZ9SOo55D7AxyXPA+8DvVdXfzF+vp6fjmL8CfCfJv6c3XfJbC/gfdiT5Hr1pvhXtPshNwGkAVfWH9O6LXAqMAu8AV0/7PRfwr5ckaYacytNEkqSODANJkmEgSTIMJEkYBpIkDANJEoaBJAn4v2IViRDR/iv3AAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "print(len(peak_preds))\n",
    "print(sum(i > 0.5 for i in peak_preds))\n",
    "plt.hist(peak_preds,color='skyblue')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Active Model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "#differential model second\n",
    "keras_model_weights2 = workingDirectory+\"/model_weights/modelWeights_300bp_C57BL6J_active_epo20_RandomShift0bp.h5\"\n",
    "keras_model_json2 = workingDirectory+\"/model_weights/modelWeights_300bp_C57BL6J_active_epo20_RandomShift0bp.json\"\n",
    "\n",
    "keras_model2 = model_from_json(open(keras_model_json2).read())\n",
    "keras_model2.load_weights(keras_model_weights2)\n",
    "keras_model2.compile(loss=keras.losses.binary_crossentropy,\n",
    "        optimizer=keras.optimizers.Adam(lr=1e-2),\n",
    "        metrics=['accuracy'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "peak_preds2 = keras_model2.predict(x_vis).flatten()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "96648\n",
      "83338\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(array([  342.,  2043.,  4074.,  3354.,  3497.,  4312.,  9310., 12650.,\n",
       "        21537., 35529.]),\n",
       " array([4.47034836e-07, 1.00000404e-01, 2.00000361e-01, 3.00000310e-01,\n",
       "        4.00000274e-01, 5.00000238e-01, 6.00000203e-01, 7.00000107e-01,\n",
       "        8.00000072e-01, 9.00000036e-01, 1.00000000e+00], dtype=float32),\n",
       " <a list of 10 Patch objects>)"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYMAAAD4CAYAAAAO9oqkAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4xLjEsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy8QZhcZAAAUIElEQVR4nO3df6zd9X3f8ecrNhC2JLUJlwjZzsw6R4sTaQ45A0+RtpRkxvBHTCUyOVKLi6y6o1C1W1SFdH+QQiY1m1IkJELnCIqp2hhG22EhmGcRomxVAB8XAhiKfAsMbm3hmxkoERoZ7L0/zsfV2eXY9/j+ONfX9/mQjs73vL+f7/d8PvhyXvf7+X7P96aqkCQtbR9Y6A5IkhaeYSBJMgwkSYaBJAnDQJIELF/oDszU+eefX2vXrl3obkjSonLgwIGfVNXY1PqiDYO1a9fS7XYXuhuStKgk+Z+D6k4TSZIMA0mSYSBJwjCQJGEYSJIwDCRJGAaSJAwDSRKGgSSJIcIgyQeTPJHkx0kOJvndVr87yUtJnmqPDa2eJLclGU/ydJKL+/a1Lcmh9tjWV/9skmfaNrclyXwMVpLmTLIwj3kyzO0o3gEuq6qfJjkL+B9JHm7rfruq7p/S/gpgXXtcCtwBXJrkPOAmoAMUcCDJnqp6vbXZATwGPARsBh5GkjQS0x4ZVM9P28uz2uNkfytzC3BP2+4xYEWSC4HLgX1VdawFwD5gc1v3kar6UfX+Buc9wFWzGJMk6RQNdc4gybIkTwFH6X2gP95W/fs2FXRrknNabRXwat/mE612svrEgPqgfuxI0k3SnZycHKbrkqQhDBUGVfVeVW0AVgOXJPk08HXgHwP/FDgP+FprPmhSq2ZQH9SPnVXVqarO2Nj77sAqSZqhU7qaqKreAH4AbK6qI20q6B3gD4FLWrMJYE3fZquBw9PUVw+oS5JGZJiricaSrGjL5wJfBP6qzfXTrvy5Cni2bbIHuKZdVbQReLOqjgB7gU1JViZZCWwC9rZ1byXZ2PZ1DfDA3A5TknQyw1xNdCGwK8kyeuFxX1U9mOT7ScboTfM8Bfzr1v4h4EpgHHgbuBagqo4luQXY39rdXFXH2vJ1wN3AufSuIvJKIkkaofQu4Fl8Op1O+ZfOJC2Yhfo61Cw/s5McqKrO1LrfQJYkGQaSJMNAkoRhIEnCMJAkYRhIkjAMJEkYBpIkDANJEoaBJAnDQJKEYSBJwjCQJGEYSJIwDCRJGAaSJAwDSRKGgSQJw0CShGEgSWKIMEjywSRPJPlxkoNJfrfVL0ryeJJDSe5Ncnarn9Nej7f1a/v29fVWfyHJ5X31za02nuTGuR+mJOlkhjkyeAe4rKr+CbAB2JxkI/At4NaqWge8Dmxv7bcDr1fVPwJube1Ish7YCnwK2Ax8J8myJMuA24ErgPXAV1pbSdKITBsG1fPT9vKs9ijgMuD+Vt8FXNWWt7TXtPVfSJJW311V71TVS8A4cEl7jFfVi1X1M2B3aytJGpGhzhm03+CfAo4C+4C/Bt6oqndbkwlgVVteBbwK0Na/CXy0vz5lmxPVB/VjR5Juku7k5OQwXZckDWGoMKiq96pqA7Ca3m/ynxzUrD3nBOtOtT6oHzurqlNVnbGxsek7LkkayildTVRVbwA/ADYCK5Isb6tWA4fb8gSwBqCt/zngWH99yjYnqkuSRmSYq4nGkqxoy+cCXwSeBx4Frm7NtgEPtOU97TVt/ferqlp9a7va6CJgHfAEsB9Y165OOpveSeY9czE4SdJwlk/fhAuBXe2qnw8A91XVg0meA3Yn+SbwJHBna38n8EdJxukdEWwFqKqDSe4DngPeBa6vqvcAktwA7AWWAXdV1cE5G6EkaVrp/dK++HQ6nep2uwvdDUlLVQad7hyBWX5mJzlQVZ2pdb+BLEkyDCRJhoEkCcNAkoRhIEnCMJAkYRhIkjAMJEkYBpIkDANJEoaBJAnDQJKEYSBJwjCQJGEYSJIwDCRJGAaSJAwDSRKGgSQJw0CSxBBhkGRNkkeTPJ/kYJLfbPVvJPmbJE+1x5V923w9yXiSF5Jc3lff3GrjSW7sq1+U5PEkh5Lcm+TsuR6oJOnEhjkyeBf4alV9EtgIXJ9kfVt3a1VtaI+HANq6rcCngM3Ad5IsS7IMuB24AlgPfKVvP99q+1oHvA5sn6PxSZKGMG0YVNWRqvrLtvwW8Dyw6iSbbAF2V9U7VfUSMA5c0h7jVfViVf0M2A1sSRLgMuD+tv0u4KqZDkiSdOpO6ZxBkrXAZ4DHW+mGJE8nuSvJylZbBbzat9lEq52o/lHgjap6d0p90PvvSNJN0p2cnDyVrkuSTmLoMEjyIeBPgd+qqr8F7gB+HtgAHAG+fbzpgM1rBvX3F6t2VlWnqjpjY2PDdl2SNI3lwzRKcha9IPjjqvozgKp6rW/9d4EH28sJYE3f5quBw215UP0nwIoky9vRQX97SdIIDHM1UYA7geer6vf76hf2NftF4Nm2vAfYmuScJBcB64AngP3Aunbl0Nn0TjLvqaoCHgWubttvAx6Y3bAkSadimCODzwG/DDyT5KlW+x16VwNtoDel8zLwawBVdTDJfcBz9K5Eur6q3gNIcgOwF1gG3FVVB9v+vgbsTvJN4El64SNJGpH0fjFffDqdTnW73YXuhqSlKoNOd47ALD+zkxyoqs7Uut9AliQZBpIkw0CShGEgScIwkCRhGEiSMAwkSRgGkiQMA0kShoEkCcNAkoRhIEnCMJAkYRhIkhjyL51J0mlpoW4jfQbyyECSZBhIkgwDSRKGgSQJw0CSxBBhkGRNkkeTPJ/kYJLfbPXzkuxLcqg9r2z1JLktyXiSp5Nc3Levba39oSTb+uqfTfJM2+a2xEsEJGmUhjkyeBf4alV9EtgIXJ9kPXAj8EhVrQMeaa8BrgDWtccO4A7ohQdwE3ApcAlw0/EAaW129G23efZDkyQNa9owqKojVfWXbfkt4HlgFbAF2NWa7QKuastbgHuq5zFgRZILgcuBfVV1rKpeB/YBm9u6j1TVj6qqgHv69iVJGoFTOmeQZC3wGeBx4GNVdQR6gQFc0JqtAl7t22yi1U5WnxhQH/T+O5J0k3QnJydPpeuSpJMYOgySfAj4U+C3qupvT9Z0QK1mUH9/sWpnVXWqqjM2NjZdlyVJQxoqDJKcRS8I/riq/qyVX2tTPLTno60+Aazp23w1cHia+uoBdUnSiAxzNVGAO4Hnq+r3+1btAY5fEbQNeKCvfk27qmgj8GabRtoLbEqysp043gTsbeveSrKxvdc1ffuSJI3AMDeq+xzwy8AzSZ5qtd8Bfg+4L8l24BXgy23dQ8CVwDjwNnAtQFUdS3ILsL+1u7mqjrXl64C7gXOBh9tDkjQi6V3As/h0Op3qdrsL3Q1JC2kpfiVplp/ZSQ5UVWdq3W8gS5IMA0mSYSBJwjCQJGEYSJIwDCRJGAaSJAwDSRKGgSQJw0CShGEgScIwkCRhGEiSMAwkSRgGkiQMA0kShoEkCcNAkoRhIEnCMJAkMUQYJLkrydEkz/bVvpHkb5I81R5X9q37epLxJC8kubyvvrnVxpPc2Fe/KMnjSQ4luTfJ2XM5QEnS9IY5Mrgb2DygfmtVbWiPhwCSrAe2Ap9q23wnybIky4DbgSuA9cBXWluAb7V9rQNeB7bPZkCSpFM3bRhU1Q+BY0Pubwuwu6reqaqXgHHgkvYYr6oXq+pnwG5gS5IAlwH3t+13AVed4hgkSbM0m3MGNyR5uk0jrWy1VcCrfW0mWu1E9Y8Cb1TVu1PqAyXZkaSbpDs5OTmLrkuS+s00DO4Afh7YABwBvt3qGdC2ZlAfqKp2VlWnqjpjY2On1mNJ0gktn8lGVfXa8eUk3wUebC8ngDV9TVcDh9vyoPpPgBVJlrejg/72kqQRmdGRQZIL+17+InD8SqM9wNYk5yS5CFgHPAHsB9a1K4fOpneSeU9VFfAocHXbfhvwwEz6JEmauWmPDJJ8D/g8cH6SCeAm4PNJNtCb0nkZ+DWAqjqY5D7gOeBd4Pqqeq/t5wZgL7AMuKuqDra3+BqwO8k3gSeBO+dsdJKkoaT3y/ni0+l0qtvtLnQ3JC2kDDrteIab5Wd2kgNV1Zla9xvIkiTDQJJkGEiSMAwkSRgGkiQMA0kShoEkCcNAksQM700kSf+fpfjlrzOMRwaSJMNAkmQYSJIwDCRJGAaSJAwDSRKGgSQJw0CShGEgScIwkCRhGEiSGCIMktyV5GiSZ/tq5yXZl+RQe17Z6klyW5LxJE8nubhvm22t/aEk2/rqn03yTNvmtsSbnEjSqA1zZHA3sHlK7UbgkapaBzzSXgNcAaxrjx3AHdALD+Am4FLgEuCm4wHS2uzo227qe0mS5tm0YVBVPwSOTSlvAXa15V3AVX31e6rnMWBFkguBy4F9VXWsql4H9gGb27qPVNWPqqqAe/r2JUkakZmeM/hYVR0BaM8XtPoq4NW+dhOtdrL6xIC6JGmE5voE8qD5/ppBffDOkx1Jukm6k5OTM+yiJGmqmYbBa22Kh/Z8tNUngDV97VYDh6eprx5QH6iqdlZVp6o6Y2NjM+y6JGmqmYbBHuD4FUHbgAf66te0q4o2Am+2aaS9wKYkK9uJ403A3rburSQb21VE1/TtS5I0ItP+2csk3wM+D5yfZILeVUG/B9yXZDvwCvDl1vwh4EpgHHgbuBagqo4luQXY39rdXFXHT0pfR++KpXOBh9tDkjRC6V3Es/h0Op3qdrsL3Q3p9OLXdM58s/zMTnKgqjpT634DWZJkGEiSDANJEoaBJAnDQJKEYSBJwjCQJGEYSJIwDCRJGAaSJAwDSRKGgSQJw0CShGEgScIwkCRhGEiSMAwkSRgGkiQMA0kShoEkiVmGQZKXkzyT5Kkk3VY7L8m+JIfa88pWT5LbkowneTrJxX372dbaH0qybXZDkiSdqrk4MviFqtpQVZ32+kbgkapaBzzSXgNcAaxrjx3AHdALD+Am4FLgEuCm4wEiSRqN+Zgm2gLsasu7gKv66vdUz2PAiiQXApcD+6rqWFW9DuwDNs9DvyRJJzDbMCjgvyU5kGRHq32sqo4AtOcLWn0V8GrfthOtdqK6JGlEls9y+89V1eEkFwD7kvzVSdpmQK1OUn//DnqBswPg4x//+Kn2VZJ0ArM6Mqiqw+35KPDn9Ob8X2vTP7Tno635BLCmb/PVwOGT1Ae9386q6lRVZ2xsbDZdlyT1mXEYJPn7ST58fBnYBDwL7AGOXxG0DXigLe8BrmlXFW0E3mzTSHuBTUlWthPHm1pNkjQis5km+hjw50mO7+dPquq/JtkP3JdkO/AK8OXW/iHgSmAceBu4FqCqjiW5Bdjf2t1cVcdm0S9J0ilK1cDp+dNep9Opbre70N2Q3i+DToNJc2SWn9lJDvR9FeDvzPYEshaLhfqAWqS/bEhLjbejkCQZBpIkp4l0JnPuXhqaYaD55QeytCg4TSRJMgwkSYaBJAnDQJKEYSBJwjCQJGEYSJLwewaj5TX3kk5THhlIkgwDSZJhIEnCMJAkYRhIkjAMJEkYBpIkDANJEqdRGCTZnOSFJONJblzo/kjSUnJafAM5yTLgduBfAhPA/iR7quq5eXrDedmtJC1Wp8uRwSXAeFW9WFU/A3YDWxa4T5K0ZJwWRwbAKuDVvtcTwKVTGyXZAexoL3+a5IUZvt/5wE9muO1i5ZiXhqU25qU2XkhmO+Z/MKh4uoTBoHmbel+haiewc9ZvlnSrqjPb/SwmjnlpWGpjXmrjhfkb8+kyTTQBrOl7vRo4vEB9kaQl53QJg/3AuiQXJTkb2ArsWeA+SdKScVpME1XVu0luAPYCy4C7qurgPL7lrKeaFiHHvDQstTEvtfHCPI05Ve+bmpckLTGnyzSRJGkBGQaSpDM7DKa7xUWSc5Lc29Y/nmTt6Hs5d4YY779N8lySp5M8kmTg9caLybC3MUlydZJKsugvQxxmzEn+Vfu3PpjkT0bdx7k2xM/2x5M8muTJ9vN95UL0c64kuSvJ0STPnmB9ktzW/ns8neTiWb9pVZ2RD3onov8a+IfA2cCPgfVT2vw68AdteStw70L3e57H+wvA32vL1y3m8Q475tbuw8APgceAzkL3ewT/zuuAJ4GV7fUFC93vEYx5J3BdW14PvLzQ/Z7lmP85cDHw7AnWXwk8TO87WhuBx2f7nmfykcEwt7jYAuxqy/cDX0gW7Y2Lph1vVT1aVW+3l4/R+z7HYjbsbUxuAf4D8L9H2bl5MsyYfxW4vapeB6iqoyPu41wbZswFfKQt/xyL/HtKVfVD4NhJmmwB7qmex4AVSS6czXueyWEw6BYXq07UpqreBd4EPjqS3s29Ycbbbzu93ywWs2nHnOQzwJqqenCUHZtHw/w7fwL4RJK/SPJYks0j6938GGbM3wB+KckE8BDwG6Pp2oI51f/fp3VafM9gngxzi4uhboOxSAw9liS/BHSAfzGvPZp/Jx1zkg8AtwK/MqoOjcAw/87L6U0VfZ7e0d9/T/Lpqnpjnvs2X4YZ81eAu6vq20n+GfBHbcz/d/67tyDm/LPrTD4yGOYWF3/XJslyeoeXJzs0O50NdUuPJF8E/h3wpap6Z0R9my/TjfnDwKeBHyR5md7c6p5FfhJ52J/rB6rq/1TVS8AL9MJhsRpmzNuB+wCq6kfAB+ndxO5MNee38DmTw2CYW1zsAba15auB71c7O7MITTveNmXyn+gFwWKfR4ZpxlxVb1bV+VW1tqrW0jtP8qWq6i5Md+fEMD/X/4XexQKkd4fLTwAvjrSXc2uYMb8CfAEgySfphcHkSHs5WnuAa9pVRRuBN6vqyGx2eMZOE9UJbnGR5GagW1V7gDvpHU6O0zsi2LpwPZ6dIcf7H4EPAf+5nSd/paq+tGCdnqUhx3xGGXLMe4FNSZ4D3gN+u6r+18L1enaGHPNXge8m+Tf0pkt+ZRH/YkeS79Gb5ju/nQe5CTgLoKr+gN55kSuBceBt4NpZv+ci/u8lSZojZ/I0kSRpSIaBJMkwkCQZBpIkDANJEoaBJAnDQJIE/D9fwFRbyxHsHQAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "print(len(peak_preds2))\n",
    "print(sum(i > 0.5 for i in peak_preds2))\n",
    "plt.hist(peak_preds2,color='red')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Cutting down the data lists"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Keep seqs where at least one of preds for (seq and rc) is above threshold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "#easiest thing is to just put both in the df then can be modular about cutting this down\n",
    "poised_preds = peak_preds[::2]\n",
    "poised_preds_rc = peak_preds[1::2]\n",
    "active_preds = peak_preds2[::2]\n",
    "active_preds_rc = peak_preds2[1::2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "merged_bed[\"Poised Predictions\"] = poised_preds\n",
    "merged_bed[\"Poised Predictions RC\"] = poised_preds_rc\n",
    "merged_bed[\"Active Predictions\"] = active_preds\n",
    "merged_bed[\"Active Predictions RC\"] = active_preds_rc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "48324\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Peakid</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Poised Predictions</th>\n",
       "      <th>Poised Predictions RC</th>\n",
       "      <th>Active Predictions</th>\n",
       "      <th>Active Predictions RC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr5</td>\n",
       "      <td>147894094</td>\n",
       "      <td>147894294</td>\n",
       "      <td>Merged-chr5-147894194-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.850177</td>\n",
       "      <td>0.580634</td>\n",
       "      <td>0.930514</td>\n",
       "      <td>0.886086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>53399224</td>\n",
       "      <td>53399424</td>\n",
       "      <td>Merged-chr15-53399324-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.843508</td>\n",
       "      <td>0.767881</td>\n",
       "      <td>0.864703</td>\n",
       "      <td>0.922945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>95323530</td>\n",
       "      <td>95323730</td>\n",
       "      <td>Merged-chr10-95323630-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.168567</td>\n",
       "      <td>0.238811</td>\n",
       "      <td>0.625731</td>\n",
       "      <td>0.631909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr12</td>\n",
       "      <td>111743421</td>\n",
       "      <td>111743621</td>\n",
       "      <td>Merged-chr12-111743521-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.922834</td>\n",
       "      <td>0.916620</td>\n",
       "      <td>0.924718</td>\n",
       "      <td>0.895533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr11</td>\n",
       "      <td>100582977</td>\n",
       "      <td>100583177</td>\n",
       "      <td>Merged-chr11-100583077-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.938389</td>\n",
       "      <td>0.930212</td>\n",
       "      <td>0.939787</td>\n",
       "      <td>0.963610</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End                    Peakid  Stat Strand  \\\n",
       "0   chr5  147894094  147894294   Merged-chr5-147894194-1   0.0      +   \n",
       "1  chr15   53399224   53399424   Merged-chr15-53399324-1   0.0      +   \n",
       "2  chr10   95323530   95323730   Merged-chr10-95323630-1   0.0      +   \n",
       "3  chr12  111743421  111743621  Merged-chr12-111743521-1   0.0      +   \n",
       "4  chr11  100582977  100583177  Merged-chr11-100583077-1   0.0      +   \n",
       "\n",
       "   Poised Predictions  Poised Predictions RC  Active Predictions  \\\n",
       "0            0.850177               0.580634            0.930514   \n",
       "1            0.843508               0.767881            0.864703   \n",
       "2            0.168567               0.238811            0.625731   \n",
       "3            0.922834               0.916620            0.924718   \n",
       "4            0.938389               0.930212            0.939787   \n",
       "\n",
       "   Active Predictions RC  \n",
       "0               0.886086  \n",
       "1               0.922945  \n",
       "2               0.631909  \n",
       "3               0.895533  \n",
       "4               0.963610  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(merged_bed))\n",
    "merged_bed.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_name = \"./bed_files/poised_active_merged_peaks_scored.bed\"\n",
    "merged_bed.to_csv(save_name, index=False, header=True, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Cut down bed df and then remake the dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(48324, 10)"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_bed.shape"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Poised model predicts positive"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "38482\n"
     ]
    }
   ],
   "source": [
    "#first cut (just take a row if one passes)\n",
    "save_rows1 = []\n",
    "for row in merged_bed.iterrows():\n",
    "    if row[1].loc['Poised Predictions'] > 0.5 or row[1].loc['Poised Predictions RC'] > 0.5:\n",
    "        save_rows1.append(row[0])\n",
    "        \n",
    "print(len(save_rows1))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "38482\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Peakid</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Poised Predictions</th>\n",
       "      <th>Poised Predictions RC</th>\n",
       "      <th>Active Predictions</th>\n",
       "      <th>Active Predictions RC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr5</td>\n",
       "      <td>147894094</td>\n",
       "      <td>147894294</td>\n",
       "      <td>Merged-chr5-147894194-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.850177</td>\n",
       "      <td>0.580634</td>\n",
       "      <td>0.930514</td>\n",
       "      <td>0.886086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>53399224</td>\n",
       "      <td>53399424</td>\n",
       "      <td>Merged-chr15-53399324-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.843508</td>\n",
       "      <td>0.767881</td>\n",
       "      <td>0.864703</td>\n",
       "      <td>0.922945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr12</td>\n",
       "      <td>111743421</td>\n",
       "      <td>111743621</td>\n",
       "      <td>Merged-chr12-111743521-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.922834</td>\n",
       "      <td>0.916620</td>\n",
       "      <td>0.924718</td>\n",
       "      <td>0.895533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr11</td>\n",
       "      <td>100582977</td>\n",
       "      <td>100583177</td>\n",
       "      <td>Merged-chr11-100583077-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.938389</td>\n",
       "      <td>0.930212</td>\n",
       "      <td>0.939787</td>\n",
       "      <td>0.963610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>chr11</td>\n",
       "      <td>98782374</td>\n",
       "      <td>98782574</td>\n",
       "      <td>Merged-chr11-98782474-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.059996</td>\n",
       "      <td>0.648414</td>\n",
       "      <td>0.830570</td>\n",
       "      <td>0.707561</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End                    Peakid  Stat Strand  \\\n",
       "0   chr5  147894094  147894294   Merged-chr5-147894194-1   0.0      +   \n",
       "1  chr15   53399224   53399424   Merged-chr15-53399324-1   0.0      +   \n",
       "3  chr12  111743421  111743621  Merged-chr12-111743521-1   0.0      +   \n",
       "4  chr11  100582977  100583177  Merged-chr11-100583077-1   0.0      +   \n",
       "5  chr11   98782374   98782574   Merged-chr11-98782474-1   0.0      +   \n",
       "\n",
       "   Poised Predictions  Poised Predictions RC  Active Predictions  \\\n",
       "0            0.850177               0.580634            0.930514   \n",
       "1            0.843508               0.767881            0.864703   \n",
       "3            0.922834               0.916620            0.924718   \n",
       "4            0.938389               0.930212            0.939787   \n",
       "5            0.059996               0.648414            0.830570   \n",
       "\n",
       "   Active Predictions RC  \n",
       "0               0.886086  \n",
       "1               0.922945  \n",
       "3               0.895533  \n",
       "4               0.963610  \n",
       "5               0.707561  "
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_bed_filter = merged_bed.iloc[save_rows1,:]\n",
    "print(len(merged_bed_filter))\n",
    "merged_bed_filter.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_name = \"./bed_files/poised_active_merged_peaks_scored_poised_filtered.bed\"\n",
    "merged_bed_filter.to_csv(save_name, index=False, header=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Active model predicts positive"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "43537\n"
     ]
    }
   ],
   "source": [
    "#first cut (just take a row if one passes)\n",
    "save_rows2 = []\n",
    "for row in merged_bed.iterrows():\n",
    "    if row[1].loc['Active Predictions'] > 0.5 or row[1].loc['Active Predictions RC'] > 0.5:\n",
    "        save_rows2.append(row[0])\n",
    "        \n",
    "print(len(save_rows2))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "43537\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Peakid</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Poised Predictions</th>\n",
       "      <th>Poised Predictions RC</th>\n",
       "      <th>Active Predictions</th>\n",
       "      <th>Active Predictions RC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr5</td>\n",
       "      <td>147894094</td>\n",
       "      <td>147894294</td>\n",
       "      <td>Merged-chr5-147894194-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.850177</td>\n",
       "      <td>0.580634</td>\n",
       "      <td>0.930514</td>\n",
       "      <td>0.886086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>53399224</td>\n",
       "      <td>53399424</td>\n",
       "      <td>Merged-chr15-53399324-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.843508</td>\n",
       "      <td>0.767881</td>\n",
       "      <td>0.864703</td>\n",
       "      <td>0.922945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>95323530</td>\n",
       "      <td>95323730</td>\n",
       "      <td>Merged-chr10-95323630-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.168567</td>\n",
       "      <td>0.238811</td>\n",
       "      <td>0.625731</td>\n",
       "      <td>0.631909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr12</td>\n",
       "      <td>111743421</td>\n",
       "      <td>111743621</td>\n",
       "      <td>Merged-chr12-111743521-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.922834</td>\n",
       "      <td>0.916620</td>\n",
       "      <td>0.924718</td>\n",
       "      <td>0.895533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr11</td>\n",
       "      <td>100582977</td>\n",
       "      <td>100583177</td>\n",
       "      <td>Merged-chr11-100583077-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.938389</td>\n",
       "      <td>0.930212</td>\n",
       "      <td>0.939787</td>\n",
       "      <td>0.963610</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End                    Peakid  Stat Strand  \\\n",
       "0   chr5  147894094  147894294   Merged-chr5-147894194-1   0.0      +   \n",
       "1  chr15   53399224   53399424   Merged-chr15-53399324-1   0.0      +   \n",
       "2  chr10   95323530   95323730   Merged-chr10-95323630-1   0.0      +   \n",
       "3  chr12  111743421  111743621  Merged-chr12-111743521-1   0.0      +   \n",
       "4  chr11  100582977  100583177  Merged-chr11-100583077-1   0.0      +   \n",
       "\n",
       "   Poised Predictions  Poised Predictions RC  Active Predictions  \\\n",
       "0            0.850177               0.580634            0.930514   \n",
       "1            0.843508               0.767881            0.864703   \n",
       "2            0.168567               0.238811            0.625731   \n",
       "3            0.922834               0.916620            0.924718   \n",
       "4            0.938389               0.930212            0.939787   \n",
       "\n",
       "   Active Predictions RC  \n",
       "0               0.886086  \n",
       "1               0.922945  \n",
       "2               0.631909  \n",
       "3               0.895533  \n",
       "4               0.963610  "
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_bed_filter_active = merged_bed.iloc[save_rows2,:]\n",
    "print(len(merged_bed_filter_active))\n",
    "merged_bed_filter_active.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_name = \"./bed_files/poised_active_merged_peaks_scored_active_filtered.bed\"\n",
    "merged_bed_filter_active.to_csv(save_name, index=False, header=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Both active and poised"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "37452\n"
     ]
    }
   ],
   "source": [
    "#first cut (just take a row if one passes)\n",
    "save_rows3 = []\n",
    "for row in merged_bed.iterrows():\n",
    "    if row[1].loc['Poised Predictions'] > 0.5 or row[1].loc['Poised Predictions RC'] > 0.5:\n",
    "        if row[1].loc['Active Predictions'] > 0.5 or row[1].loc['Active Predictions RC'] > 0.5:\n",
    "            save_rows3.append(row[0])\n",
    "        \n",
    "print(len(save_rows3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "37452\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Peakid</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Poised Predictions</th>\n",
       "      <th>Poised Predictions RC</th>\n",
       "      <th>Active Predictions</th>\n",
       "      <th>Active Predictions RC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr5</td>\n",
       "      <td>147894094</td>\n",
       "      <td>147894294</td>\n",
       "      <td>Merged-chr5-147894194-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.850177</td>\n",
       "      <td>0.580634</td>\n",
       "      <td>0.930514</td>\n",
       "      <td>0.886086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>53399224</td>\n",
       "      <td>53399424</td>\n",
       "      <td>Merged-chr15-53399324-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.843508</td>\n",
       "      <td>0.767881</td>\n",
       "      <td>0.864703</td>\n",
       "      <td>0.922945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr12</td>\n",
       "      <td>111743421</td>\n",
       "      <td>111743621</td>\n",
       "      <td>Merged-chr12-111743521-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.922834</td>\n",
       "      <td>0.916620</td>\n",
       "      <td>0.924718</td>\n",
       "      <td>0.895533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr11</td>\n",
       "      <td>100582977</td>\n",
       "      <td>100583177</td>\n",
       "      <td>Merged-chr11-100583077-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.938389</td>\n",
       "      <td>0.930212</td>\n",
       "      <td>0.939787</td>\n",
       "      <td>0.963610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>chr11</td>\n",
       "      <td>98782374</td>\n",
       "      <td>98782574</td>\n",
       "      <td>Merged-chr11-98782474-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.059996</td>\n",
       "      <td>0.648414</td>\n",
       "      <td>0.830570</td>\n",
       "      <td>0.707561</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End                    Peakid  Stat Strand  \\\n",
       "0   chr5  147894094  147894294   Merged-chr5-147894194-1   0.0      +   \n",
       "1  chr15   53399224   53399424   Merged-chr15-53399324-1   0.0      +   \n",
       "3  chr12  111743421  111743621  Merged-chr12-111743521-1   0.0      +   \n",
       "4  chr11  100582977  100583177  Merged-chr11-100583077-1   0.0      +   \n",
       "5  chr11   98782374   98782574   Merged-chr11-98782474-1   0.0      +   \n",
       "\n",
       "   Poised Predictions  Poised Predictions RC  Active Predictions  \\\n",
       "0            0.850177               0.580634            0.930514   \n",
       "1            0.843508               0.767881            0.864703   \n",
       "3            0.922834               0.916620            0.924718   \n",
       "4            0.938389               0.930212            0.939787   \n",
       "5            0.059996               0.648414            0.830570   \n",
       "\n",
       "   Active Predictions RC  \n",
       "0               0.886086  \n",
       "1               0.922945  \n",
       "3               0.895533  \n",
       "4               0.963610  \n",
       "5               0.707561  "
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_bed_filter_both = merged_bed.iloc[save_rows3,:]\n",
    "print(len(merged_bed_filter_both))\n",
    "merged_bed_filter_both.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "save_name = \"./bed_files/poised_active_merged_peaks_scored_poised_active_filtered.bed\"\n",
    "merged_bed_filter_both.to_csv(save_name, index=False, header=False, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Now remake the datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaled to 300\n",
      "Generate replicates by shifting 0 bp\n",
      "38482\n"
     ]
    }
   ],
   "source": [
    "#poised --> delete first col!\n",
    "data_path = \"./bed_files/poised_active_merged_peaks_scored_poised_filtered.bed\"\n",
    "poised_pos_data = data_prep(data_path, genomes, 1, \"C57\", size=300, rep=1, shift=None)\n",
    "print(len(poised_pos_data))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaled to 300\n",
      "Generate replicates by shifting 0 bp\n",
      "43537\n"
     ]
    }
   ],
   "source": [
    "#amln+ --> delete first col!\n",
    "data_path = \"./bed_files/poised_active_merged_peaks_scored_active_filtered.bed\"\n",
    "active_pos_data = data_prep(data_path, genomes, 1, \"C57\", size=300, rep=1, shift=None)\n",
    "print(len(active_pos_data))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Scaled to 300\n",
      "Generate replicates by shifting 0 bp\n",
      "37452\n"
     ]
    }
   ],
   "source": [
    "#amln+ AND control+ --> delete first col!\n",
    "data_path = \"./bed_files/poised_active_merged_peaks_scored_poised_active_filtered.bed\"\n",
    "both_pos_data = data_prep(data_path, genomes, 1, \"C57\", size=300, rep=1, shift=None)\n",
    "print(len(both_pos_data))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Score nucleotide importance with DeepLIFT\n",
    "## Do this twice for each model... the overlap set and the individual set"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "import deeplift\n",
    "from deeplift.layers import NonlinearMxtsMode\n",
    "import deeplift.conversion.kerasapi_conversion as kc\n",
    "from collections import OrderedDict\n",
    "from deeplift.util import get_shuffle_seq_ref_function\n",
    "from deeplift.dinuc_shuffle import dinuc_shuffle"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Define scoring function"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nonlinear_mxts_mode is set to: DeepLIFT_GenomicsDefault\n",
      "For layer 2 the preceding linear layer is 0 of type Conv1D;\n",
      "In accordance with nonlinear_mxts_mode=DeepLIFT_GenomicsDefault we are setting the NonlinearMxtsMode to Rescale\n",
      "Heads-up: current implementation assumes maxpool layer is followed by a linear transformation (conv/dense layer)\n",
      "For layer 7 the preceding linear layer is 5 of type Conv1D;\n",
      "In accordance with nonlinear_mxts_mode=DeepLIFT_GenomicsDefault we are setting the NonlinearMxtsMode to Rescale\n",
      "Heads-up: current implementation assumes maxpool layer is followed by a linear transformation (conv/dense layer)\n",
      "For layer 12 the preceding linear layer is 10 of type Conv1D;\n",
      "In accordance with nonlinear_mxts_mode=DeepLIFT_GenomicsDefault we are setting the NonlinearMxtsMode to Rescale\n",
      "For layer 16 the preceding linear layer is 15 of type Dense;\n",
      "In accordance with nonlinear_mxts_modeDeepLIFT_GenomicsDefault we are setting the NonlinearMxtsMode to RevealCancel\n",
      "Heads-up: I assume sigmoid is the output layer, not an intermediate one; if it's an intermediate layer then please bug me and I will implement the grad func\n",
      "For layer 18 the preceding linear layer is 17 of type Dense;\n",
      "In accordance with nonlinear_mxts_modeDeepLIFT_GenomicsDefault we are setting the NonlinearMxtsMode to RevealCancel\n",
      "TARGET LAYER SET TO 17\n"
     ]
    }
   ],
   "source": [
    "#control model\n",
    "#already compiled... making new one here that I'm not compiling\n",
    "keras_model_weights = workingDirectory+\"/model_weights/modelWeights_300bp_C57BL6J_poised_epo20_RandomShift0bp.h5\"\n",
    "keras_model_json = workingDirectory+\"/model_weights/modelWeights_300bp_C57BL6J_poised_epo20_RandomShift0bp.json\"\n",
    "\n",
    "keras_model1 = model_from_json(open(keras_model_json).read())\n",
    "keras_model1.load_weights(keras_model_weights)\n",
    "\n",
    "#new to run\n",
    "nonlinear_mxts_mode = NonlinearMxtsMode.DeepLIFT_GenomicsDefault\n",
    "deeplift_method = kc.convert_model_from_saved_files(\n",
    "            h5_file=keras_model_weights,\n",
    "            json_file=keras_model_json,\n",
    "            nonlinear_mxts_mode=nonlinear_mxts_mode)\n",
    "scoring_func = deeplift_method.get_target_contribs_func(find_scores_layer_idx=0, target_layer_idx=-2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Need to rerun this one too!"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "nonlinear_mxts_mode is set to: DeepLIFT_GenomicsDefault\n",
      "For layer 2 the preceding linear layer is 0 of type Conv1D;\n",
      "In accordance with nonlinear_mxts_mode=DeepLIFT_GenomicsDefault we are setting the NonlinearMxtsMode to Rescale\n",
      "Heads-up: current implementation assumes maxpool layer is followed by a linear transformation (conv/dense layer)\n",
      "For layer 7 the preceding linear layer is 5 of type Conv1D;\n",
      "In accordance with nonlinear_mxts_mode=DeepLIFT_GenomicsDefault we are setting the NonlinearMxtsMode to Rescale\n",
      "Heads-up: current implementation assumes maxpool layer is followed by a linear transformation (conv/dense layer)\n",
      "For layer 12 the preceding linear layer is 10 of type Conv1D;\n",
      "In accordance with nonlinear_mxts_mode=DeepLIFT_GenomicsDefault we are setting the NonlinearMxtsMode to Rescale\n",
      "For layer 16 the preceding linear layer is 15 of type Dense;\n",
      "In accordance with nonlinear_mxts_modeDeepLIFT_GenomicsDefault we are setting the NonlinearMxtsMode to RevealCancel\n",
      "Heads-up: I assume sigmoid is the output layer, not an intermediate one; if it's an intermediate layer then please bug me and I will implement the grad func\n",
      "For layer 18 the preceding linear layer is 17 of type Dense;\n",
      "In accordance with nonlinear_mxts_modeDeepLIFT_GenomicsDefault we are setting the NonlinearMxtsMode to RevealCancel\n",
      "TARGET LAYER SET TO 17\n"
     ]
    }
   ],
   "source": [
    "#control model\n",
    "#already compiled... making new one here that I'm not compiling\n",
    "keras_model_weights2 = workingDirectory+\"/model_weights/modelWeights_300bp_C57BL6J_active_epo20_RandomShift0bp.h5\"\n",
    "keras_model_json2 = workingDirectory+\"/model_weights/modelWeights_300bp_C57BL6J_active_epo20_RandomShift0bp.json\"\n",
    "\n",
    "keras_model2 = model_from_json(open(keras_model_json2).read())\n",
    "keras_model2.load_weights(keras_model_weights2)\n",
    "\n",
    "#new to run\n",
    "nonlinear_mxts_mode = NonlinearMxtsMode.DeepLIFT_GenomicsDefault\n",
    "deeplift_method2 = kc.convert_model_from_saved_files(\n",
    "            h5_file=keras_model_weights2,\n",
    "            json_file=keras_model_json2,\n",
    "            nonlinear_mxts_mode=nonlinear_mxts_mode)\n",
    "scoring_func2 = deeplift_method2.get_target_contribs_func(find_scores_layer_idx=0, target_layer_idx=-2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Specify reference and compute importance"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "37452"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(both_pos_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "43537"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(active_pos_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "38482"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(poised_pos_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "#making 3 onehotseqs from all 3 pos data sets\n",
    "x_vis_poised, y_vis_poised, info_vis_poised = dataset2onehot(poised_pos_data, False)\n",
    "x_vis_active, y_vis_active, info_vis_active = dataset2onehot(active_pos_data, False)\n",
    "x_vis_both, y_vis_both, info_vis_both = dataset2onehot(both_pos_data, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(76964, 300, 4)\n",
      "(76964,)\n",
      "(87074, 300, 4)\n",
      "(87074,)\n",
      "(74904, 300, 4)\n",
      "(74904,)\n"
     ]
    }
   ],
   "source": [
    "print(x_vis_poised.shape)\n",
    "print(y_vis_poised.shape)\n",
    "print(x_vis_active.shape)\n",
    "print(y_vis_active.shape)\n",
    "print(x_vis_both.shape)\n",
    "print(y_vis_both.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Poised model \n",
    "- Scores 1 = poised pos data\n",
    "- Scores 2 = active + poised pos data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done 0\n",
      "Done 16000\n",
      "Done 32000\n",
      "Done 48000\n",
      "Done 64000\n",
      "CPU times: user 2h 57min 16s, sys: 18min 43s, total: 3h 16min\n",
      "Wall time: 5min 48s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "batch_size = 128\n",
    "background = OrderedDict([('A', 0.25), ('C', 0.25), ('G', 0.25), ('T', 0.25)])\n",
    "scores1 = np.array(scoring_func(task_idx=0, input_data_list=[x_vis_poised], batch_size=batch_size, \n",
    "                               input_references_list=[np.array([background['A'], \n",
    "                                                                background['C'], \n",
    "                                                                background['G'], \n",
    "                                                                background['T']])[None,None,:]],\n",
    "                               progress_update=1000))\n",
    "scores1 = np.sum(scores1, axis=2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done 0\n",
      "Done 16000\n",
      "Done 32000\n",
      "Done 48000\n",
      "Done 64000\n",
      "CPU times: user 2h 48min 32s, sys: 16min 50s, total: 3h 5min 23s\n",
      "Wall time: 6min 9s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "batch_size = 128 \n",
    "background = OrderedDict([('A', 0.25), ('C', 0.25), ('G', 0.25), ('T', 0.25)])\n",
    "scores2 = np.array(scoring_func(task_idx=0, input_data_list=[x_vis_both], batch_size=batch_size, \n",
    "                               input_references_list=[np.array([background['A'], \n",
    "                                                                background['C'], \n",
    "                                                                background['G'], \n",
    "                                                                background['T']])[None,None,:]],\n",
    "                               progress_update=1000))\n",
    "scores2 = np.sum(scores2, axis=2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Active model\n",
    "- Scores 3 = active pos data\n",
    "- Scores 4 = active + poised pos data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done 0\n",
      "Done 16000\n",
      "Done 32000\n",
      "Done 48000\n",
      "Done 64000\n",
      "Done 80000\n",
      "CPU times: user 3h 15min 3s, sys: 20min 34s, total: 3h 35min 38s\n",
      "Wall time: 7min 24s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "batch_size = 128 \n",
    "background = OrderedDict([('A', 0.25), ('C', 0.25), ('G', 0.25), ('T', 0.25)])\n",
    "scores3 = np.array(scoring_func2(task_idx=0, input_data_list=[x_vis_active], batch_size=batch_size, \n",
    "                               input_references_list=[np.array([background['A'], \n",
    "                                                                background['C'], \n",
    "                                                                background['G'], \n",
    "                                                                background['T']])[None,None,:]],\n",
    "                               progress_update=1000))\n",
    "scores3 = np.sum(scores3, axis=2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Done 0\n",
      "Done 16000\n",
      "Done 32000\n",
      "Done 48000\n",
      "Done 64000\n",
      "CPU times: user 2h 46min 50s, sys: 16min 29s, total: 3h 3min 19s\n",
      "Wall time: 6min 9s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "batch_size = 128 \n",
    "background = OrderedDict([('A', 0.25), ('C', 0.25), ('G', 0.25), ('T', 0.25)])\n",
    "scores4 = np.array(scoring_func2(task_idx=0, input_data_list=[x_vis_both], batch_size=batch_size, \n",
    "                               input_references_list=[np.array([background['A'], \n",
    "                                                                background['C'], \n",
    "                                                                background['G'], \n",
    "                                                                background['T']])[None,None,:]],\n",
    "                               progress_update=1000))\n",
    "scores4 = np.sum(scores4, axis=2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Get nucleotide-level max importance from original and reverse compliment sequences"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Select peak names\n",
    "As an edit to Hannah's code I want to also propagate the peak id feature so that I can easily match with differential sequences from our analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Peakid</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Poised Predictions</th>\n",
       "      <th>Poised Predictions RC</th>\n",
       "      <th>Active Predictions</th>\n",
       "      <th>Active Predictions RC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr5</td>\n",
       "      <td>147894094</td>\n",
       "      <td>147894294</td>\n",
       "      <td>Merged-chr5-147894194-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.850177</td>\n",
       "      <td>0.580634</td>\n",
       "      <td>0.930514</td>\n",
       "      <td>0.886086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>53399224</td>\n",
       "      <td>53399424</td>\n",
       "      <td>Merged-chr15-53399324-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.843508</td>\n",
       "      <td>0.767881</td>\n",
       "      <td>0.864703</td>\n",
       "      <td>0.922945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr12</td>\n",
       "      <td>111743421</td>\n",
       "      <td>111743621</td>\n",
       "      <td>Merged-chr12-111743521-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.922834</td>\n",
       "      <td>0.916620</td>\n",
       "      <td>0.924718</td>\n",
       "      <td>0.895533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr11</td>\n",
       "      <td>100582977</td>\n",
       "      <td>100583177</td>\n",
       "      <td>Merged-chr11-100583077-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.938389</td>\n",
       "      <td>0.930212</td>\n",
       "      <td>0.939787</td>\n",
       "      <td>0.963610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>chr11</td>\n",
       "      <td>98782374</td>\n",
       "      <td>98782574</td>\n",
       "      <td>Merged-chr11-98782474-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.059996</td>\n",
       "      <td>0.648414</td>\n",
       "      <td>0.830570</td>\n",
       "      <td>0.707561</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End                    Peakid  Stat Strand  \\\n",
       "0   chr5  147894094  147894294   Merged-chr5-147894194-1   0.0      +   \n",
       "1  chr15   53399224   53399424   Merged-chr15-53399324-1   0.0      +   \n",
       "3  chr12  111743421  111743621  Merged-chr12-111743521-1   0.0      +   \n",
       "4  chr11  100582977  100583177  Merged-chr11-100583077-1   0.0      +   \n",
       "5  chr11   98782374   98782574   Merged-chr11-98782474-1   0.0      +   \n",
       "\n",
       "   Poised Predictions  Poised Predictions RC  Active Predictions  \\\n",
       "0            0.850177               0.580634            0.930514   \n",
       "1            0.843508               0.767881            0.864703   \n",
       "3            0.922834               0.916620            0.924718   \n",
       "4            0.938389               0.930212            0.939787   \n",
       "5            0.059996               0.648414            0.830570   \n",
       "\n",
       "   Active Predictions RC  \n",
       "0               0.886086  \n",
       "1               0.922945  \n",
       "3               0.895533  \n",
       "4               0.963610  \n",
       "5               0.707561  "
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_bed_filter.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Peakid</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Poised Predictions</th>\n",
       "      <th>Poised Predictions RC</th>\n",
       "      <th>Active Predictions</th>\n",
       "      <th>Active Predictions RC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr5</td>\n",
       "      <td>147894094</td>\n",
       "      <td>147894294</td>\n",
       "      <td>Merged-chr5-147894194-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.850177</td>\n",
       "      <td>0.580634</td>\n",
       "      <td>0.930514</td>\n",
       "      <td>0.886086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>53399224</td>\n",
       "      <td>53399424</td>\n",
       "      <td>Merged-chr15-53399324-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.843508</td>\n",
       "      <td>0.767881</td>\n",
       "      <td>0.864703</td>\n",
       "      <td>0.922945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>chr10</td>\n",
       "      <td>95323530</td>\n",
       "      <td>95323730</td>\n",
       "      <td>Merged-chr10-95323630-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.168567</td>\n",
       "      <td>0.238811</td>\n",
       "      <td>0.625731</td>\n",
       "      <td>0.631909</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr12</td>\n",
       "      <td>111743421</td>\n",
       "      <td>111743621</td>\n",
       "      <td>Merged-chr12-111743521-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.922834</td>\n",
       "      <td>0.916620</td>\n",
       "      <td>0.924718</td>\n",
       "      <td>0.895533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr11</td>\n",
       "      <td>100582977</td>\n",
       "      <td>100583177</td>\n",
       "      <td>Merged-chr11-100583077-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.938389</td>\n",
       "      <td>0.930212</td>\n",
       "      <td>0.939787</td>\n",
       "      <td>0.963610</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End                    Peakid  Stat Strand  \\\n",
       "0   chr5  147894094  147894294   Merged-chr5-147894194-1   0.0      +   \n",
       "1  chr15   53399224   53399424   Merged-chr15-53399324-1   0.0      +   \n",
       "2  chr10   95323530   95323730   Merged-chr10-95323630-1   0.0      +   \n",
       "3  chr12  111743421  111743621  Merged-chr12-111743521-1   0.0      +   \n",
       "4  chr11  100582977  100583177  Merged-chr11-100583077-1   0.0      +   \n",
       "\n",
       "   Poised Predictions  Poised Predictions RC  Active Predictions  \\\n",
       "0            0.850177               0.580634            0.930514   \n",
       "1            0.843508               0.767881            0.864703   \n",
       "2            0.168567               0.238811            0.625731   \n",
       "3            0.922834               0.916620            0.924718   \n",
       "4            0.938389               0.930212            0.939787   \n",
       "\n",
       "   Active Predictions RC  \n",
       "0               0.886086  \n",
       "1               0.922945  \n",
       "2               0.631909  \n",
       "3               0.895533  \n",
       "4               0.963610  "
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_bed_filter_active.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chr</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Peakid</th>\n",
       "      <th>Stat</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Poised Predictions</th>\n",
       "      <th>Poised Predictions RC</th>\n",
       "      <th>Active Predictions</th>\n",
       "      <th>Active Predictions RC</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>chr5</td>\n",
       "      <td>147894094</td>\n",
       "      <td>147894294</td>\n",
       "      <td>Merged-chr5-147894194-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.850177</td>\n",
       "      <td>0.580634</td>\n",
       "      <td>0.930514</td>\n",
       "      <td>0.886086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>chr15</td>\n",
       "      <td>53399224</td>\n",
       "      <td>53399424</td>\n",
       "      <td>Merged-chr15-53399324-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.843508</td>\n",
       "      <td>0.767881</td>\n",
       "      <td>0.864703</td>\n",
       "      <td>0.922945</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>chr12</td>\n",
       "      <td>111743421</td>\n",
       "      <td>111743621</td>\n",
       "      <td>Merged-chr12-111743521-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.922834</td>\n",
       "      <td>0.916620</td>\n",
       "      <td>0.924718</td>\n",
       "      <td>0.895533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>chr11</td>\n",
       "      <td>100582977</td>\n",
       "      <td>100583177</td>\n",
       "      <td>Merged-chr11-100583077-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.938389</td>\n",
       "      <td>0.930212</td>\n",
       "      <td>0.939787</td>\n",
       "      <td>0.963610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>chr11</td>\n",
       "      <td>98782374</td>\n",
       "      <td>98782574</td>\n",
       "      <td>Merged-chr11-98782474-1</td>\n",
       "      <td>0.0</td>\n",
       "      <td>+</td>\n",
       "      <td>0.059996</td>\n",
       "      <td>0.648414</td>\n",
       "      <td>0.830570</td>\n",
       "      <td>0.707561</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Chr      Start        End                    Peakid  Stat Strand  \\\n",
       "0   chr5  147894094  147894294   Merged-chr5-147894194-1   0.0      +   \n",
       "1  chr15   53399224   53399424   Merged-chr15-53399324-1   0.0      +   \n",
       "3  chr12  111743421  111743621  Merged-chr12-111743521-1   0.0      +   \n",
       "4  chr11  100582977  100583177  Merged-chr11-100583077-1   0.0      +   \n",
       "5  chr11   98782374   98782574   Merged-chr11-98782474-1   0.0      +   \n",
       "\n",
       "   Poised Predictions  Poised Predictions RC  Active Predictions  \\\n",
       "0            0.850177               0.580634            0.930514   \n",
       "1            0.843508               0.767881            0.864703   \n",
       "3            0.922834               0.916620            0.924718   \n",
       "4            0.938389               0.930212            0.939787   \n",
       "5            0.059996               0.648414            0.830570   \n",
       "\n",
       "   Active Predictions RC  \n",
       "0               0.886086  \n",
       "1               0.922945  \n",
       "3               0.895533  \n",
       "4               0.963610  \n",
       "5               0.707561  "
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_bed_filter_both.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {},
   "outputs": [],
   "source": [
    "#building an index for use in all, peak_ids (chr_start_end)\n",
    "#bed_df_both.head()\n",
    "\n",
    "peak_ids_poised = []\n",
    "merge_ids_poised = merged_bed_filter['Peakid'].tolist()\n",
    "for row in range(len(merged_bed_filter)):\n",
    "    peak = str(merged_bed_filter.iloc[row,0]) + \"_\" + str(merged_bed_filter.iloc[row,1]) + \"_\" + str(merged_bed_filter.iloc[row,2])\n",
    "    peak_ids_poised.append(peak)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "#building an index for use in all, peak_ids (chr_start_end)\n",
    "#bed_df_both.head()\n",
    "\n",
    "peak_ids_active = []\n",
    "merge_ids_active = merged_bed_filter_active['Peakid'].tolist()\n",
    "\n",
    "for row in range(len(merged_bed_filter_active)):\n",
    "    peak = str(merged_bed_filter_active.iloc[row,0]) + \"_\" + str(merged_bed_filter_active.iloc[row,1]) + \"_\" + str(merged_bed_filter_active.iloc[row,2])\n",
    "    peak_ids_active.append(peak)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [],
   "source": [
    "#building an index for use in all, peak_ids (chr_start_end)\n",
    "#merged_bed_filter_both.head()\n",
    "\n",
    "peak_ids_both = []\n",
    "merge_ids_both = merged_bed_filter_both['Peakid'].tolist()\n",
    "for row in range(len(merged_bed_filter_both)):\n",
    "    peak = str(merged_bed_filter_both.iloc[row,0]) + \"_\" + str(merged_bed_filter_both.iloc[row,1]) + \"_\" + str(merged_bed_filter_both.iloc[row,2])\n",
    "    peak_ids_both.append(peak)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "43537\n",
      "['chr5_147894094_147894294', 'chr15_53399224_53399424', 'chr10_95323530_95323730', 'chr12_111743421_111743621', 'chr11_100582977_100583177']\n",
      "37452\n",
      "['chr5_147894094_147894294', 'chr15_53399224_53399424', 'chr12_111743421_111743621', 'chr11_100582977_100583177', 'chr11_98782374_98782574']\n"
     ]
    }
   ],
   "source": [
    "# print(len(peak_ids_poised))\n",
    "# print(peak_ids_poised[:5])\n",
    "# print(merge_ids_poised[:5])\n",
    "print(len(peak_ids_active))\n",
    "print(peak_ids_active[:5])\n",
    "print(len(peak_ids_both))\n",
    "print(peak_ids_both[:5])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Write out scores"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "if not os.path.isdir('./region_scores/'):\n",
    "    os.mkdir('./region_scores/')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##  Poised Model\n",
    "- Scores 1 = poised pos data\n",
    "- Scores 2 = active pos data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Poised model, poised data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(38482, 300)"
      ]
     },
     "execution_count": 138,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "weights1 = []\n",
    "for sample_index in range(len(scores1)//2):\n",
    "    sample_scores1 = np.array([scores1[sample_index*2], scores1[sample_index*2+1][::-1]])\n",
    "    sample_score1 = np.array([sample_scores1[j,i] for i,j in enumerate(np.argmax(np.absolute(sample_scores1), axis=0))])\n",
    "    weights1.append(sample_score1)\n",
    "weights1 = np.array(weights1)\n",
    "weights1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(38482, 300)\n"
     ]
    }
   ],
   "source": [
    "#writing this to a file so can use in other notebooks\n",
    "#want to include some sort of index here too... maybe peak IDs?\n",
    "\n",
    "control_nt_weights1 = \"./region_scores/poised_model_poised_peak_weights.txt\"\n",
    "weights_df1 = pd.DataFrame(weights1)\n",
    "weights_df1.index = peak_ids_poised\n",
    "print(weights_df1.shape)\n",
    "#control_weights_df.head()\n",
    "weights_df1.to_csv(control_nt_weights1,index = True, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Poised model, both data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(37452, 300)"
      ]
     },
     "execution_count": 112,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "weights2 = []\n",
    "for sample_index in range(len(scores2)//2):\n",
    "    sample_scores2 = np.array([scores2[sample_index*2], scores2[sample_index*2+1][::-1]])\n",
    "    sample_score2 = np.array([sample_scores2[j,i] for i,j in enumerate(np.argmax(np.absolute(sample_scores2), axis=0))])\n",
    "    weights2.append(sample_score2)\n",
    "weights2 = np.array(weights2)\n",
    "weights2.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(37452, 300)\n"
     ]
    }
   ],
   "source": [
    "#writing this to a file so can use in other notebooks\n",
    "#want to include some sort of index here too... maybe peak IDs?\n",
    "\n",
    "control_nt_weights2 = \"./region_scores/poised_model_poised_active_peak_weights.txt\"\n",
    "weights_df2 = pd.DataFrame(weights2)\n",
    "weights_df2.index = peak_ids_both\n",
    "print(weights_df2.shape)\n",
    "#control_weights_df.head()\n",
    "weights_df2.to_csv(control_nt_weights2,index = True, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Active model\n",
    "- Scores 3 = Active pos data\n",
    "- Scores 4 = Active pos data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Active model, Active positional data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(43537, 300)"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "weights3 = []\n",
    "for sample_index in range(len(scores3)//2):\n",
    "    sample_scores3 = np.array([scores3[sample_index*2], scores3[sample_index*2+1][::-1]])\n",
    "    sample_score3 = np.array([sample_scores3[j,i] for i,j in enumerate(np.argmax(np.absolute(sample_scores3), axis=0))])\n",
    "    weights3.append(sample_score3)\n",
    "weights3 = np.array(weights3)\n",
    "weights3.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(43537, 300)\n"
     ]
    }
   ],
   "source": [
    "#writing this to a file so can use in other notebooks\n",
    "#want to include some sort of index here too... maybe peak IDs?\n",
    "\n",
    "diff_nt_weights3 = \"./region_scores/active_model_active_peak_weights.txt\"\n",
    "weights_df3 = pd.DataFrame(weights3)\n",
    "weights_df3.index = peak_ids_active\n",
    "print(weights_df3.shape)\n",
    "#control_weights_df.head()\n",
    "weights_df3.to_csv(diff_nt_weights3,index = True, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Active model, active + poised positional data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(37452, 300)"
      ]
     },
     "execution_count": 118,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "weights4 = []\n",
    "for sample_index in range(len(scores4)//2):\n",
    "    sample_scores4 = np.array([scores4[sample_index*2], scores4[sample_index*2+1][::-1]])\n",
    "    sample_score4 = np.array([sample_scores4[j,i] for i,j in enumerate(np.argmax(np.absolute(sample_scores4), axis=0))])\n",
    "    weights4.append(sample_score4)\n",
    "weights4 = np.array(weights4)\n",
    "weights4.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(37452, 300)\n"
     ]
    }
   ],
   "source": [
    "#writing this to a file so can use in other notebooks\n",
    "#want to include some sort of index here too... maybe peak IDs?\n",
    "\n",
    "diff_nt_weights4 = \"./region_scores/active_model_active_poised_peak_weights.txt\"\n",
    "weights_df4 = pd.DataFrame(weights4)\n",
    "weights_df4.index = peak_ids_both\n",
    "print(weights_df4.shape)\n",
    "# control_weights_df.head()\n",
    "weights_df4.to_csv(diff_nt_weights4,index = True, sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Smooth Weights"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(38482, 300)\n"
     ]
    }
   ],
   "source": [
    "proc_range = 1\n",
    "smooth_weights1 = np.array([[np.mean(np.abs(w[max(0, i-proc_range):min(len(w), i+proc_range+1)]))\n",
    "                         for i in range(len(w))] for w in weights1])\n",
    "print(smooth_weights1.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(37452, 300)\n"
     ]
    }
   ],
   "source": [
    "proc_range = 1\n",
    "smooth_weights2 = np.array([[np.mean(np.abs(w[max(0, i-proc_range):min(len(w), i+proc_range+1)]))\n",
    "                         for i in range(len(w))] for w in weights2])\n",
    "print(smooth_weights2.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(43537, 300)\n"
     ]
    }
   ],
   "source": [
    "proc_range = 1\n",
    "smooth_weights3 = np.array([[np.mean(np.abs(w[max(0, i-proc_range):min(len(w), i+proc_range+1)]))\n",
    "                         for i in range(len(w))] for w in weights3])\n",
    "print(smooth_weights3.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(37452, 300)\n"
     ]
    }
   ],
   "source": [
    "proc_range = 1\n",
    "smooth_weights4 = np.array([[np.mean(np.abs(w[max(0, i-proc_range):min(len(w), i+proc_range+1)]))\n",
    "                         for i in range(len(w))] for w in weights4])\n",
    "print(smooth_weights4.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Visualize importance scores for individual site\n",
    "\n",
    "For now we will just use score1 and score3 since they are the ones that were properly trained."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [],
   "source": [
    "# active_pos_data[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# both_pos_data[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "ename": "IndexError",
     "evalue": "index 0 is out of bounds for axis 0 with size 0",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-212-5d65b197461c>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0msns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mset\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstyle\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'white'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfont_scale\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1.5\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mplt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfigure\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfigsize\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m17\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;36m3.2\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m \u001b[0mw\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mabs\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msmooth_weights1\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marray\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpoised_pos_data\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m-\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0ms2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      7\u001b[0m \u001b[0mplt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscatter\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0marange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mw\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mw\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0ms\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m10\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mc\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'#3A90E3'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mzorder\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m10\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0mplt\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maxhline\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msort\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mw\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m240\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mc\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'orangered'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlinestyle\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'--'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlinewidth\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m3\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mIndexError\u001b[0m: index 0 is out of bounds for axis 0 with size 0"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Figure size 1224x230.4 with 0 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "s = 'chr5_147894044_147894344_C57'\n",
    "s2 = 'chr19_17276031_17276331_C57'\n",
    "\n",
    "sns.set(style='white', font_scale=1.5)\n",
    "plt.figure(figsize=(17,3.2))\n",
    "w = np.abs(smooth_weights1[np.array(poised_pos_data)[:,-1] == s2][0])\n",
    "plt.scatter(np.arange(len(w)), w, s=10, c='#3A90E3',zorder=10)\n",
    "plt.axhline(np.sort(w)[240], c='orangered', linestyle='--', linewidth=3)\n",
    "plt.xlim(0,300)\n",
    "plt.xlabel('position')\n",
    "plt.ylabel('abs. DeepLIFT score')\n",
    "sns.despine(top=True, bottom=True)\n",
    "plt.tight_layout()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# s = 'chr5_147894044_147894344_C57'\n",
    "\n",
    "# sns.set(style='white', font_scale=1.5)\n",
    "# plt.figure(figsize=(17,3.2))\n",
    "# w = np.abs(smooth_weights3[np.array(active_pos_data)[:,-1] == s][0])\n",
    "# plt.scatter(np.arange(len(w)), w, s=10, c='#3A90E3',zorder=10)\n",
    "# plt.axhline(np.sort(w)[240], c='orangered', linestyle='--', linewidth=3)\n",
    "# plt.xlim(0,300)\n",
    "# plt.xlabel('position')\n",
    "# plt.ylabel('abs. DeepLIFT score')\n",
    "# sns.despine(top=True, bottom=True)\n",
    "# plt.tight_layout()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "zs-deeplift",
   "language": "python",
   "name": "zs-deeplift"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
